<!DOCTYPE html>
<html >

<head>

  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <title>Clustering and differential analysis of CFU-E cells with different conditions</title>
  <meta name="description" content="“Clustering and differential analysis of CFU-E cells with different conditions”">
  <meta name="generator" content="bookdown 0.7 and GitBook 2.6.7">

  <meta property="og:title" content="Clustering and differential analysis of CFU-E cells with different conditions" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Clustering and differential analysis of CFU-E cells with different conditions" />
  
  
  

<meta name="author" content="Ronak Shah">



  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black">
  
  


<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />









<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="0.1" data-path=""><a href="#table-of-contents"><i class="fa fa-check"></i><b>0.1</b> Table of Contents</a></li>
<li class="chapter" data-level="0.2" data-path=""><a href="#summary-of-experiment"><i class="fa fa-check"></i><b>0.2</b> Summary of Experiment:</a></li>
<li class="chapter" data-level="0.3" data-path=""><a href="#exploring-data-with-deseq2"><i class="fa fa-check"></i><b>0.3</b> Exploring data with DESeq2</a><ul>
<li class="chapter" data-level="0.3.1" data-path=""><a href="#running-deseq2"><i class="fa fa-check"></i><b>0.3.1</b> Running DESeq2</a></li>
<li class="chapter" data-level="0.3.2" data-path=""><a href="#make-counts-table-to-explore-that-data"><i class="fa fa-check"></i><b>0.3.2</b> Make counts table to explore that data</a></li>
<li class="chapter" data-level="0.3.3" data-path=""><a href="#histograms-of-counts-per-gene"><i class="fa fa-check"></i><b>0.3.3</b> Histograms of counts per gene</a></li>
<li class="chapter" data-level="0.3.4" data-path=""><a href="#eliminate-undetected-genes"><i class="fa fa-check"></i><b>0.3.4</b> Eliminate undetected genes</a></li>
<li class="chapter" data-level="0.3.5" data-path=""><a href="#re-run-deseq2-normalization"><i class="fa fa-check"></i><b>0.3.5</b> Re-run DESeq2 &amp; Normalization</a></li>
<li class="chapter" data-level="0.3.6" data-path=""><a href="#count-variance-is-related-to-mean"><i class="fa fa-check"></i><b>0.3.6</b> Count variance is related to mean</a></li>
<li class="chapter" data-level="0.3.7" data-path=""><a href="#estimated-dispersion-for-each-gene"><i class="fa fa-check"></i><b>0.3.7</b> Estimated Dispersion for each gene</a></li>
<li class="chapter" data-level="0.3.8" data-path=""><a href="#pca-amd-heatmap-on-all-samples-using-reguralized-log-transformation"><i class="fa fa-check"></i><b>0.3.8</b> PCA amd heatmap on all samples using reguralized log transformation</a></li>
<li class="chapter" data-level="0.3.9" data-path=""><a href="#perform-differential-expression-call"><i class="fa fa-check"></i><b>0.3.9</b> Perform differential expression call</a></li>
</ul></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">“Clustering and differential analysis of CFU-E cells with different conditions”</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="header">
<h1 class="title">“Clustering and differential analysis of CFU-E cells with different conditions”</h1>
<h4 class="author"><em>Ronak Shah</em></h4>
<h4 class="date"><em>July 27th 2017</em></h4>
</div>
<div id="table-of-contents" class="section level2">
<h2><span class="header-section-number">0.1</span> Table of Contents</h2>
<ul>
<li><a href="#summary-of-experiment:">Summary of Experiment:</a></li>
<li><a href="#exploring-data-with-deseq2">Exploring data with DESeq2</a>
<ul>
<li><a href="#running-deseq2">Running DESeq2</a></li>
<li><a href="#make-counts-table-to-explore-that-data">Make counts table to explore that data</a></li>
<li><a href="#histograms-of-counts-per-gene">Histograms of counts per gene</a>
<ul>
<li><a href="#matrix">matrix</a></li>
<li><a href="#truncated-matrix">truncated matrix</a></li>
<li><a href="#box-plotdensity-plots">box-plotdensity-plots</a></li>
<li><a href="#scatter-plots">scatter-plots</a></li>
</ul></li>
<li><a href="#eliminate-undetected-genes">Eliminate undetected genes</a></li>
<li><a href="#-re-run-deseq2-&amp;-normalization">Re-run DESeq2 &amp; Normalization</a>
<ul>
<li><a href="#plot-normalized-data">Plot normalized data</a></li>
</ul></li>
<li><a href="#count-variance-is-related-to-mean">Count variance is related to mean</a></li>
<li><a href="#estimated-dispersion-for-each-gene">Estimated Dispersion for each gene</a></li>
<li><a href="#pca-amd-heatmap-on-all-samples-using-reguralized-log-transformation">PCA amd heatmap on all samples using reguralized log transformation</a>
<ul>
<li><a href="#pca-using-rlog">PCA using rlog</a></li>
<li><a href="#heatmap-of-count-matrix-for-first-50-genes">Heatmap of count matrix for first 50 genes</a></li>
<li><a href="#heatmap-of-sample-distance-based-on-rlog">Heatmap of sample distance based on rlog</a></li>
</ul></li>
<li><a href="#perform-differential-expression-call">Perform differential expression call</a>
<ul>
<li><a href="#set-standard-cutoff-and-enhanced-volcano-plots">Set standard cutoff and Enhanced Volcano Plots</a></li>
<li><a href="#untreated-(wt)-vs-epo-with-vehicle-(ev)">Untreated (WT) vs EPO with Vehicle (EV)</a></li>
<li><a href="#untreated-(wt)-vs-epo-with-hmgb1-(eh)">Untreated (WT) vs EPO with HMGB1 (EH)</a></li>
<li><a href="#untreated-(wt)-vs-hmgb1-(h)">Untreated (WT) vs HMGB1 (H)</a></li>
<li><a href="#epo-with-vehicle-(ev)-vs-epo-with-hmgb1-(eh)">EPO with Vehicle (EV) vs EPO with HMGB1 (EH)</a></li>
<li><a href="#epo-with-vehicle-(ev)-vs-hmgb1-(h)">EPO with Vehicle (EV) vs HMGB1 (H)</a></li>
<li><a href="#epo-with-hmgb1-(eh)-vs-hmgb1-(h)">EPO with HMGB1 (EH) vs HMGB1 (H)</a></li>
</ul></li>
</ul></li>
</ul>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Load libraries</span>
<span class="kw">set.seed</span>(<span class="dv">0</span>)
<span class="kw">source</span>(<span class="st">&quot;https://bioconductor.org/biocLite.R&quot;</span>)
<span class="kw">library</span>(<span class="st">&quot;DESeq2&quot;</span>)
<span class="kw">library</span>(<span class="st">&quot;RColorBrewer&quot;</span>)
<span class="kw">library</span>(<span class="st">&quot;vsn&quot;</span>)
<span class="kw">library</span>(<span class="st">&quot;pheatmap&quot;</span>)
<span class="kw">library</span>(<span class="st">&quot;EnhancedVolcano&quot;</span>)
<span class="kw">library</span>(<span class="st">&quot;gridExtra&quot;</span>)
<span class="kw">library</span>(<span class="st">&quot;grid&quot;</span>)
<span class="kw">library</span>(<span class="st">&quot;ggplot2&quot;</span>)
<span class="kw">library</span>(<span class="st">&quot;gplots&quot;</span>)
<span class="kw">library</span>(<span class="st">&quot;affy&quot;</span>)
<span class="kw">library</span>(<span class="st">&quot;reshape2&quot;</span>)
<span class="kw">library</span>(<span class="st">&quot;gProfileR&quot;</span>)
<span class="kw">library</span>(<span class="st">&quot;ComplexHeatmap&quot;</span>)</code></pre></div>
<hr />
</div>
<div id="summary-of-experiment" class="section level2">
<h2><span class="header-section-number">0.2</span> Summary of Experiment:</h2>
<ol style="list-style-type: decimal">
<li>Flow-sorted Colony-Forming Units of Erythroid (CFU-E) cells. The cells are then starved for six hours to stop the Erythropoietin (EPO). Then these cells are treated with:
<ul>
<li>EPO + Vehicle (Peripheral Blood Smear (PBS))</li>
<li>EPO + High mobility group box 1 (HMGB1) protein</li>
<li>HMGB1</li>
</ul></li>
<li><p>The treatment occurs for sixty minutes and then RNA was extracted. Three biological replicates are used for reproducibility.</p></li>
<li><p>Single-end stranded RNA-seq performed on wild-type EPO starved cells and treated with 1,2,3 as stated above.</p></li>
<li>Labels:
<ul>
<li>WT =&gt; Untreated</li>
<li>EV =&gt; EPO + Vehicle</li>
<li>EH =&gt; EPO + HMGB1</li>
<li>H =&gt; HMGB1</li>
<li>E1 =&gt; Samples from Patient 1</li>
<li>E2 =&gt; Samples from Patient 2</li>
<li>E3 =&gt; Samples from Patient 3</li>
</ul></li>
</ol>
</div>
<div id="exploring-data-with-deseq2" class="section level2">
<h2><span class="header-section-number">0.3</span> Exploring data with DESeq2</h2>
<!-- ### Basic DESeq2 setup -->
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">directory &lt;-<span class="st"> &quot;/Users/rshah22/Documents/Projects/RNAseq_Erythrocid/counts_analysis/&quot;</span>
files &lt;-<span class="st"> </span><span class="kw">read.table</span>(<span class="kw">paste</span>(directory, 
    <span class="st">&quot;deseq2_sampleinfo.txt&quot;</span>, <span class="dt">sep =</span> <span class="st">&quot;&quot;</span>), 
    <span class="dt">header =</span> T)
condition &lt;-<span class="st"> </span><span class="kw">read.csv</span>(<span class="kw">paste</span>(directory, 
    <span class="st">&quot;deseq2_phenotype.txt&quot;</span>, <span class="dt">sep =</span> <span class="st">&quot;&quot;</span>), 
    <span class="dt">header =</span> T)
sampleTable &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">sampleName =</span> files<span class="op">$</span>sampleName, 
    <span class="dt">fileName =</span> files<span class="op">$</span>fileName, 
    <span class="dt">condition.type =</span> condition<span class="op">$</span>condition, 
    <span class="dt">condition.treatment =</span> condition<span class="op">$</span>treatment, 
    <span class="dt">condition.experiment =</span> condition<span class="op">$</span>experiment)
ddsHTSeq &lt;-<span class="st"> </span><span class="kw">DESeqDataSetFromHTSeqCount</span>(<span class="dt">sampleTable =</span> sampleTable, 
    <span class="dt">directory =</span> directory, <span class="dt">design =</span> <span class="op">~</span>condition.type <span class="op">+</span><span class="st"> </span>
<span class="st">        </span>condition.experiment)
<span class="co"># ddsHTSeq</span>
ddsHTSeq<span class="op">$</span>condition.type &lt;-<span class="st"> </span><span class="kw">factor</span>(ddsHTSeq<span class="op">$</span>condition.type, 
    <span class="dt">levels =</span> <span class="kw">c</span>(<span class="st">&quot;WT&quot;</span>, <span class="st">&quot;EV&quot;</span>, <span class="st">&quot;EH&quot;</span>, 
        <span class="st">&quot;H&quot;</span>))
ddsHTSeq<span class="op">$</span>condition.experiment &lt;-<span class="st"> </span><span class="kw">factor</span>(ddsHTSeq<span class="op">$</span>condition.experiment, 
    <span class="dt">levels =</span> <span class="kw">c</span>(<span class="st">&quot;E1&quot;</span>, <span class="st">&quot;E2&quot;</span>, <span class="st">&quot;E3&quot;</span>))</code></pre></div>
<div id="running-deseq2" class="section level3">
<h3><span class="header-section-number">0.3.1</span> Running DESeq2</h3>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">dds &lt;-<span class="st"> </span><span class="kw">DESeq</span>(ddsHTSeq)</code></pre></div>
</div>
<div id="make-counts-table-to-explore-that-data" class="section level3">
<h3><span class="header-section-number">0.3.2</span> Make counts table to explore that data</h3>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">count.table &lt;-<span class="st"> </span><span class="kw">counts</span>(dds)
count.table &lt;-<span class="st"> </span>count.table[<span class="op">-</span><span class="kw">c</span>((<span class="kw">grep</span>(<span class="st">&quot;^HIST|RNA|^RP11&quot;</span>, 
    <span class="kw">rownames</span>(count.table)))), ]
stats.per.sample &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="kw">t</span>(<span class="kw">do.call</span>(cbind, 
    <span class="kw">lapply</span>(count.table, summary))))
stats.per.sample<span class="op">$</span>libsum &lt;-<span class="st"> </span><span class="kw">apply</span>(count.table, 
    <span class="dv">2</span>, sum)  ## libsum
stats.per.sample<span class="op">$</span>perc05 &lt;-<span class="st"> </span><span class="kw">apply</span>(count.table, 
    <span class="dv">2</span>, quantile, <span class="fl">0.05</span>)
stats.per.sample<span class="op">$</span>perc10 &lt;-<span class="st"> </span><span class="kw">apply</span>(count.table, 
    <span class="dv">2</span>, quantile, <span class="fl">0.1</span>)
stats.per.sample<span class="op">$</span>perc90 &lt;-<span class="st"> </span><span class="kw">apply</span>(count.table, 
    <span class="dv">2</span>, quantile, <span class="fl">0.9</span>)
stats.per.sample<span class="op">$</span>perc95 &lt;-<span class="st"> </span><span class="kw">apply</span>(count.table, 
    <span class="dv">2</span>, quantile, <span class="fl">0.95</span>)
stats.per.sample<span class="op">$</span>zeros &lt;-<span class="st"> </span><span class="kw">apply</span>(count.table <span class="op">==</span><span class="st"> </span>
<span class="st">    </span><span class="dv">0</span>, <span class="dv">2</span>, sum)
stats.per.sample<span class="op">$</span>percent.zeros &lt;-<span class="st"> </span><span class="dv">100</span> <span class="op">*</span><span class="st"> </span>
<span class="st">    </span>stats.per.sample<span class="op">$</span>zeros<span class="op">/</span><span class="kw">nrow</span>(count.table)
<span class="kw">kable</span>(stats.per.sample[<span class="kw">sample</span>(<span class="dv">1</span><span class="op">:</span><span class="kw">ncol</span>(count.table), 
    <span class="dt">size =</span> <span class="dv">10</span>), ], <span class="dt">caption =</span> <span class="st">&quot;**Table: statistics per sample. ** We only display a random selection of 10 samples. &quot;</span>)</code></pre></div>
<table>
<caption>(#tab:code block 4)<strong>Table: statistics per sample. </strong> We only display a random selection of 10 samples.</caption>
<thead>
<tr class="header">
<th></th>
<th align="right">Min.</th>
<th align="right">X1st.Qu.</th>
<th align="right">Median</th>
<th align="right">Mean</th>
<th align="right">X3rd.Qu.</th>
<th align="right">Max.</th>
<th align="right">libsum</th>
<th align="right">perc05</th>
<th align="right">perc10</th>
<th align="right">perc90</th>
<th align="right">perc95</th>
<th align="right">zeros</th>
<th align="right">percent.zeros</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>11</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">16025018</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">864.0</td>
<td align="right">1727.2</td>
<td align="right">24354</td>
<td align="right">54.92929</td>
</tr>
<tr class="even">
<td>3</td>
<td align="right">5</td>
<td align="right">5</td>
<td align="right">5</td>
<td align="right">5</td>
<td align="right">5</td>
<td align="right">5</td>
<td align="right">15796424</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">800.4</td>
<td align="right">1621.0</td>
<td align="right">24715</td>
<td align="right">55.74351</td>
</tr>
<tr class="odd">
<td>4</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">15273891</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">774.0</td>
<td align="right">1569.2</td>
<td align="right">24872</td>
<td align="right">56.09762</td>
</tr>
<tr class="even">
<td>6</td>
<td align="right">11</td>
<td align="right">11</td>
<td align="right">11</td>
<td align="right">11</td>
<td align="right">11</td>
<td align="right">11</td>
<td align="right">13073035</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">645.0</td>
<td align="right">1292.0</td>
<td align="right">24544</td>
<td align="right">55.35783</td>
</tr>
<tr class="odd">
<td>8</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">16006088</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">804.0</td>
<td align="right">1620.0</td>
<td align="right">24365</td>
<td align="right">54.95410</td>
</tr>
<tr class="even">
<td>2</td>
<td align="right">3</td>
<td align="right">3</td>
<td align="right">3</td>
<td align="right">3</td>
<td align="right">3</td>
<td align="right">3</td>
<td align="right">15509458</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">793.0</td>
<td align="right">1603.0</td>
<td align="right">24712</td>
<td align="right">55.73674</td>
</tr>
<tr class="odd">
<td>9</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">14905408</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">789.0</td>
<td align="right">1569.0</td>
<td align="right">24223</td>
<td align="right">54.63383</td>
</tr>
<tr class="even">
<td>5</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">16355877</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">811.4</td>
<td align="right">1628.2</td>
<td align="right">24136</td>
<td align="right">54.43760</td>
</tr>
<tr class="odd">
<td>12</td>
<td align="right">882</td>
<td align="right">882</td>
<td align="right">882</td>
<td align="right">882</td>
<td align="right">882</td>
<td align="right">882</td>
<td align="right">13745586</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">728.0</td>
<td align="right">1477.0</td>
<td align="right">24754</td>
<td align="right">55.83147</td>
</tr>
<tr class="even">
<td>7</td>
<td align="right">33</td>
<td align="right">33</td>
<td align="right">33</td>
<td align="right">33</td>
<td align="right">33</td>
<td align="right">33</td>
<td align="right">16044909</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">801.0</td>
<td align="right">1620.2</td>
<td align="right">24460</td>
<td align="right">55.16837</td>
</tr>
</tbody>
</table>
<hr />
<!--### Assign color based on sample type -->
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">col.sampletype &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dt">WT =</span> <span class="st">&quot;#9E0142&quot;</span>, 
    <span class="dt">EV =</span> <span class="st">&quot;#D0384D&quot;</span>, <span class="dt">EH =</span> <span class="st">&quot;#EE6445&quot;</span>, 
    <span class="dt">H =</span> <span class="st">&quot;#FA9C58&quot;</span>)
expDesign &lt;-<span class="st"> </span>condition
expDesign<span class="op">$</span>color &lt;-<span class="st"> </span>col.sampletype[<span class="kw">as.vector</span>(expDesign<span class="op">$</span>condition)]</code></pre></div>
</div>
<div id="histograms-of-counts-per-gene" class="section level3 tabset tabset-fade tabset-pills">
<h3><span class="header-section-number">0.3.3</span> Histograms of counts per gene</h3>
<div id="matrix" class="section level4">
<h4><span class="header-section-number">0.3.3.1</span> matrix</h4>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">plot.new</span>()
<span class="kw">hist</span>(<span class="kw">as.matrix</span>(count.table), <span class="dt">col =</span> <span class="st">&quot;blue&quot;</span>, 
    <span class="dt">border =</span> <span class="st">&quot;white&quot;</span>, <span class="dt">breaks =</span> <span class="dv">100</span>)</code></pre></div>
<p><img src="_main_files/figure-html/code%20block%206-1.png" width="672" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">invisible</span>(<span class="kw">dev.off</span>())</code></pre></div>
</div>
<div id="truncated-matrix" class="section level4">
<h4><span class="header-section-number">0.3.3.2</span> truncated matrix</h4>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">plot.new</span>()
<span class="kw">hist</span>(<span class="kw">as.matrix</span>(count.table), <span class="dt">col =</span> <span class="st">&quot;blue&quot;</span>, 
    <span class="dt">border =</span> <span class="st">&quot;white&quot;</span>, <span class="dt">breaks =</span> <span class="dv">20000</span>, 
    <span class="dt">xlim =</span> <span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">500</span>), <span class="dt">main =</span> <span class="st">&quot;Counts per gene&quot;</span>, 
    <span class="dt">xlab =</span> <span class="st">&quot;Counts (truncated axis)&quot;</span>, 
    <span class="dt">ylab =</span> <span class="st">&quot;Number of genes&quot;</span>, <span class="dt">las =</span> <span class="dv">1</span>, 
    <span class="dt">cex.axis =</span> <span class="fl">0.7</span>)</code></pre></div>
<p><img src="_main_files/figure-html/code%20block%207-1.png" width="672" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">invisible</span>(<span class="kw">dev.off</span>())</code></pre></div>
</div>
<div id="log2-transformation" class="section level4">
<h4><span class="header-section-number">0.3.3.3</span> log2-transformation</h4>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">epsilon &lt;-<span class="st"> </span><span class="dv">1</span>  <span class="co"># pseudo-count to avoid problems with log(0)</span>
<span class="kw">plot.new</span>()
<span class="kw">hist</span>(<span class="kw">as.matrix</span>(<span class="kw">log2</span>(count.table <span class="op">+</span><span class="st"> </span>
<span class="st">    </span>epsilon)), <span class="dt">breaks =</span> <span class="dv">100</span>, <span class="dt">col =</span> <span class="st">&quot;blue&quot;</span>, 
    <span class="dt">border =</span> <span class="st">&quot;white&quot;</span>, <span class="dt">main =</span> <span class="st">&quot;Log2-transformed counts per gene&quot;</span>, 
    <span class="dt">xlab =</span> <span class="st">&quot;log2(counts+1)&quot;</span>, <span class="dt">ylab =</span> <span class="st">&quot;Number of genes&quot;</span>, 
    <span class="dt">las =</span> <span class="dv">1</span>, <span class="dt">cex.axis =</span> <span class="fl">0.7</span>)</code></pre></div>
<p><img src="_main_files/figure-html/code%20block%208-1.png" width="672" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">invisible</span>(<span class="kw">dev.off</span>())</code></pre></div>
</div>
<div id="box-plot" class="section level4">
<h4><span class="header-section-number">0.3.3.4</span> box-plot</h4>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">plot.new</span>()
<span class="kw">boxplot</span>(<span class="kw">log2</span>(count.table <span class="op">+</span><span class="st"> </span>epsilon), 
    <span class="dt">col =</span> expDesign<span class="op">$</span>color, <span class="dt">pch =</span> <span class="st">&quot;.&quot;</span>, 
    <span class="dt">horizontal =</span> <span class="ot">TRUE</span>, <span class="dt">cex.axis =</span> <span class="fl">0.5</span>, 
    <span class="dt">las =</span> <span class="dv">1</span>, <span class="dt">ylab =</span> <span class="st">&quot;Samples&quot;</span>, 
    <span class="dt">xlab =</span> <span class="st">&quot;log2(Counts +1)&quot;</span>)</code></pre></div>
<p><img src="_main_files/figure-html/code%20block%209-1.png" width="672" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">invisible</span>(<span class="kw">dev.off</span>())</code></pre></div>
</div>
<div id="density-plots" class="section level4">
<h4><span class="header-section-number">0.3.3.5</span> density-plots</h4>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">plot.new</span>()
<span class="kw">plotDensity</span>(<span class="kw">log2</span>(count.table <span class="op">+</span><span class="st"> </span>
<span class="st">    </span>epsilon), <span class="dt">lty =</span> <span class="dv">1</span>, <span class="dt">col =</span> expDesign<span class="op">$</span>color, 
    <span class="dt">lwd =</span> <span class="dv">2</span>)
<span class="kw">grid</span>()
<span class="kw">legend</span>(<span class="st">&quot;topright&quot;</span>, <span class="dt">legend =</span> <span class="kw">names</span>(col.sampletype), 
    <span class="dt">col =</span> col.sampletype, <span class="dt">lwd =</span> <span class="dv">2</span>)</code></pre></div>
<p><img src="_main_files/figure-html/code%20block%2010-1.png" width="672" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">invisible</span>(<span class="kw">dev.off</span>())</code></pre></div>
</div>
<div id="scatter-plots" class="section level4">
<h4><span class="header-section-number">0.3.3.6</span> scatter-plots</h4>
<ul>
<li>all points are aligned along the diagonal, with a relatively wider dispersion at the bottom, corresponding to small number fluctuations.</li>
<li>comparing across samples types there are points discarding from the diagonal.</li>
</ul>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">plotFun &lt;-<span class="st"> </span><span class="cf">function</span>(x, y) {
    dns &lt;-<span class="st"> </span><span class="kw">densCols</span>(x, y)
    <span class="kw">points</span>(x, y, <span class="dt">col =</span> dns, <span class="dt">pch =</span> <span class="st">&quot;.&quot;</span>, 
        <span class="dt">panel.first =</span> <span class="kw">grid</span>())
    <span class="kw">abline</span>(<span class="dt">a =</span> <span class="dv">0</span>, <span class="dt">b =</span> <span class="dv">1</span>, <span class="dt">col =</span> <span class="st">&quot;brown&quot;</span>)
}
<span class="kw">plot.new</span>()
<span class="kw">pairs</span>(<span class="kw">log2</span>(count.table[, <span class="kw">sample</span>(<span class="kw">ncol</span>(count.table), 
    <span class="dv">12</span>)] <span class="op">+</span><span class="st"> </span>epsilon), <span class="dt">panel =</span> plotFun, 
    <span class="dt">lower.panel =</span> <span class="ot">NULL</span>)</code></pre></div>
<p><img src="_main_files/figure-html/code%20block%2011-1.png" width="672" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">invisible</span>(<span class="kw">dev.off</span>())</code></pre></div>
</div>
</div>
<div id="eliminate-undetected-genes" class="section level3">
<h3><span class="header-section-number">0.3.4</span> Eliminate undetected genes</h3>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">prop.null &lt;-<span class="st"> </span><span class="kw">apply</span>(count.table, 
    <span class="dv">2</span>, <span class="cf">function</span>(x) <span class="dv">100</span> <span class="op">*</span><span class="st"> </span><span class="kw">mean</span>(x <span class="op">==</span><span class="st"> </span>
<span class="st">        </span><span class="dv">0</span>))
<span class="kw">plot.new</span>()
<span class="kw">barplot</span>(prop.null, <span class="dt">main =</span> <span class="st">&quot;Percentage of null counts per sample&quot;</span>, 
    <span class="dt">horiz =</span> <span class="ot">TRUE</span>, <span class="dt">cex.names =</span> <span class="fl">0.5</span>, 
    <span class="dt">las =</span> <span class="dv">1</span>, <span class="dt">col =</span> expDesign<span class="op">$</span>color, 
    <span class="dt">ylab =</span> <span class="st">&quot;Samples&quot;</span>, <span class="dt">xlab =</span> <span class="st">&quot;% of null counts&quot;</span>)</code></pre></div>
<p><img src="_main_files/figure-html/code%20block%2012-1.png" width="672" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">invisible</span>(<span class="kw">dev.off</span>())
count.table &lt;-<span class="st"> </span>count.table[<span class="kw">rowSums</span>(count.table) <span class="op">&gt;</span><span class="st"> </span>
<span class="st">    </span><span class="dv">0</span>, ]</code></pre></div>
</div>
<div id="re-run-deseq2-normalization" class="section level3">
<h3><span class="header-section-number">0.3.5</span> Re-run DESeq2 &amp; Normalization</h3>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">dds0 &lt;-<span class="st"> </span><span class="kw">DESeqDataSetFromMatrix</span>(<span class="dt">countData =</span> count.table, 
    <span class="dt">colData =</span> expDesign, <span class="dt">design =</span> <span class="op">~</span>condition <span class="op">+</span><span class="st"> </span>
<span class="st">        </span>experiment)
dds.norm &lt;-<span class="st"> </span><span class="kw">estimateSizeFactors</span>(dds0)
<span class="co"># sizeFactors(dds.norm)</span></code></pre></div>
<div id="plot-normalized-data" class="section level4 tabset tabset-fade tabset-pills">
<h4><span class="header-section-number">0.3.5.1</span> Plot normalized data</h4>
<div id="box-plot-1" class="section level5">
<h5><span class="header-section-number">0.3.5.1.1</span> box-plot</h5>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">par</span>(<span class="dt">mfrow =</span> <span class="kw">c</span>(<span class="dv">1</span>, <span class="dv">2</span>), <span class="dt">cex.lab =</span> <span class="fl">0.7</span>)
<span class="kw">boxplot</span>(<span class="kw">log2</span>(<span class="kw">counts</span>(dds.norm) <span class="op">+</span><span class="st"> </span>
<span class="st">    </span>epsilon), <span class="dt">col =</span> col.sampletype, 
    <span class="dt">cex.axis =</span> <span class="fl">0.7</span>, <span class="dt">las =</span> <span class="dv">1</span>, <span class="dt">xlab =</span> <span class="st">&quot;log2(counts)&quot;</span>, 
    <span class="dt">horizontal =</span> <span class="ot">TRUE</span>, <span class="dt">main =</span> <span class="st">&quot;Raw counts&quot;</span>)
<span class="kw">boxplot</span>(<span class="kw">log2</span>(<span class="kw">counts</span>(dds.norm, <span class="dt">normalized =</span> <span class="ot">TRUE</span>) <span class="op">+</span><span class="st"> </span>
<span class="st">    </span>epsilon), <span class="dt">col =</span> col.sampletype, 
    <span class="dt">cex.axis =</span> <span class="fl">0.7</span>, <span class="dt">las =</span> <span class="dv">1</span>, <span class="dt">xlab =</span> <span class="st">&quot;log2(normalized counts)&quot;</span>, 
    <span class="dt">horizontal =</span> <span class="ot">TRUE</span>, <span class="dt">main =</span> <span class="st">&quot;Normalized counts&quot;</span>)</code></pre></div>
<p><img src="_main_files/figure-html/code%20block%2014-1.png" width="672" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">invisible</span>(<span class="kw">dev.off</span>())</code></pre></div>
</div>
<div id="density-plot" class="section level5">
<h5><span class="header-section-number">0.3.5.1.2</span> density-plot</h5>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">par</span>(<span class="dt">mfrow =</span> <span class="kw">c</span>(<span class="dv">1</span>, <span class="dv">2</span>), <span class="dt">cex.lab =</span> <span class="fl">0.7</span>)
<span class="kw">plotDensity</span>(<span class="kw">log2</span>(<span class="kw">counts</span>(dds.norm) <span class="op">+</span><span class="st"> </span>
<span class="st">    </span>epsilon), <span class="dt">col =</span> col.sampletype, 
    <span class="dt">xlab =</span> <span class="st">&quot;log2(counts)&quot;</span>, <span class="dt">cex.lab =</span> <span class="fl">0.7</span>)
<span class="kw">plotDensity</span>(<span class="kw">log2</span>(<span class="kw">counts</span>(dds.norm, 
    <span class="dt">normalized =</span> <span class="ot">TRUE</span>) <span class="op">+</span><span class="st"> </span>epsilon), 
    <span class="dt">col =</span> col.sampletype, <span class="dt">xlab =</span> <span class="st">&quot;log2(normalized counts)&quot;</span>, 
    <span class="dt">cex.lab =</span> <span class="fl">0.7</span>)</code></pre></div>
<p><img src="_main_files/figure-html/code%20block%2015-1.png" width="672" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">invisible</span>(<span class="kw">dev.off</span>())</code></pre></div>
</div>
</div>
</div>
<div id="count-variance-is-related-to-mean" class="section level3">
<h3><span class="header-section-number">0.3.6</span> Count variance is related to mean</h3>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">## Computing mean and variance
norm.counts &lt;-<span class="st"> </span><span class="kw">counts</span>(dds.norm, 
    <span class="dt">normalized =</span> <span class="ot">TRUE</span>)
mean.counts &lt;-<span class="st"> </span><span class="kw">rowMeans</span>(norm.counts)
variance.counts &lt;-<span class="st"> </span><span class="kw">apply</span>(norm.counts, 
    <span class="dv">1</span>, var)
## sum(mean.counts==0) # Number
## of completely undetected
## genes
norm.counts.stats &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">min =</span> <span class="kw">apply</span>(norm.counts, 
    <span class="dv">2</span>, min), <span class="dt">mean =</span> <span class="kw">apply</span>(norm.counts, 
    <span class="dv">2</span>, mean), <span class="dt">median =</span> <span class="kw">apply</span>(norm.counts, 
    <span class="dv">2</span>, median), <span class="dt">max =</span> <span class="kw">apply</span>(norm.counts, 
    <span class="dv">2</span>, max), <span class="dt">zeros =</span> <span class="kw">apply</span>(norm.counts <span class="op">==</span><span class="st"> </span>
<span class="st">    </span><span class="dv">0</span>, <span class="dv">2</span>, sum), <span class="dt">percent.zeros =</span> <span class="dv">100</span> <span class="op">*</span><span class="st"> </span>
<span class="st">    </span><span class="kw">apply</span>(norm.counts <span class="op">==</span><span class="st"> </span><span class="dv">0</span>, <span class="dv">2</span>, 
        sum)<span class="op">/</span><span class="kw">nrow</span>(norm.counts), 
    <span class="dt">perc05 =</span> <span class="kw">apply</span>(norm.counts, 
        <span class="dv">2</span>, quantile, <span class="fl">0.05</span>), <span class="dt">perc10 =</span> <span class="kw">apply</span>(norm.counts, 
        <span class="dv">2</span>, quantile, <span class="fl">0.1</span>), <span class="dt">perc90 =</span> <span class="kw">apply</span>(norm.counts, 
        <span class="dv">2</span>, quantile, <span class="fl">0.9</span>), <span class="dt">perc95 =</span> <span class="kw">apply</span>(norm.counts, 
        <span class="dv">2</span>, quantile, <span class="fl">0.95</span>))
<span class="kw">kable</span>(norm.counts.stats)</code></pre></div>
<table>
<thead>
<tr class="header">
<th></th>
<th align="right">min</th>
<th align="right">mean</th>
<th align="right">median</th>
<th align="right">max</th>
<th align="right">zeros</th>
<th align="right">percent.zeros</th>
<th align="right">perc05</th>
<th align="right">perc10</th>
<th align="right">perc90</th>
<th align="right">perc95</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>S01</td>
<td align="right">0</td>
<td align="right">573.3060</td>
<td align="right">10.42074</td>
<td align="right">302131.5</td>
<td align="right">6882</td>
<td align="right">26.11269</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1358.864</td>
<td align="right">2394.581</td>
</tr>
<tr class="even">
<td>S02</td>
<td align="right">0</td>
<td align="right">568.1465</td>
<td align="right">10.61987</td>
<td align="right">302188.5</td>
<td align="right">6730</td>
<td align="right">25.53595</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1333.856</td>
<td align="right">2346.992</td>
</tr>
<tr class="odd">
<td>S03</td>
<td align="right">0</td>
<td align="right">579.9790</td>
<td align="right">10.64411</td>
<td align="right">321006.9</td>
<td align="right">6733</td>
<td align="right">25.54733</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1346.963</td>
<td align="right">2362.604</td>
</tr>
<tr class="even">
<td>S04</td>
<td align="right">0</td>
<td align="right">581.0388</td>
<td align="right">10.02579</td>
<td align="right">350783.2</td>
<td align="right">6890</td>
<td align="right">26.14305</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1359.497</td>
<td align="right">2363.379</td>
</tr>
<tr class="odd">
<td>S05</td>
<td align="right">0</td>
<td align="right">572.3780</td>
<td align="right">11.98990</td>
<td align="right">374807.9</td>
<td align="right">6154</td>
<td align="right">23.35041</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1294.909</td>
<td align="right">2275.314</td>
</tr>
<tr class="even">
<td>S06</td>
<td align="right">0</td>
<td align="right">568.7111</td>
<td align="right">12.61162</td>
<td align="right">383578.0</td>
<td align="right">6562</td>
<td align="right">24.89850</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1278.360</td>
<td align="right">2241.773</td>
</tr>
<tr class="odd">
<td>S07</td>
<td align="right">0</td>
<td align="right">579.0621</td>
<td align="right">12.36500</td>
<td align="right">380738.5</td>
<td align="right">6478</td>
<td align="right">24.57978</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1304.032</td>
<td align="right">2315.395</td>
</tr>
<tr class="even">
<td>S08</td>
<td align="right">0</td>
<td align="right">584.3166</td>
<td align="right">11.54536</td>
<td align="right">366004.1</td>
<td align="right">6383</td>
<td align="right">24.21931</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1328.870</td>
<td align="right">2354.868</td>
</tr>
<tr class="odd">
<td>S09</td>
<td align="right">0</td>
<td align="right">543.7968</td>
<td align="right">12.49969</td>
<td align="right">260979.1</td>
<td align="right">6241</td>
<td align="right">23.68052</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1306.698</td>
<td align="right">2255.713</td>
</tr>
<tr class="even">
<td>S10</td>
<td align="right">0</td>
<td align="right">560.4016</td>
<td align="right">11.20687</td>
<td align="right">265525.5</td>
<td align="right">6653</td>
<td align="right">25.24379</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1335.656</td>
<td align="right">2337.041</td>
</tr>
<tr class="odd">
<td>S11</td>
<td align="right">0</td>
<td align="right">555.9397</td>
<td align="right">10.97169</td>
<td align="right">230312.2</td>
<td align="right">6372</td>
<td align="right">24.17758</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1360.489</td>
<td align="right">2350.684</td>
</tr>
<tr class="even">
<td>S12</td>
<td align="right">0</td>
<td align="right">567.6020</td>
<td align="right">10.88288</td>
<td align="right">264424.5</td>
<td align="right">6772</td>
<td align="right">25.69531</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1381.255</td>
<td align="right">2399.239</td>
</tr>
</tbody>
</table>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">## Mean and variance
## relationship
mean.var.col &lt;-<span class="st"> </span><span class="kw">densCols</span>(<span class="dt">x =</span> <span class="kw">log2</span>(mean.counts), 
    <span class="dt">y =</span> <span class="kw">log2</span>(variance.counts))
<span class="kw">plot.new</span>()
<span class="kw">plot</span>(<span class="dt">x =</span> <span class="kw">log2</span>(mean.counts), <span class="dt">y =</span> <span class="kw">log2</span>(variance.counts), 
    <span class="dt">pch =</span> <span class="dv">16</span>, <span class="dt">cex =</span> <span class="fl">0.5</span>, <span class="dt">col =</span> mean.var.col, 
    <span class="dt">main =</span> <span class="st">&quot;Mean-variance relationship&quot;</span>, 
    <span class="dt">xlab =</span> <span class="st">&quot;Mean log2(normalized counts) per gene&quot;</span>, 
    <span class="dt">ylab =</span> <span class="st">&quot;Variance of log2(normalized counts)&quot;</span>, 
    <span class="dt">panel.first =</span> <span class="kw">grid</span>())
<span class="kw">abline</span>(<span class="dt">a =</span> <span class="dv">0</span>, <span class="dt">b =</span> <span class="dv">1</span>, <span class="dt">col =</span> <span class="st">&quot;brown&quot;</span>)</code></pre></div>
<p><img src="_main_files/figure-html/code%20block%2016-1.png" width="672" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">invisible</span>(<span class="kw">dev.off</span>())</code></pre></div>
</div>
<div id="estimated-dispersion-for-each-gene" class="section level3">
<h3><span class="header-section-number">0.3.7</span> Estimated Dispersion for each gene</h3>
<ul>
<li>Shows the mean of normalized counts (x axis) and dispersion estimate for each genes</li>
</ul>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">## Performing estimation of
## dispersion parameter
dds.disp &lt;-<span class="st"> </span><span class="kw">estimateDispersions</span>(dds.norm)
<span class="kw">plotDispEsts</span>(dds.disp)</code></pre></div>
<p><img src="_main_files/figure-html/code%20block%2017-1.png" width="672" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">invisible</span>(<span class="kw">dev.off</span>())</code></pre></div>
</div>
<div id="pca-amd-heatmap-on-all-samples-using-reguralized-log-transformation" class="section level3 tabset tabset-fade tabset-pills">
<h3><span class="header-section-number">0.3.8</span> PCA amd heatmap on all samples using reguralized log transformation</h3>
<ul>
<li>Transforms the count data to the log2 scale in a way which minimizes differences between samples for rows with small counts, and which normalizes with respect to library size.</li>
</ul>
<div id="pca-using-rlog" class="section level4">
<h4><span class="header-section-number">0.3.8.1</span> PCA using rlog</h4>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">rld &lt;-<span class="st"> </span><span class="kw">rlog</span>(dds.disp, <span class="dt">blind =</span> <span class="ot">TRUE</span>)
hmcol &lt;-<span class="st"> </span><span class="kw">colorRampPalette</span>(<span class="kw">brewer.pal</span>(<span class="dv">11</span>, 
    <span class="st">&quot;Spectral&quot;</span>))(<span class="dv">12</span>)
<span class="kw">plotPCA</span>(rld, <span class="dt">ntop =</span> <span class="dv">5000</span>, <span class="dt">intgroup =</span> <span class="kw">c</span>(<span class="st">&quot;condition&quot;</span>, 
    <span class="st">&quot;experiment&quot;</span>)) <span class="op">+</span><span class="st"> </span><span class="kw">scale_color_manual</span>(<span class="dt">values =</span> hmcol)</code></pre></div>
<p><img src="_main_files/figure-html/code%20block%2018-1.png" width="672" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">invisible</span>(<span class="kw">dev.off</span>())</code></pre></div>
</div>
<div id="heatmap-of-count-matrix-for-first-50-genes" class="section level4">
<h4><span class="header-section-number">0.3.8.2</span> Heatmap of count matrix for first 50 genes</h4>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">select &lt;-<span class="st"> </span><span class="kw">order</span>(<span class="kw">rowMedians</span>(<span class="kw">counts</span>(dds.disp, 
    <span class="dt">normalized =</span> <span class="ot">TRUE</span>)), <span class="dt">decreasing =</span> <span class="ot">TRUE</span>)[<span class="dv">1</span><span class="op">:</span><span class="dv">50</span>]
df &lt;-<span class="st"> </span><span class="kw">as.data.frame</span>(<span class="kw">colData</span>(dds.disp)[, 
    <span class="kw">c</span>(<span class="st">&quot;condition&quot;</span>, <span class="st">&quot;experiment&quot;</span>)])
<span class="co"># Heatmap of count matrix</span>
<span class="kw">pheatmap</span>(<span class="kw">assay</span>(rld)[select, ], 
    <span class="dt">cluster_rows =</span> <span class="ot">TRUE</span>, <span class="dt">show_rownames =</span> <span class="ot">TRUE</span>, 
    <span class="dt">cluster_cols =</span> <span class="ot">TRUE</span>, <span class="dt">annotation_col =</span> df)</code></pre></div>
<p><img src="_main_files/figure-html/code%20block%2019-1.png" width="1152" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">invisible</span>(<span class="kw">dev.off</span>())</code></pre></div>
</div>
<div id="heatmap-of-sample-distance-based-on-rlog" class="section level4">
<h4><span class="header-section-number">0.3.8.3</span> Heatmap of sample distance based on rlog</h4>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">sampleDists &lt;-<span class="st"> </span><span class="kw">dist</span>(<span class="kw">t</span>(<span class="kw">assay</span>(rld)))
sampleDistMatrix &lt;-<span class="st"> </span><span class="kw">as.matrix</span>(sampleDists)
<span class="kw">rownames</span>(sampleDistMatrix) &lt;-<span class="st"> </span><span class="kw">paste</span>(rld<span class="op">$</span>condition, 
    rld<span class="op">$</span>experiment, <span class="dt">sep =</span> <span class="st">&quot;-&quot;</span>)
<span class="kw">colnames</span>(sampleDistMatrix) &lt;-<span class="st"> </span><span class="ot">NULL</span>
colors &lt;-<span class="st"> </span><span class="kw">colorRampPalette</span>(<span class="kw">rev</span>(<span class="kw">brewer.pal</span>(<span class="dv">9</span>, 
    <span class="st">&quot;Blues&quot;</span>)))(<span class="dv">255</span>)
<span class="kw">pheatmap</span>(sampleDistMatrix, <span class="dt">clustering_distance_rows =</span> sampleDists, 
    <span class="dt">clustering_distance_cols =</span> sampleDists, 
    <span class="dt">col =</span> colors)</code></pre></div>
<p><img src="_main_files/figure-html/code%20block%2020-1.png" width="672" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">invisible</span>(<span class="kw">dev.off</span>())</code></pre></div>
</div>
</div>
<div id="perform-differential-expression-call" class="section level3">
<h3><span class="header-section-number">0.3.9</span> Perform differential expression call</h3>
<div id="set-standard-cutoff-and-enhanced-volcano-plots" class="section level4">
<h4><span class="header-section-number">0.3.9.1</span> Set standard cutoff and Enhanced Volcano Plots</h4>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">p_cutoff =<span class="st"> </span><span class="fl">0.1</span>
fc_cutoff =<span class="st"> </span><span class="fl">0.5</span>
xlim &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dv">1</span>, <span class="fl">1e+05</span>)
ylim &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="op">-</span><span class="dv">6</span>, <span class="dv">6</span>)
wald.test &lt;-<span class="st"> </span><span class="kw">nbinomWaldTest</span>(dds.disp)
enhanced_volcano_plots &lt;-<span class="st"> </span><span class="cf">function</span>(title, 
    results, pvalue, foldchange) {
    p =<span class="st"> </span><span class="kw">EnhancedVolcano</span>(results, 
        <span class="dt">lab =</span> <span class="kw">rownames</span>(results), 
        <span class="dt">x =</span> <span class="st">&quot;log2FoldChange&quot;</span>, <span class="dt">y =</span> <span class="st">&quot;padj&quot;</span>, 
        <span class="dt">xlab =</span> <span class="kw">bquote</span>(<span class="op">~</span>Log[<span class="dv">2</span>] <span class="op">~</span><span class="st"> </span>
<span class="st">            &quot;fold change&quot;</span>), <span class="dt">ylab =</span> <span class="kw">bquote</span>(<span class="op">~-</span>Log[<span class="dv">10</span>] <span class="op">~</span><span class="st"> </span>
<span class="st">            </span>adjusted <span class="op">~</span><span class="st"> </span><span class="kw">italic</span>(P)), 
        <span class="dt">pCutoff =</span> pvalue, <span class="dt">FCcutoff =</span> foldchange, 
        <span class="dt">xlim =</span> <span class="kw">c</span>(<span class="op">-</span><span class="dv">6</span>, <span class="dv">6</span>), <span class="dt">transcriptLabSize =</span> <span class="dv">3</span>, 
        <span class="dt">title =</span> title, <span class="dt">colAlpha =</span> <span class="dv">1</span>, 
        <span class="dt">legend =</span> <span class="kw">c</span>(<span class="st">&quot;NS&quot;</span>, <span class="st">&quot;Log2 FC&quot;</span>, 
            <span class="st">&quot;Adjusted p-value&quot;</span>, 
            <span class="st">&quot;Adjusted p-value &amp; Log2 FC&quot;</span>), 
        <span class="dt">legendPosition =</span> <span class="st">&quot;bottom&quot;</span>, 
        <span class="dt">legendLabSize =</span> <span class="dv">10</span>, <span class="dt">legendIconSize =</span> <span class="dv">3</span>, 
        <span class="dt">DrawConnectors =</span> <span class="ot">TRUE</span>, 
        <span class="dt">widthConnectors =</span> <span class="fl">0.5</span>, 
        <span class="dt">colConnectors =</span> <span class="st">&quot;black&quot;</span>)
    <span class="kw">return</span>(p)
}
plotCounts_gg &lt;-<span class="st"> </span><span class="cf">function</span>(i, dds, 
    intgroup) {
    group &lt;-<span class="st"> </span><span class="cf">if</span> (<span class="kw">length</span>(intgroup) <span class="op">==</span><span class="st"> </span>
<span class="st">        </span><span class="dv">1</span>) {
        <span class="kw">colData</span>(dds)[[intgroup]]
    } <span class="cf">else</span> <span class="cf">if</span> (<span class="kw">length</span>(intgroup) <span class="op">==</span><span class="st"> </span>
<span class="st">        </span><span class="dv">2</span>) {
        lvls &lt;-<span class="st"> </span><span class="kw">as.vector</span>(<span class="kw">t</span>(<span class="kw">outer</span>(<span class="kw">levels</span>(<span class="kw">colData</span>(dds)[[intgroup[<span class="dv">1</span>]]]), 
            <span class="kw">levels</span>(<span class="kw">colData</span>(dds)[[intgroup[<span class="dv">2</span>]]]), 
            <span class="cf">function</span>(x, y) <span class="kw">paste</span>(x, 
                y, <span class="dt">sep =</span> <span class="st">&quot; : &quot;</span>))))
        <span class="kw">droplevels</span>(<span class="kw">factor</span>(<span class="kw">apply</span>(<span class="kw">as.data.frame</span>(<span class="kw">colData</span>(dds)[, 
            intgroup, <span class="dt">drop =</span> <span class="ot">FALSE</span>]), 
            <span class="dv">1</span>, paste, <span class="dt">collapse =</span> <span class="st">&quot; : &quot;</span>), 
            <span class="dt">levels =</span> lvls))
    } <span class="cf">else</span> {
        <span class="kw">factor</span>(<span class="kw">apply</span>(<span class="kw">as.data.frame</span>(<span class="kw">colData</span>(dds)[, 
            intgroup, <span class="dt">drop =</span> <span class="ot">FALSE</span>]), 
            <span class="dv">1</span>, paste, <span class="dt">collapse =</span> <span class="st">&quot; : &quot;</span>))
    }
    data &lt;-<span class="st"> </span><span class="kw">plotCounts</span>(dds, <span class="dt">gene =</span> i, 
        <span class="dt">intgroup =</span> intgroup, <span class="dt">returnData =</span> <span class="ot">TRUE</span>)
    data &lt;-<span class="st"> </span><span class="kw">cbind</span>(data, <span class="kw">data.frame</span>(<span class="dt">group =</span> group))
    main &lt;-<span class="st"> </span><span class="kw">rownames</span>(dds)[i]
    <span class="kw">ggplot</span>(data, <span class="kw">aes</span>(<span class="dt">x =</span> group, 
        <span class="dt">y =</span> count)) <span class="op">+</span><span class="st"> </span><span class="kw">geom_boxplot</span>() <span class="op">+</span><span class="st"> </span>
<span class="st">        </span><span class="kw">ylab</span>(<span class="st">&quot;Normalized count&quot;</span>) <span class="op">+</span><span class="st"> </span>
<span class="st">        </span><span class="kw">ggtitle</span>(main) <span class="op">+</span><span class="st"> </span><span class="kw">coord_trans</span>(<span class="dt">y =</span> <span class="st">&quot;log2&quot;</span>) <span class="op">+</span><span class="st"> </span>
<span class="st">        </span><span class="kw">scale_x_discrete</span>(<span class="dt">limits =</span> <span class="kw">c</span>(<span class="st">&quot;WT&quot;</span>, 
            <span class="st">&quot;EV&quot;</span>, <span class="st">&quot;EH&quot;</span>, <span class="st">&quot;H&quot;</span>))
}</code></pre></div>
</div>
<div id="untreated-wt-vs-epo-with-vehicle-ev" class="section level4">
<h4><span class="header-section-number">0.3.9.2</span> Untreated (WT) vs EPO with Vehicle (EV)</h4>
<div id="plots" class="section level5 tabset tabset-fade tabset-pills">
<h5><span class="header-section-number">0.3.9.2.1</span> Plots</h5>
<div id="ma-plot" class="section level6">
<h6><span class="header-section-number">0.3.9.2.1.1</span> MA-plot</h6>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">res_WT_EV &lt;-<span class="st"> </span><span class="kw">results</span>(wald.test, 
    <span class="dt">contrast =</span> <span class="kw">c</span>(<span class="st">&quot;condition&quot;</span>, <span class="st">&quot;WT&quot;</span>, 
        <span class="st">&quot;EV&quot;</span>), <span class="dt">alpha =</span> p_cutoff, 
    <span class="dt">pAdjustMethod =</span> <span class="st">&quot;BH&quot;</span>)
res_WT_EV_ashr &lt;-<span class="st"> </span><span class="kw">lfcShrink</span>(wald.test, 
    <span class="dt">contrast =</span> <span class="kw">c</span>(<span class="st">&quot;condition&quot;</span>, <span class="st">&quot;WT&quot;</span>, 
        <span class="st">&quot;EV&quot;</span>), <span class="dt">res =</span> res_WT_EV, 
    <span class="dt">type =</span> <span class="st">&quot;ashr&quot;</span>)
<span class="kw">par</span>(<span class="dt">mfrow =</span> <span class="kw">c</span>(<span class="dv">1</span>, <span class="dv">2</span>))
<span class="kw">plotMA</span>(res_WT_EV, <span class="dt">xlim =</span> xlim, 
    <span class="dt">ylim =</span> ylim, <span class="dt">main =</span> <span class="st">&quot;normal&quot;</span>)
<span class="kw">plotMA</span>(res_WT_EV_ashr, <span class="dt">xlim =</span> xlim, 
    <span class="dt">ylim =</span> ylim, <span class="dt">main =</span> <span class="st">&quot;ashr&quot;</span>)</code></pre></div>
<p><img src="_main_files/figure-html/code%20block%2022-1.png" width="672" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">invisible</span>(<span class="kw">dev.off</span>())</code></pre></div>
</div>
<div id="historgram-of-adjusted-p-values" class="section level6">
<h6><span class="header-section-number">0.3.9.2.1.2</span> Historgram of adjusted p-values</h6>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">hist</span>(res_WT_EV<span class="op">$</span>padj, <span class="dt">breaks =</span> <span class="dv">20</span>, 
    <span class="dt">col =</span> <span class="st">&quot;grey&quot;</span>, <span class="dt">main =</span> <span class="st">&quot;DESeq2 p-value distribution&quot;</span>, 
    <span class="dt">xlab =</span> <span class="st">&quot;DESeq2 P-value&quot;</span>, <span class="dt">ylab =</span> <span class="st">&quot;Number of genes&quot;</span>)</code></pre></div>
<p><img src="_main_files/figure-html/code%20block%2023-1.png" width="672" /></p>
</div>
<div id="volacano-plot" class="section level6">
<h6><span class="header-section-number">0.3.9.2.1.3</span> Volacano-plot</h6>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">resOrdered_WT_EV &lt;-<span class="st"> </span>res_WT_EV[<span class="kw">order</span>(res_WT_EV<span class="op">$</span>padj), 
    ]
<span class="kw">write.csv</span>(<span class="kw">as.data.frame</span>(resOrdered_WT_EV), 
    <span class="dt">file =</span> <span class="st">&quot;condition_WT_EV_alpha0.1_results.csv&quot;</span>)
WT_EV_p1 &lt;-<span class="st"> </span><span class="kw">enhanced_volcano_plots</span>(<span class="st">&quot;WT vs. EV (padj=0.05,log2fc=0.5)&quot;</span>, 
    res_WT_EV, <span class="fl">0.05</span>, fc_cutoff)
WT_EV_p2 &lt;-<span class="st"> </span><span class="kw">enhanced_volcano_plots</span>(<span class="st">&quot;WT vs. EV (padj=0.1,log2fc=0.5)&quot;</span>, 
    res_WT_EV, p_cutoff, fc_cutoff)
<span class="kw">grid.arrange</span>(WT_EV_p1, WT_EV_p2, 
    <span class="dt">nrow =</span> <span class="dv">2</span>, <span class="dt">ncol =</span> <span class="dv">1</span>)
<span class="kw">grid.rect</span>(<span class="dt">gp =</span> <span class="kw">gpar</span>(<span class="dt">fill =</span> <span class="ot">NA</span>))</code></pre></div>
<p><img src="_main_files/figure-html/code%20block%2024-1.png" width="960" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">invisible</span>(<span class="kw">dev.off</span>())</code></pre></div>
</div>
<div id="plot-counts" class="section level6">
<h6><span class="header-section-number">0.3.9.2.1.4</span> Plot counts</h6>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">nBestFeatures =<span class="st"> </span><span class="dv">20</span>
ord &lt;-<span class="st"> </span><span class="kw">order</span>(res_WT_EV<span class="op">$</span>padj, <span class="dt">decreasing =</span> <span class="ot">FALSE</span>)
<span class="cf">for</span> (i <span class="cf">in</span> <span class="kw">head</span>(ord, nBestFeatures)) {
    <span class="kw">print</span>(<span class="kw">plotCounts_gg</span>(i, <span class="dt">dds =</span> wald.test, 
        <span class="dt">intgroup =</span> <span class="kw">c</span>(<span class="st">&quot;condition&quot;</span>)))
}</code></pre></div>
<p><img src="_main_files/figure-html/code%20block%2025-1.png" width="672" /><img src="_main_files/figure-html/code%20block%2025-2.png" width="672" /><img src="_main_files/figure-html/code%20block%2025-3.png" width="672" /><img src="_main_files/figure-html/code%20block%2025-4.png" width="672" /><img src="_main_files/figure-html/code%20block%2025-5.png" width="672" /><img src="_main_files/figure-html/code%20block%2025-6.png" width="672" /><img src="_main_files/figure-html/code%20block%2025-7.png" width="672" /><img src="_main_files/figure-html/code%20block%2025-8.png" width="672" /><img src="_main_files/figure-html/code%20block%2025-9.png" width="672" /><img src="_main_files/figure-html/code%20block%2025-10.png" width="672" /><img src="_main_files/figure-html/code%20block%2025-11.png" width="672" /><img src="_main_files/figure-html/code%20block%2025-12.png" width="672" /><img src="_main_files/figure-html/code%20block%2025-13.png" width="672" /><img src="_main_files/figure-html/code%20block%2025-14.png" width="672" /><img src="_main_files/figure-html/code%20block%2025-15.png" width="672" /><img src="_main_files/figure-html/code%20block%2025-16.png" width="672" /><img src="_main_files/figure-html/code%20block%2025-17.png" width="672" /><img src="_main_files/figure-html/code%20block%2025-18.png" width="672" /><img src="_main_files/figure-html/code%20block%2025-19.png" width="672" /><img src="_main_files/figure-html/code%20block%2025-20.png" width="672" /></p>
</div>
</div>
<div id="select-genes-based-on-fdr-and-make-heatmap" class="section level5">
<h5><span class="header-section-number">0.3.9.2.2</span> Select genes based on FDR and make heatmap</h5>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">  gene.kept &lt;-<span class="st"> </span><span class="kw">rownames</span>(res_WT_EV)[res_WT_EV<span class="op">$</span>padj <span class="op">&lt;=</span><span class="st"> </span>p_cutoff <span class="op">&amp;</span><span class="st"> </span><span class="op">!</span><span class="kw">is.na</span>(res_WT_EV<span class="op">$</span>padj) <span class="op">&amp;</span><span class="st"> </span>( res_WT_EV<span class="op">$</span>log2FoldChange <span class="op">&lt;=</span><span class="st"> </span><span class="op">-</span><span class="fl">0.5</span> <span class="op">|</span><span class="st"> </span>res_WT_EV<span class="op">$</span>log2FoldChange <span class="op">&gt;=</span><span class="st"> </span><span class="fl">0.5</span>)]
  count.table.kept &lt;-<span class="st"> </span><span class="kw">log2</span>(count.table <span class="op">+</span><span class="st"> </span>epsilon)[gene.kept, ]
  <span class="kw">heatmap.2</span>(<span class="kw">as.matrix</span>(count.table.kept), 
            <span class="dt">scale=</span><span class="st">&quot;row&quot;</span>, 
            <span class="dt">hclust=</span><span class="cf">function</span>(x) <span class="kw">hclust</span>(x,<span class="dt">method=</span><span class="st">&quot;average&quot;</span>), 
            <span class="dt">distfun=</span><span class="cf">function</span>(x) <span class="kw">as.dist</span>((<span class="dv">1</span><span class="op">-</span><span class="kw">cor</span>(<span class="kw">t</span>(x)))<span class="op">/</span><span class="dv">2</span>), 
            <span class="dt">trace=</span><span class="st">&quot;none&quot;</span>, 
            <span class="dt">density=</span><span class="st">&quot;none&quot;</span>, 
            <span class="co">#labRow=&quot;&quot;,</span>
            <span class="dt">cexCol=</span><span class="fl">0.7</span>)</code></pre></div>
<p><img src="_main_files/figure-html/code%20black%2026-1.png" width="960" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">  <span class="kw">invisible</span>(<span class="kw">dev.off</span>())</code></pre></div>
</div>
<div id="do-functional-enrichment" class="section level5">
<h5><span class="header-section-number">0.3.9.2.3</span> Do functional enrichment</h5>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">res_WT_EV.df &lt;-<span class="st"> </span><span class="kw">na.omit</span>(<span class="kw">data.frame</span>(res_WT_EV))
induced.sign &lt;-<span class="st"> </span><span class="kw">rownames</span>(res_WT_EV.df)[res_WT_EV.df<span class="op">$</span>log2FoldChange <span class="op">&gt;=</span><span class="st"> </span>
<span class="st">    </span><span class="fl">0.5</span> <span class="op">&amp;</span><span class="st"> </span>res_WT_EV.df<span class="op">$</span>padj <span class="op">&lt;</span><span class="st"> </span>p_cutoff]
<span class="co"># head(induced.sign)</span>
<span class="co"># names(term.induced)</span>
<span class="cf">if</span> (<span class="kw">identical</span>(induced.sign, <span class="kw">character</span>(<span class="dv">0</span>))) {
    <span class="kw">cat</span>(<span class="st">&quot;No genes found that have induced expression&quot;</span>)
} <span class="cf">else</span> {
    term.induced &lt;-<span class="st"> </span><span class="kw">gprofiler</span>(<span class="dt">query =</span> induced.sign, 
        <span class="dt">organism =</span> <span class="st">&quot;hsapiens&quot;</span>)
    term.induced &lt;-<span class="st"> </span>term.induced[<span class="kw">order</span>(term.induced<span class="op">$</span>p.value), 
        ]
    <span class="co"># term.induced$p.value</span>
    <span class="kw">kable</span>(term.induced[<span class="dv">1</span><span class="op">:</span><span class="dv">10</span>, <span class="kw">c</span>(<span class="st">&quot;term.name&quot;</span>, 
        <span class="st">&quot;term.size&quot;</span>, <span class="st">&quot;query.size&quot;</span>, 
        <span class="st">&quot;overlap.size&quot;</span>, <span class="st">&quot;recall&quot;</span>, 
        <span class="st">&quot;precision&quot;</span>, <span class="st">&quot;p.value&quot;</span>, 
        <span class="st">&quot;intersection&quot;</span>)], <span class="dt">format.args =</span> <span class="kw">c</span>(<span class="dt">engeneer =</span> <span class="ot">TRUE</span>, 
        <span class="dt">digits =</span> <span class="dv">3</span>), <span class="dt">caption =</span> <span class="st">&quot;**Table: induced gene functional analysis wit gProfileR. ** &quot;</span>)
}</code></pre></div>
<pre><code>## No genes found that have induced expression</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">repressed.sign &lt;-<span class="st"> </span><span class="kw">rownames</span>(res_WT_EV.df)[res_WT_EV.df<span class="op">$</span>log2FoldChange <span class="op">&lt;=</span><span class="st"> </span>
<span class="st">    </span><span class="op">-</span><span class="fl">0.5</span> <span class="op">&amp;</span><span class="st"> </span>res_WT_EV.df<span class="op">$</span>padj <span class="op">&lt;</span><span class="st"> </span>
<span class="st">    </span>p_cutoff]
<span class="cf">if</span> (<span class="kw">identical</span>(repressed.sign, <span class="kw">character</span>(<span class="dv">0</span>))) {
    <span class="kw">cat</span>(<span class="st">&quot;No genes found that have repressed expression&quot;</span>)
} <span class="cf">else</span> {
    term.repressed &lt;-<span class="st"> </span><span class="kw">gprofiler</span>(<span class="dt">query =</span> repressed.sign, 
        <span class="dt">organism =</span> <span class="st">&quot;hsapiens&quot;</span>)
    term.repressed &lt;-<span class="st"> </span>term.repressed[<span class="kw">order</span>(term.repressed<span class="op">$</span>p.value), 
        ]
    <span class="kw">kable</span>(term.repressed[<span class="dv">1</span><span class="op">:</span><span class="dv">10</span>, 
        <span class="kw">c</span>(<span class="st">&quot;term.name&quot;</span>, <span class="st">&quot;term.size&quot;</span>, 
            <span class="st">&quot;query.size&quot;</span>, <span class="st">&quot;overlap.size&quot;</span>, 
            <span class="st">&quot;recall&quot;</span>, <span class="st">&quot;precision&quot;</span>, 
            <span class="st">&quot;p.value&quot;</span>, <span class="st">&quot;intersection&quot;</span>)], 
        <span class="dt">format.args =</span> <span class="kw">c</span>(<span class="dt">engeneer =</span> <span class="ot">TRUE</span>, 
            <span class="dt">digits =</span> <span class="dv">3</span>), <span class="dt">caption =</span> <span class="st">&quot;**Table: repressed genes functional analysis with gProfileR. ** &quot;</span>)
}</code></pre></div>
<table>
<caption>(#tab:code block 27)<strong>Table: repressed genes functional analysis with gProfileR. </strong></caption>
<thead>
<tr class="header">
<th></th>
<th align="left">term.name</th>
<th align="right">term.size</th>
<th align="right">query.size</th>
<th align="right">overlap.size</th>
<th align="right">recall</th>
<th align="right">precision</th>
<th align="right">p.value</th>
<th align="left">intersection</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>13</td>
<td align="left">Jak-STAT signaling pathway</td>
<td align="right">162</td>
<td align="right">22</td>
<td align="right">8</td>
<td align="right">0.049</td>
<td align="right">0.364</td>
<td align="right">7.00e-07</td>
<td align="left">IL4R,OSM,CISH,SOCS2,CDKN1A,MYC,PIM1,SOCS3</td>
</tr>
<tr class="even">
<td>17</td>
<td align="left">Signaling by Interleukins</td>
<td align="right">463</td>
<td align="right">21</td>
<td align="right">10</td>
<td align="right">0.022</td>
<td align="right">0.476</td>
<td align="right">1.60e-06</td>
<td align="left">IL4R,OSM,CISH,SOCS2,CD80,CDKN1A,MYC,PIM1,SOCS3,IRS2</td>
</tr>
<tr class="odd">
<td>15</td>
<td align="left">Cytokine Signaling in Immune system</td>
<td align="right">685</td>
<td align="right">21</td>
<td align="right">11</td>
<td align="right">0.016</td>
<td align="right">0.524</td>
<td align="right">4.60e-06</td>
<td align="left">IL4R,OSM,CISH,EGR1,SOCS2,CD80,CDKN1A,MYC,PIM1,SOCS3,IRS2</td>
</tr>
<tr class="even">
<td>18</td>
<td align="left">Interleukin-4 and 13 signaling</td>
<td align="right">111</td>
<td align="right">21</td>
<td align="right">6</td>
<td align="right">0.054</td>
<td align="right">0.286</td>
<td align="right">1.68e-05</td>
<td align="left">IL4R,OSM,CDKN1A,MYC,PIM1,SOCS3</td>
</tr>
<tr class="odd">
<td>16</td>
<td align="left">Growth hormone receptor signaling</td>
<td align="right">24</td>
<td align="right">21</td>
<td align="right">4</td>
<td align="right">0.167</td>
<td align="right">0.190</td>
<td align="right">3.57e-05</td>
<td align="left">CISH,SOCS2,SOCS3,IRS2</td>
</tr>
<tr class="even">
<td>8</td>
<td align="left">response to insulin</td>
<td align="right">255</td>
<td align="right">37</td>
<td align="right">7</td>
<td align="right">0.027</td>
<td align="right">0.189</td>
<td align="right">1.06e-03</td>
<td align="left">CISH,EGR1,SOCS2,MYC,SOCS3,IRS2,PPARA</td>
</tr>
<tr class="odd">
<td>14</td>
<td align="left">Immune System</td>
<td align="right">2010</td>
<td align="right">21</td>
<td align="right">13</td>
<td align="right">0.006</td>
<td align="right">0.619</td>
<td align="right">5.50e-03</td>
<td align="left">IL4R,OSM,CISH,EGR1,SOCS2,CD80,CDKN1A,MYC,PIM1,MB21D1,HSPA6,SOCS3,IRS2</td>
</tr>
<tr class="even">
<td>1</td>
<td align="left">response to oxygen-containing compound</td>
<td align="right">1555</td>
<td align="right">37</td>
<td align="right">13</td>
<td align="right">0.008</td>
<td align="right">0.351</td>
<td align="right">7.17e-03</td>
<td align="left">CPEB4,CISH,KLF9,EGR1,SOCS2,CDKN1A,MYC,PIM1,AQP3,RARG,SOCS3,IRS2,PPARA</td>
</tr>
<tr class="odd">
<td>20</td>
<td align="left">TFAP2 (AP-2) family regulates transcription of cell cycle factors</td>
<td align="right">5</td>
<td align="right">21</td>
<td align="right">2</td>
<td align="right">0.400</td>
<td align="right">0.095</td>
<td align="right">1.11e-02</td>
<td align="left">CDKN1A,MYC</td>
</tr>
<tr class="even">
<td>19</td>
<td align="left">Interleukin-7 signaling</td>
<td align="right">36</td>
<td align="right">21</td>
<td align="right">3</td>
<td align="right">0.083</td>
<td align="right">0.143</td>
<td align="right">1.38e-02</td>
<td align="left">CISH,SOCS2,IRS2</td>
</tr>
</tbody>
</table>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># kable(head(term.induced[,c(&#39;p.value&#39;,</span>
<span class="co"># &#39;term.name&#39;,&#39;intersection&#39;)],</span>
<span class="co"># 10))</span></code></pre></div>
</div>
</div>
<div id="untreated-wt-vs-epo-with-hmgb1-eh" class="section level4">
<h4><span class="header-section-number">0.3.9.3</span> Untreated (WT) vs EPO with HMGB1 (EH)</h4>
<div id="plots-1" class="section level5 tabset tabset-fade tabset-pills">
<h5><span class="header-section-number">0.3.9.3.1</span> Plots</h5>
<div id="ma-plot-1" class="section level6">
<h6><span class="header-section-number">0.3.9.3.1.1</span> MA-plot</h6>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">res_WT_EH &lt;-<span class="st"> </span><span class="kw">results</span>(wald.test, 
    <span class="dt">contrast =</span> <span class="kw">c</span>(<span class="st">&quot;condition&quot;</span>, <span class="st">&quot;WT&quot;</span>, 
        <span class="st">&quot;EH&quot;</span>), <span class="dt">alpha =</span> p_cutoff, 
    <span class="dt">pAdjustMethod =</span> <span class="st">&quot;BH&quot;</span>)
res_WT_EH_ashr &lt;-<span class="st"> </span><span class="kw">lfcShrink</span>(wald.test, 
    <span class="dt">contrast =</span> <span class="kw">c</span>(<span class="st">&quot;condition&quot;</span>, <span class="st">&quot;WT&quot;</span>, 
        <span class="st">&quot;EH&quot;</span>), <span class="dt">res =</span> res_WT_EV, 
    <span class="dt">type =</span> <span class="st">&quot;ashr&quot;</span>)
<span class="kw">par</span>(<span class="dt">mfrow =</span> <span class="kw">c</span>(<span class="dv">1</span>, <span class="dv">2</span>))
<span class="kw">plotMA</span>(res_WT_EH, <span class="dt">xlim =</span> xlim, 
    <span class="dt">ylim =</span> ylim, <span class="dt">main =</span> <span class="st">&quot;normal&quot;</span>)
<span class="kw">plotMA</span>(res_WT_EH_ashr, <span class="dt">xlim =</span> xlim, 
    <span class="dt">ylim =</span> ylim, <span class="dt">main =</span> <span class="st">&quot;ashr&quot;</span>)</code></pre></div>
<p><img src="_main_files/figure-html/code%20block%2028-1.png" width="672" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">invisible</span>(<span class="kw">dev.off</span>())</code></pre></div>
</div>
<div id="historgram-of-adjusted-p-values-1" class="section level6">
<h6><span class="header-section-number">0.3.9.3.1.2</span> Historgram of adjusted p-values</h6>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">hist</span>(res_WT_EH<span class="op">$</span>padj, <span class="dt">breaks =</span> <span class="dv">20</span>, 
    <span class="dt">col =</span> <span class="st">&quot;grey&quot;</span>, <span class="dt">main =</span> <span class="st">&quot;DESeq2 p-value distribution&quot;</span>, 
    <span class="dt">xlab =</span> <span class="st">&quot;DESeq2 P-value&quot;</span>, <span class="dt">ylab =</span> <span class="st">&quot;Number of genes&quot;</span>)</code></pre></div>
<p><img src="_main_files/figure-html/code%20block%2029-1.png" width="672" /></p>
</div>
<div id="volacano-plot-1" class="section level6">
<h6><span class="header-section-number">0.3.9.3.1.3</span> Volacano-plot</h6>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">resOrdered_WT_EH &lt;-<span class="st"> </span>res_WT_EH[<span class="kw">order</span>(res_WT_EH<span class="op">$</span>padj), 
    ]
<span class="kw">write.csv</span>(<span class="kw">as.data.frame</span>(resOrdered_WT_EH), 
    <span class="dt">file =</span> <span class="st">&quot;condition_WT_EH_alpha0.1_results.csv&quot;</span>)
WT_EH_p1 &lt;-<span class="st"> </span><span class="kw">enhanced_volcano_plots</span>(<span class="st">&quot;WT vs. EH (padj=0.05,log2fc=0.5)&quot;</span>, 
    res_WT_EH, <span class="fl">0.05</span>, fc_cutoff)
WT_EH_p2 &lt;-<span class="st"> </span><span class="kw">enhanced_volcano_plots</span>(<span class="st">&quot;WT vs. EH (padj=0.1,log2fc=0.5)&quot;</span>, 
    res_WT_EH, p_cutoff, fc_cutoff)
<span class="kw">grid.arrange</span>(WT_EH_p1, WT_EH_p2, 
    <span class="dt">nrow =</span> <span class="dv">2</span>, <span class="dt">ncol =</span> <span class="dv">1</span>)
<span class="kw">grid.rect</span>(<span class="dt">gp =</span> <span class="kw">gpar</span>(<span class="dt">fill =</span> <span class="ot">NA</span>))</code></pre></div>
<p><img src="_main_files/figure-html/code%20block%2030-1.png" width="960" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">invisible</span>(<span class="kw">dev.off</span>())</code></pre></div>
</div>
<div id="plot-counts-1" class="section level6">
<h6><span class="header-section-number">0.3.9.3.1.4</span> Plot counts</h6>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">nBestFeatures =<span class="st"> </span><span class="dv">20</span>
ord &lt;-<span class="st"> </span><span class="kw">order</span>(res_WT_EH<span class="op">$</span>padj, <span class="dt">decreasing =</span> <span class="ot">FALSE</span>)
<span class="cf">for</span> (i <span class="cf">in</span> <span class="kw">head</span>(ord, nBestFeatures)) {
    <span class="kw">print</span>(<span class="kw">plotCounts_gg</span>(i, <span class="dt">dds =</span> wald.test, 
        <span class="dt">intgroup =</span> <span class="kw">c</span>(<span class="st">&quot;condition&quot;</span>)))
}</code></pre></div>
<p><img src="_main_files/figure-html/code%20block%2031-1.png" width="672" /><img src="_main_files/figure-html/code%20block%2031-2.png" width="672" /><img src="_main_files/figure-html/code%20block%2031-3.png" width="672" /><img src="_main_files/figure-html/code%20block%2031-4.png" width="672" /><img src="_main_files/figure-html/code%20block%2031-5.png" width="672" /><img src="_main_files/figure-html/code%20block%2031-6.png" width="672" /><img src="_main_files/figure-html/code%20block%2031-7.png" width="672" /><img src="_main_files/figure-html/code%20block%2031-8.png" width="672" /><img src="_main_files/figure-html/code%20block%2031-9.png" width="672" /><img src="_main_files/figure-html/code%20block%2031-10.png" width="672" /><img src="_main_files/figure-html/code%20block%2031-11.png" width="672" /><img src="_main_files/figure-html/code%20block%2031-12.png" width="672" /><img src="_main_files/figure-html/code%20block%2031-13.png" width="672" /><img src="_main_files/figure-html/code%20block%2031-14.png" width="672" /><img src="_main_files/figure-html/code%20block%2031-15.png" width="672" /><img src="_main_files/figure-html/code%20block%2031-16.png" width="672" /><img src="_main_files/figure-html/code%20block%2031-17.png" width="672" /><img src="_main_files/figure-html/code%20block%2031-18.png" width="672" /><img src="_main_files/figure-html/code%20block%2031-19.png" width="672" /><img src="_main_files/figure-html/code%20block%2031-20.png" width="672" /></p>
</div>
</div>
<div id="select-genes-based-on-fdr-and-make-heatmap-1" class="section level5">
<h5><span class="header-section-number">0.3.9.3.2</span> Select genes based on FDR and make heatmap</h5>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">  gene.kept &lt;-<span class="st"> </span><span class="kw">rownames</span>(res_WT_EH)[res_WT_EH<span class="op">$</span>padj <span class="op">&lt;=</span><span class="st"> </span>p_cutoff <span class="op">&amp;</span><span class="st"> </span><span class="op">!</span><span class="kw">is.na</span>(res_WT_EH<span class="op">$</span>padj) <span class="op">&amp;</span><span class="st"> </span>( res_WT_EH<span class="op">$</span>log2FoldChange <span class="op">&lt;=</span><span class="st"> </span><span class="op">-</span><span class="fl">0.5</span> <span class="op">|</span><span class="st"> </span>res_WT_EH<span class="op">$</span>log2FoldChange <span class="op">&gt;=</span><span class="st"> </span><span class="fl">0.5</span>)]
  count.table.kept &lt;-<span class="st"> </span><span class="kw">log2</span>(count.table <span class="op">+</span><span class="st"> </span>epsilon)[gene.kept, ]
  <span class="kw">heatmap.2</span>(<span class="kw">as.matrix</span>(count.table.kept), 
            <span class="dt">scale=</span><span class="st">&quot;row&quot;</span>, 
            <span class="dt">hclust=</span><span class="cf">function</span>(x) <span class="kw">hclust</span>(x,<span class="dt">method=</span><span class="st">&quot;average&quot;</span>), 
            <span class="dt">distfun=</span><span class="cf">function</span>(x) <span class="kw">as.dist</span>((<span class="dv">1</span><span class="op">-</span><span class="kw">cor</span>(<span class="kw">t</span>(x)))<span class="op">/</span><span class="dv">2</span>), 
            <span class="dt">trace=</span><span class="st">&quot;none&quot;</span>, 
            <span class="dt">density=</span><span class="st">&quot;none&quot;</span>, 
            <span class="co">#labRow=&quot;&quot;,</span>
            <span class="dt">cexCol=</span><span class="fl">0.7</span>)</code></pre></div>
<p><img src="_main_files/figure-html/code%20black%2032-1.png" width="960" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">  <span class="kw">invisible</span>(<span class="kw">dev.off</span>())</code></pre></div>
</div>
<div id="do-functional-enrichment-1" class="section level5">
<h5><span class="header-section-number">0.3.9.3.3</span> Do functional enrichment</h5>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">res_WT_EH.df &lt;-<span class="st"> </span><span class="kw">na.omit</span>(<span class="kw">data.frame</span>(res_WT_EH))
induced.sign &lt;-<span class="st"> </span><span class="kw">rownames</span>(res_WT_EV.df)[res_WT_EH.df<span class="op">$</span>log2FoldChange <span class="op">&gt;=</span><span class="st"> </span>
<span class="st">    </span><span class="fl">0.5</span> <span class="op">&amp;</span><span class="st"> </span>res_WT_EH.df<span class="op">$</span>padj <span class="op">&lt;</span><span class="st"> </span>p_cutoff]
<span class="co"># head(induced.sign)</span>
<span class="co"># names(term.induced)</span>
<span class="cf">if</span> (<span class="kw">identical</span>(induced.sign, <span class="kw">character</span>(<span class="dv">0</span>))) {
    <span class="kw">cat</span>(<span class="st">&quot;No genes found that have induced expression&quot;</span>)
} <span class="cf">else</span> {
    term.induced &lt;-<span class="st"> </span><span class="kw">gprofiler</span>(<span class="dt">query =</span> induced.sign, 
        <span class="dt">organism =</span> <span class="st">&quot;hsapiens&quot;</span>)
    term.induced &lt;-<span class="st"> </span>term.induced[<span class="kw">order</span>(term.induced<span class="op">$</span>p.value), 
        ]
    <span class="co"># term.induced$p.value</span>
    <span class="kw">kable</span>(term.induced[<span class="dv">1</span><span class="op">:</span><span class="dv">10</span>, <span class="kw">c</span>(<span class="st">&quot;term.name&quot;</span>, 
        <span class="st">&quot;term.size&quot;</span>, <span class="st">&quot;query.size&quot;</span>, 
        <span class="st">&quot;overlap.size&quot;</span>, <span class="st">&quot;recall&quot;</span>, 
        <span class="st">&quot;precision&quot;</span>, <span class="st">&quot;p.value&quot;</span>, 
        <span class="st">&quot;intersection&quot;</span>)], <span class="dt">format.args =</span> <span class="kw">c</span>(<span class="dt">engeneer =</span> <span class="ot">TRUE</span>, 
        <span class="dt">digits =</span> <span class="dv">3</span>), <span class="dt">caption =</span> <span class="st">&quot;**Table: induced gene functional analysis wit gProfileR. ** &quot;</span>)
}</code></pre></div>
<table>
<caption>(#tab:code block 33)<strong>Table: induced gene functional analysis wit gProfileR. </strong></caption>
<thead>
<tr class="header">
<th></th>
<th align="left">term.name</th>
<th align="right">term.size</th>
<th align="right">query.size</th>
<th align="right">overlap.size</th>
<th align="right">recall</th>
<th align="right">precision</th>
<th align="right">p.value</th>
<th align="left">intersection</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>1</td>
<td align="left">macrolide binding</td>
<td align="right">16</td>
<td align="right">45</td>
<td align="right">3</td>
<td align="right">0.188</td>
<td align="right">0.067</td>
<td align="right">0.0201</td>
<td align="left">FKBP4,FKBP3,FKBP2</td>
</tr>
<tr class="even">
<td>2</td>
<td align="left">FK506 binding</td>
<td align="right">16</td>
<td align="right">45</td>
<td align="right">3</td>
<td align="right">0.188</td>
<td align="right">0.067</td>
<td align="right">0.0201</td>
<td align="left">FKBP4,FKBP3,FKBP2</td>
</tr>
<tr class="odd">
<td>3</td>
<td align="left">Factor: E2F-3; motif: GGCGGGN; match class: 1</td>
<td align="right">9297</td>
<td align="right">45</td>
<td align="right">35</td>
<td align="right">0.004</td>
<td align="right">0.778</td>
<td align="right">0.0464</td>
<td align="left">FKBP4,RANBP3,SNX24,XPO1,ZFAND6,KIF16B,FKBP3,TRIB3,RELB,FAM184A,ZNF142,LGALSL,ACTR10,TET1,COL6A1,DCAF6,CDCA7,TAGLN,HNRNPDL,TRIM11,OTUD6B,EYA3,ERCC3,DACT1,FAM69B,SPATA33,DDB1,UPF3A,DUS1L,ZNF524,FKBP2,TALDO1,LIN28B,DACT3,ZNF521</td>
</tr>
<tr class="even">
<td>NA</td>
<td align="left">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td>NA.1</td>
<td align="left">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td>NA.2</td>
<td align="left">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td>NA.3</td>
<td align="left">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td>NA.4</td>
<td align="left">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td>NA.5</td>
<td align="left">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td>NA.6</td>
<td align="left">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="left">NA</td>
</tr>
</tbody>
</table>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">repressed.sign &lt;-<span class="st"> </span><span class="kw">rownames</span>(res_WT_EH.df)[res_WT_EH.df<span class="op">$</span>log2FoldChange <span class="op">&lt;=</span><span class="st"> </span>
<span class="st">    </span><span class="op">-</span><span class="fl">0.5</span> <span class="op">&amp;</span><span class="st"> </span>res_WT_EH.df<span class="op">$</span>padj <span class="op">&lt;</span><span class="st"> </span>
<span class="st">    </span>p_cutoff]
<span class="cf">if</span> (<span class="kw">identical</span>(repressed.sign, <span class="kw">character</span>(<span class="dv">0</span>))) {
    <span class="kw">cat</span>(<span class="st">&quot;No genes found that have repressed expression&quot;</span>)
} <span class="cf">else</span> {
    term.repressed &lt;-<span class="st"> </span><span class="kw">gprofiler</span>(<span class="dt">query =</span> repressed.sign, 
        <span class="dt">organism =</span> <span class="st">&quot;hsapiens&quot;</span>)
    term.repressed &lt;-<span class="st"> </span>term.repressed[<span class="kw">order</span>(term.repressed<span class="op">$</span>p.value), 
        ]
    <span class="kw">kable</span>(term.repressed[<span class="dv">1</span><span class="op">:</span><span class="dv">10</span>, 
        <span class="kw">c</span>(<span class="st">&quot;term.name&quot;</span>, <span class="st">&quot;term.size&quot;</span>, 
            <span class="st">&quot;query.size&quot;</span>, <span class="st">&quot;overlap.size&quot;</span>, 
            <span class="st">&quot;recall&quot;</span>, <span class="st">&quot;precision&quot;</span>, 
            <span class="st">&quot;p.value&quot;</span>, <span class="st">&quot;intersection&quot;</span>)], 
        <span class="dt">format.args =</span> <span class="kw">c</span>(<span class="dt">engeneer =</span> <span class="ot">TRUE</span>, 
            <span class="dt">digits =</span> <span class="dv">3</span>), <span class="dt">caption =</span> <span class="st">&quot;**Table: repressed genes functional analysis with gProfileR. ** &quot;</span>)
}</code></pre></div>
<table>
<caption>(#tab:code block 33)<strong>Table: repressed genes functional analysis with gProfileR. </strong></caption>
<thead>
<tr class="header">
<th></th>
<th align="left">term.name</th>
<th align="right">term.size</th>
<th align="right">query.size</th>
<th align="right">overlap.size</th>
<th align="right">recall</th>
<th align="right">precision</th>
<th align="right">p.value</th>
<th align="left">intersection</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>19</td>
<td align="left">Cytokine Signaling in Immune system</td>
<td align="right">685</td>
<td align="right">22</td>
<td align="right">11</td>
<td align="right">0.016</td>
<td align="right">0.500</td>
<td align="right">7.90e-06</td>
<td align="left">ICAM1,OSM,NFKBIA,RELB,CISH,EGR1,SOCS2,IRF1,MYC,PIM1,JUN</td>
</tr>
<tr class="even">
<td>18</td>
<td align="left">Immune System</td>
<td align="right">2010</td>
<td align="right">22</td>
<td align="right">16</td>
<td align="right">0.008</td>
<td align="right">0.727</td>
<td align="right">1.80e-05</td>
<td align="left">ICAM1,OSM,NFKBIA,RELB,CISH,TNFAIP3,EGR1,SOCS2,IRF1,MYC,PIM1,NFKBIE,MB21D1,SLCO4C1,JUN,UBOX5</td>
</tr>
<tr class="odd">
<td>13</td>
<td align="left">Epstein-Barr virus infection</td>
<td align="right">197</td>
<td align="right">26</td>
<td align="right">7</td>
<td align="right">0.036</td>
<td align="right">0.269</td>
<td align="right">2.44e-04</td>
<td align="left">ICAM1,NFKBIA,RELB,TNFAIP3,MYC,NFKBIE,JUN</td>
</tr>
<tr class="even">
<td>20</td>
<td align="left">Signaling by Interleukins</td>
<td align="right">463</td>
<td align="right">22</td>
<td align="right">8</td>
<td align="right">0.017</td>
<td align="right">0.364</td>
<td align="right">6.51e-04</td>
<td align="left">ICAM1,OSM,NFKBIA,CISH,SOCS2,MYC,PIM1,JUN</td>
</tr>
<tr class="odd">
<td>3</td>
<td align="left">I-kappaB/NF-kappaB complex</td>
<td align="right">7</td>
<td align="right">42</td>
<td align="right">3</td>
<td align="right">0.429</td>
<td align="right">0.071</td>
<td align="right">8.70e-04</td>
<td align="left">BCL3,NFKBIA,RELB</td>
</tr>
<tr class="even">
<td>22</td>
<td align="left">Factor: E2F; motif: GGCGSG; match class: 1</td>
<td align="right">10205</td>
<td align="right">42</td>
<td align="right">37</td>
<td align="right">0.004</td>
<td align="right">0.881</td>
<td align="right">1.14e-03</td>
<td align="left">MSMO1,VMP1,KLF6,BCL3,ICAM1,OSM,NFKBIA,RELB,ARRDC3,CISH,TNFAIP3,EGR1,SOCS2,ARL4A,IRF1,A4GALT,TMEM160,NINJ1,BHLHE40,STX11,MYC,PIM1,NFKBIZ,NFKBIE,RAB11FIP1,ETS2,ANKRD33B,MB21D1,KCNK5,LETM2,STARD5,SLCO4C1,JUN,DDN,H1FX,UBOX5,TSC22D2</td>
</tr>
<tr class="odd">
<td>11</td>
<td align="left">HTLV-I infection</td>
<td align="right">253</td>
<td align="right">26</td>
<td align="right">7</td>
<td align="right">0.028</td>
<td align="right">0.269</td>
<td align="right">1.26e-03</td>
<td align="left">ICAM1,NFKBIA,RELB,EGR1,MYC,ETS2,JUN</td>
</tr>
<tr class="even">
<td>14</td>
<td align="left">C-type lectin receptor signaling pathway</td>
<td align="right">104</td>
<td align="right">26</td>
<td align="right">5</td>
<td align="right">0.048</td>
<td align="right">0.192</td>
<td align="right">1.70e-03</td>
<td align="left">BCL3,NFKBIA,RELB,IRF1,JUN</td>
</tr>
<tr class="odd">
<td>15</td>
<td align="left">TNF signaling pathway</td>
<td align="right">108</td>
<td align="right">26</td>
<td align="right">5</td>
<td align="right">0.046</td>
<td align="right">0.192</td>
<td align="right">2.04e-03</td>
<td align="left">BCL3,ICAM1,NFKBIA,TNFAIP3,JUN</td>
</tr>
<tr class="even">
<td>9</td>
<td align="left">TNF-alpha/NF-kappa B signaling complex (CHUK, KPNA3, NFKB2, NFKBIB, REL, IKBKG, NFKB1, NFKBIE, RELB, NFKBIA, RELA, TNIP2)</td>
<td align="right">12</td>
<td align="right">12</td>
<td align="right">3</td>
<td align="right">0.250</td>
<td align="right">0.250</td>
<td align="right">2.54e-03</td>
<td align="left">NFKBIA,RELB,NFKBIE</td>
</tr>
</tbody>
</table>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># kable(head(term.induced[,c(&#39;p.value&#39;,</span>
<span class="co"># &#39;term.name&#39;,&#39;intersection&#39;)],</span>
<span class="co"># 10))</span></code></pre></div>
<hr />
</div>
</div>
<div id="untreated-wt-vs-hmgb1-h" class="section level4">
<h4><span class="header-section-number">0.3.9.4</span> Untreated (WT) vs HMGB1 (H)</h4>
<div id="plots-2" class="section level5 tabset tabset-fade tabset-pills">
<h5><span class="header-section-number">0.3.9.4.1</span> Plots</h5>
<div id="ma-plot-2" class="section level6">
<h6><span class="header-section-number">0.3.9.4.1.1</span> MA-plot</h6>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">res_WT_H &lt;-<span class="st"> </span><span class="kw">results</span>(wald.test, 
    <span class="dt">contrast =</span> <span class="kw">c</span>(<span class="st">&quot;condition&quot;</span>, <span class="st">&quot;WT&quot;</span>, 
        <span class="st">&quot;H&quot;</span>), <span class="dt">alpha =</span> p_cutoff, 
    <span class="dt">pAdjustMethod =</span> <span class="st">&quot;BH&quot;</span>)
res_WT_H_ashr &lt;-<span class="st"> </span><span class="kw">lfcShrink</span>(wald.test, 
    <span class="dt">contrast =</span> <span class="kw">c</span>(<span class="st">&quot;condition&quot;</span>, <span class="st">&quot;WT&quot;</span>, 
        <span class="st">&quot;H&quot;</span>), <span class="dt">res =</span> res_WT_H, <span class="dt">type =</span> <span class="st">&quot;ashr&quot;</span>)
<span class="kw">par</span>(<span class="dt">mfrow =</span> <span class="kw">c</span>(<span class="dv">1</span>, <span class="dv">2</span>))
<span class="kw">plotMA</span>(res_WT_H, <span class="dt">xlim =</span> xlim, <span class="dt">ylim =</span> ylim, 
    <span class="dt">main =</span> <span class="st">&quot;normal&quot;</span>)
<span class="kw">plotMA</span>(res_WT_H_ashr, <span class="dt">xlim =</span> xlim, 
    <span class="dt">ylim =</span> ylim, <span class="dt">main =</span> <span class="st">&quot;ashr&quot;</span>)</code></pre></div>
<p><img src="_main_files/figure-html/code%20block%2034-1.png" width="672" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">invisible</span>(<span class="kw">dev.off</span>())</code></pre></div>
</div>
<div id="historgram-of-adjusted-p-values-2" class="section level6">
<h6><span class="header-section-number">0.3.9.4.1.2</span> Historgram of adjusted p-values</h6>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">hist</span>(res_WT_H<span class="op">$</span>padj, <span class="dt">breaks =</span> <span class="dv">20</span>, 
    <span class="dt">col =</span> <span class="st">&quot;grey&quot;</span>, <span class="dt">main =</span> <span class="st">&quot;DESeq2 p-value distribution&quot;</span>, 
    <span class="dt">xlab =</span> <span class="st">&quot;DESeq2 P-value&quot;</span>, <span class="dt">ylab =</span> <span class="st">&quot;Number of genes&quot;</span>)</code></pre></div>
<p><img src="_main_files/figure-html/code%20block%2035-1.png" width="672" /></p>
</div>
<div id="volacano-plot-2" class="section level6">
<h6><span class="header-section-number">0.3.9.4.1.3</span> Volacano-plot</h6>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">resOrdered_WT_H &lt;-<span class="st"> </span>res_WT_H[<span class="kw">order</span>(res_WT_H<span class="op">$</span>padj), 
    ]
<span class="kw">write.csv</span>(<span class="kw">as.data.frame</span>(resOrdered_WT_H), 
    <span class="dt">file =</span> <span class="st">&quot;condition_WT_H_alpha0.1_results.csv&quot;</span>)
WT_H_p1 &lt;-<span class="st"> </span><span class="kw">enhanced_volcano_plots</span>(<span class="st">&quot;WT vs. H (padj=0.05,log2fc=0.5)&quot;</span>, 
    res_WT_H, <span class="fl">0.05</span>, fc_cutoff)
WT_H_p2 &lt;-<span class="st"> </span><span class="kw">enhanced_volcano_plots</span>(<span class="st">&quot;WT vs. H (padj=0.1,log2fc=0.5)&quot;</span>, 
    res_WT_H, p_cutoff, fc_cutoff)
<span class="kw">grid.arrange</span>(WT_H_p1, WT_H_p2, 
    <span class="dt">nrow =</span> <span class="dv">2</span>, <span class="dt">ncol =</span> <span class="dv">1</span>)
<span class="kw">grid.rect</span>(<span class="dt">gp =</span> <span class="kw">gpar</span>(<span class="dt">fill =</span> <span class="ot">NA</span>))</code></pre></div>
<p><img src="_main_files/figure-html/code%20block%2036-1.png" width="960" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">invisible</span>(<span class="kw">dev.off</span>())</code></pre></div>
</div>
<div id="plot-counts-2" class="section level6">
<h6><span class="header-section-number">0.3.9.4.1.4</span> Plot counts</h6>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">nBestFeatures =<span class="st"> </span><span class="dv">20</span>
ord &lt;-<span class="st"> </span><span class="kw">order</span>(res_WT_H<span class="op">$</span>padj, <span class="dt">decreasing =</span> <span class="ot">FALSE</span>)
<span class="cf">for</span> (i <span class="cf">in</span> <span class="kw">head</span>(ord, nBestFeatures)) {
    <span class="kw">print</span>(<span class="kw">plotCounts_gg</span>(i, <span class="dt">dds =</span> wald.test, 
        <span class="dt">intgroup =</span> <span class="kw">c</span>(<span class="st">&quot;condition&quot;</span>)))
}</code></pre></div>
<p><img src="_main_files/figure-html/code%20block%2037-1.png" width="672" /><img src="_main_files/figure-html/code%20block%2037-2.png" width="672" /><img src="_main_files/figure-html/code%20block%2037-3.png" width="672" /><img src="_main_files/figure-html/code%20block%2037-4.png" width="672" /><img src="_main_files/figure-html/code%20block%2037-5.png" width="672" /><img src="_main_files/figure-html/code%20block%2037-6.png" width="672" /><img src="_main_files/figure-html/code%20block%2037-7.png" width="672" /><img src="_main_files/figure-html/code%20block%2037-8.png" width="672" /><img src="_main_files/figure-html/code%20block%2037-9.png" width="672" /><img src="_main_files/figure-html/code%20block%2037-10.png" width="672" /><img src="_main_files/figure-html/code%20block%2037-11.png" width="672" /><img src="_main_files/figure-html/code%20block%2037-12.png" width="672" /><img src="_main_files/figure-html/code%20block%2037-13.png" width="672" /><img src="_main_files/figure-html/code%20block%2037-14.png" width="672" /><img src="_main_files/figure-html/code%20block%2037-15.png" width="672" /><img src="_main_files/figure-html/code%20block%2037-16.png" width="672" /><img src="_main_files/figure-html/code%20block%2037-17.png" width="672" /><img src="_main_files/figure-html/code%20block%2037-18.png" width="672" /><img src="_main_files/figure-html/code%20block%2037-19.png" width="672" /><img src="_main_files/figure-html/code%20block%2037-20.png" width="672" /></p>
</div>
</div>
<div id="select-genes-based-on-fdr-and-make-heatmap-2" class="section level5">
<h5><span class="header-section-number">0.3.9.4.2</span> Select genes based on FDR and make heatmap</h5>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">  gene.kept &lt;-<span class="st"> </span><span class="kw">rownames</span>(res_WT_H)[res_WT_H<span class="op">$</span>padj <span class="op">&lt;=</span><span class="st"> </span>p_cutoff <span class="op">&amp;</span><span class="st"> </span><span class="op">!</span><span class="kw">is.na</span>(res_WT_H<span class="op">$</span>padj) <span class="op">&amp;</span><span class="st"> </span>( res_WT_H<span class="op">$</span>log2FoldChange <span class="op">&lt;=</span><span class="st"> </span><span class="op">-</span><span class="fl">0.5</span> <span class="op">|</span><span class="st"> </span>res_WT_H<span class="op">$</span>log2FoldChange <span class="op">&gt;=</span><span class="st"> </span><span class="fl">0.5</span>)]
  count.table.kept &lt;-<span class="st"> </span><span class="kw">log2</span>(count.table <span class="op">+</span><span class="st"> </span>epsilon)[gene.kept, ]
  <span class="kw">heatmap.2</span>(<span class="kw">as.matrix</span>(count.table.kept), 
            <span class="dt">scale=</span><span class="st">&quot;row&quot;</span>, 
            <span class="dt">hclust=</span><span class="cf">function</span>(x) <span class="kw">hclust</span>(x,<span class="dt">method=</span><span class="st">&quot;average&quot;</span>), 
            <span class="dt">distfun=</span><span class="cf">function</span>(x) <span class="kw">as.dist</span>((<span class="dv">1</span><span class="op">-</span><span class="kw">cor</span>(<span class="kw">t</span>(x)))<span class="op">/</span><span class="dv">2</span>), 
            <span class="dt">trace=</span><span class="st">&quot;none&quot;</span>, 
            <span class="dt">density=</span><span class="st">&quot;none&quot;</span>, 
            <span class="co">#labRow=&quot;&quot;,</span>
            <span class="dt">cexCol=</span><span class="fl">0.7</span>)</code></pre></div>
<p><img src="_main_files/figure-html/code%20black%2038-1.png" width="960" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">  <span class="kw">invisible</span>(<span class="kw">dev.off</span>())</code></pre></div>
</div>
<div id="do-functional-enrichment-2" class="section level5">
<h5><span class="header-section-number">0.3.9.4.3</span> Do functional enrichment</h5>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">res_WT_H.df &lt;-<span class="st"> </span><span class="kw">na.omit</span>(<span class="kw">data.frame</span>(res_WT_H))
induced.sign &lt;-<span class="st"> </span><span class="kw">rownames</span>(res_WT_H.df)[res_WT_H.df<span class="op">$</span>log2FoldChange <span class="op">&gt;=</span><span class="st"> </span>
<span class="st">    </span><span class="fl">0.5</span> <span class="op">&amp;</span><span class="st"> </span>res_WT_H.df<span class="op">$</span>padj <span class="op">&lt;</span><span class="st"> </span>p_cutoff]
<span class="co"># head(induced.sign)</span>
<span class="co"># names(term.induced)</span>
<span class="cf">if</span> (<span class="kw">identical</span>(induced.sign, <span class="kw">character</span>(<span class="dv">0</span>))) {
    <span class="kw">cat</span>(<span class="st">&quot;No genes found that have induced expression&quot;</span>)
} <span class="cf">else</span> {
    term.induced &lt;-<span class="st"> </span><span class="kw">gprofiler</span>(<span class="dt">query =</span> induced.sign, 
        <span class="dt">organism =</span> <span class="st">&quot;hsapiens&quot;</span>)
    term.induced &lt;-<span class="st"> </span>term.induced[<span class="kw">order</span>(term.induced<span class="op">$</span>p.value), 
        ]
    <span class="co"># term.induced$p.value</span>
    <span class="kw">kable</span>(term.induced[<span class="dv">1</span><span class="op">:</span><span class="dv">10</span>, <span class="kw">c</span>(<span class="st">&quot;term.name&quot;</span>, 
        <span class="st">&quot;term.size&quot;</span>, <span class="st">&quot;query.size&quot;</span>, 
        <span class="st">&quot;overlap.size&quot;</span>, <span class="st">&quot;recall&quot;</span>, 
        <span class="st">&quot;precision&quot;</span>, <span class="st">&quot;p.value&quot;</span>, 
        <span class="st">&quot;intersection&quot;</span>)], <span class="dt">format.args =</span> <span class="kw">c</span>(<span class="dt">engeneer =</span> <span class="ot">TRUE</span>, 
        <span class="dt">digits =</span> <span class="dv">3</span>), <span class="dt">caption =</span> <span class="st">&quot;**Table: induced gene functional analysis wit gProfileR. ** &quot;</span>)
}</code></pre></div>
<table>
<caption>(#tab:code block 39)<strong>Table: induced gene functional analysis wit gProfileR. </strong></caption>
<thead>
<tr class="header">
<th></th>
<th align="left">term.name</th>
<th align="left">term.size</th>
<th align="left">query.size</th>
<th align="left">overlap.size</th>
<th align="left">recall</th>
<th align="left">precision</th>
<th align="left">p.value</th>
<th align="left">intersection</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td>NA.1</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td>NA.2</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td>NA.3</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td>NA.4</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td>NA.5</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td>NA.6</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td>NA.7</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td>NA.8</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td>NA.9</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
</tr>
</tbody>
</table>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">repressed.sign &lt;-<span class="st"> </span><span class="kw">rownames</span>(res_WT_H.df)[res_WT_H.df<span class="op">$</span>log2FoldChange <span class="op">&lt;=</span><span class="st"> </span>
<span class="st">    </span><span class="op">-</span><span class="fl">0.5</span> <span class="op">&amp;</span><span class="st"> </span>res_WT_H.df<span class="op">$</span>padj <span class="op">&lt;</span><span class="st"> </span>p_cutoff]
<span class="cf">if</span> (<span class="kw">identical</span>(repressed.sign, <span class="kw">character</span>(<span class="dv">0</span>))) {
    <span class="kw">cat</span>(<span class="st">&quot;No genes found that have repressed expression&quot;</span>)
} <span class="cf">else</span> {
    term.repressed &lt;-<span class="st"> </span><span class="kw">gprofiler</span>(<span class="dt">query =</span> repressed.sign, 
        <span class="dt">organism =</span> <span class="st">&quot;hsapiens&quot;</span>)
    term.repressed &lt;-<span class="st"> </span>term.repressed[<span class="kw">order</span>(term.repressed<span class="op">$</span>p.value), 
        ]
    <span class="kw">kable</span>(term.repressed[<span class="dv">1</span><span class="op">:</span><span class="dv">10</span>, 
        <span class="kw">c</span>(<span class="st">&quot;term.name&quot;</span>, <span class="st">&quot;term.size&quot;</span>, 
            <span class="st">&quot;query.size&quot;</span>, <span class="st">&quot;overlap.size&quot;</span>, 
            <span class="st">&quot;recall&quot;</span>, <span class="st">&quot;precision&quot;</span>, 
            <span class="st">&quot;p.value&quot;</span>, <span class="st">&quot;intersection&quot;</span>)], 
        <span class="dt">format.args =</span> <span class="kw">c</span>(<span class="dt">engeneer =</span> <span class="ot">TRUE</span>, 
            <span class="dt">digits =</span> <span class="dv">3</span>), <span class="dt">caption =</span> <span class="st">&quot;**Table: repressed genes functional analysis with gProfileR. ** &quot;</span>)
}</code></pre></div>
<table>
<caption>(#tab:code block 39)<strong>Table: repressed genes functional analysis with gProfileR. </strong></caption>
<thead>
<tr class="header">
<th></th>
<th align="left">term.name</th>
<th align="right">term.size</th>
<th align="right">query.size</th>
<th align="right">overlap.size</th>
<th align="right">recall</th>
<th align="right">precision</th>
<th align="right">p.value</th>
<th align="left">intersection</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>10</td>
<td align="left">TNF signaling pathway</td>
<td align="right">108</td>
<td align="right">14</td>
<td align="right">5</td>
<td align="right">0.046</td>
<td align="right">0.357</td>
<td align="right">6.81e-05</td>
<td align="left">BCL3,ICAM1,NFKBIA,TNFAIP3,JUN</td>
</tr>
<tr class="even">
<td>7</td>
<td align="left">Epstein-Barr virus infection</td>
<td align="right">197</td>
<td align="right">14</td>
<td align="right">5</td>
<td align="right">0.025</td>
<td align="right">0.357</td>
<td align="right">1.31e-03</td>
<td align="left">ICAM1,NFKBIA,TNFAIP3,NFKBIE,JUN</td>
</tr>
<tr class="odd">
<td>3</td>
<td align="left">CHUK-NFKB2-REL-IKBKG-SPAG9-NFKB1-NFKBIE-COPB2-TNIP1-NFKBIA-RELA-TNIP2 complex</td>
<td align="right">12</td>
<td align="right">6</td>
<td align="right">2</td>
<td align="right">0.167</td>
<td align="right">0.333</td>
<td align="right">1.11e-02</td>
<td align="left">NFKBIA,NFKBIE</td>
</tr>
<tr class="even">
<td>4</td>
<td align="left">TNF-alpha/NF-kappa B signaling complex (CHUK, KPNA3, NFKB2, NFKBIB, REL, IKBKG, NFKB1, NFKBIE, RELB, NFKBIA, RELA, TNIP2)</td>
<td align="right">12</td>
<td align="right">6</td>
<td align="right">2</td>
<td align="right">0.167</td>
<td align="right">0.333</td>
<td align="right">1.11e-02</td>
<td align="left">NFKBIA,NFKBIE</td>
</tr>
<tr class="odd">
<td>2</td>
<td align="left">regulation of DNA binding transcription factor activity</td>
<td align="right">400</td>
<td align="right">23</td>
<td align="right">6</td>
<td align="right">0.015</td>
<td align="right">0.261</td>
<td align="right">1.34e-02</td>
<td align="left">ICAM1,NFKBIA,TNFAIP3,BHLHE40,NFKBIE,JUN</td>
</tr>
<tr class="even">
<td>9</td>
<td align="left">B cell receptor signaling pathway</td>
<td align="right">70</td>
<td align="right">14</td>
<td align="right">3</td>
<td align="right">0.043</td>
<td align="right">0.214</td>
<td align="right">1.68e-02</td>
<td align="left">NFKBIA,NFKBIE,JUN</td>
</tr>
<tr class="odd">
<td>1</td>
<td align="left">nucleotide-binding oligomerization domain containing 1 signaling pathway</td>
<td align="right">4</td>
<td align="right">23</td>
<td align="right">2</td>
<td align="right">0.500</td>
<td align="right">0.087</td>
<td align="right">1.93e-02</td>
<td align="left">NFKBIA,TNFAIP3</td>
</tr>
<tr class="even">
<td>12</td>
<td align="left">Factor: NF-kappaB; motif: GGGGATYCCC</td>
<td align="right">6390</td>
<td align="right">22</td>
<td align="right">17</td>
<td align="right">0.003</td>
<td align="right">0.773</td>
<td align="right">3.27e-02</td>
<td align="left">BCL3,ICAM1,HIVEP1,NFKBIA,ARRDC3,TNFAIP3,NINJ1,STX11,NFKBIZ,NFKBIE,ZFP36L2,KLF15,TPSAB1,SLCO4C1,FAM174A,JUN,H1F0</td>
</tr>
<tr class="odd">
<td>13</td>
<td align="left">Factor: NF-kappaB; motif: GGGGATYCCC; match class: 0</td>
<td align="right">6390</td>
<td align="right">22</td>
<td align="right">17</td>
<td align="right">0.003</td>
<td align="right">0.773</td>
<td align="right">3.27e-02</td>
<td align="left">BCL3,ICAM1,HIVEP1,NFKBIA,ARRDC3,TNFAIP3,NINJ1,STX11,NFKBIZ,NFKBIE,ZFP36L2,KLF15,TPSAB1,SLCO4C1,FAM174A,JUN,H1F0</td>
</tr>
<tr class="even">
<td>11</td>
<td align="left">Th1 and Th2 cell differentiation</td>
<td align="right">90</td>
<td align="right">14</td>
<td align="right">3</td>
<td align="right">0.033</td>
<td align="right">0.214</td>
<td align="right">3.53e-02</td>
<td align="left">NFKBIA,NFKBIE,JUN</td>
</tr>
</tbody>
</table>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># kable(head(term.induced[,c(&#39;p.value&#39;,</span>
<span class="co"># &#39;term.name&#39;,&#39;intersection&#39;)],</span>
<span class="co"># 10))</span></code></pre></div>
</div>
</div>
<div id="epo-with-vehicle-ev-vs-epo-with-hmgb1-eh" class="section level4">
<h4><span class="header-section-number">0.3.9.5</span> EPO with Vehicle (EV) vs EPO with HMGB1 (EH)</h4>
<div id="plots-3" class="section level5 tabset tabset-fade tabset-pills">
<h5><span class="header-section-number">0.3.9.5.1</span> Plots</h5>
<div id="ma-plot-3" class="section level6">
<h6><span class="header-section-number">0.3.9.5.1.1</span> MA-plot</h6>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">res_EV_EH &lt;-<span class="st"> </span><span class="kw">results</span>(wald.test, 
    <span class="dt">contrast =</span> <span class="kw">c</span>(<span class="st">&quot;condition&quot;</span>, <span class="st">&quot;EV&quot;</span>, 
        <span class="st">&quot;EH&quot;</span>), <span class="dt">alpha =</span> p_cutoff, 
    <span class="dt">pAdjustMethod =</span> <span class="st">&quot;BH&quot;</span>)
res_EV_EH_ashr &lt;-<span class="st"> </span><span class="kw">lfcShrink</span>(wald.test, 
    <span class="dt">contrast =</span> <span class="kw">c</span>(<span class="st">&quot;condition&quot;</span>, <span class="st">&quot;EV&quot;</span>, 
        <span class="st">&quot;EH&quot;</span>), <span class="dt">res =</span> res_EV_EH, 
    <span class="dt">type =</span> <span class="st">&quot;ashr&quot;</span>)
<span class="kw">par</span>(<span class="dt">mfrow =</span> <span class="kw">c</span>(<span class="dv">1</span>, <span class="dv">2</span>))
<span class="kw">plotMA</span>(res_EV_EH, <span class="dt">xlim =</span> xlim, 
    <span class="dt">ylim =</span> ylim, <span class="dt">main =</span> <span class="st">&quot;normal&quot;</span>)
<span class="kw">plotMA</span>(res_EV_EH_ashr, <span class="dt">xlim =</span> xlim, 
    <span class="dt">ylim =</span> ylim, <span class="dt">main =</span> <span class="st">&quot;ashr&quot;</span>)</code></pre></div>
<p><img src="_main_files/figure-html/code%20block%2040-1.png" width="672" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">invisible</span>(<span class="kw">dev.off</span>())</code></pre></div>
</div>
<div id="historgram-of-adjusted-p-values-3" class="section level6">
<h6><span class="header-section-number">0.3.9.5.1.2</span> Historgram of adjusted p-values</h6>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">hist</span>(res_EV_EH<span class="op">$</span>padj, <span class="dt">breaks =</span> <span class="dv">20</span>, 
    <span class="dt">col =</span> <span class="st">&quot;grey&quot;</span>, <span class="dt">main =</span> <span class="st">&quot;DESeq2 p-value distribution&quot;</span>, 
    <span class="dt">xlab =</span> <span class="st">&quot;DESeq2 P-value&quot;</span>, <span class="dt">ylab =</span> <span class="st">&quot;Number of genes&quot;</span>)</code></pre></div>
<p><img src="_main_files/figure-html/code%20block%2041-1.png" width="672" /></p>
</div>
<div id="volacano-plot-3" class="section level6">
<h6><span class="header-section-number">0.3.9.5.1.3</span> Volacano-plot</h6>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">resOrdered_EV_EH &lt;-<span class="st"> </span>res_EV_EH[<span class="kw">order</span>(res_EV_EH<span class="op">$</span>padj), 
    ]
<span class="kw">write.csv</span>(<span class="kw">as.data.frame</span>(resOrdered_EV_EH), 
    <span class="dt">file =</span> <span class="st">&quot;condition_EV_EH_alpha0.1_results.csv&quot;</span>)
EV_EH_p1 &lt;-<span class="st"> </span><span class="kw">enhanced_volcano_plots</span>(<span class="st">&quot;EV vs. EH (padj=0.05,log2fc=0.5)&quot;</span>, 
    res_EV_EH, <span class="fl">0.05</span>, fc_cutoff)
EV_EH_p2 &lt;-<span class="st"> </span><span class="kw">enhanced_volcano_plots</span>(<span class="st">&quot;EV vs. EH (padj=0.1,log2fc=0.5)&quot;</span>, 
    res_EV_EH, p_cutoff, fc_cutoff)
<span class="kw">grid.arrange</span>(EV_EH_p1, EV_EH_p2, 
    <span class="dt">nrow =</span> <span class="dv">2</span>, <span class="dt">ncol =</span> <span class="dv">1</span>)
<span class="kw">grid.rect</span>(<span class="dt">gp =</span> <span class="kw">gpar</span>(<span class="dt">fill =</span> <span class="ot">NA</span>))</code></pre></div>
<p><img src="_main_files/figure-html/code%20block%2042-1.png" width="960" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">invisible</span>(<span class="kw">dev.off</span>())</code></pre></div>
</div>
<div id="plot-counts-3" class="section level6">
<h6><span class="header-section-number">0.3.9.5.1.4</span> Plot counts</h6>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">nBestFeatures =<span class="st"> </span><span class="dv">20</span>
ord &lt;-<span class="st"> </span><span class="kw">order</span>(res_EV_EH<span class="op">$</span>padj, <span class="dt">decreasing =</span> <span class="ot">FALSE</span>)
<span class="cf">for</span> (i <span class="cf">in</span> <span class="kw">head</span>(ord, nBestFeatures)) {
    <span class="kw">print</span>(<span class="kw">plotCounts_gg</span>(i, <span class="dt">dds =</span> wald.test, 
        <span class="dt">intgroup =</span> <span class="kw">c</span>(<span class="st">&quot;condition&quot;</span>)))
}</code></pre></div>
<p><img src="_main_files/figure-html/code%20block%2043-1.png" width="672" /><img src="_main_files/figure-html/code%20block%2043-2.png" width="672" /><img src="_main_files/figure-html/code%20block%2043-3.png" width="672" /><img src="_main_files/figure-html/code%20block%2043-4.png" width="672" /><img src="_main_files/figure-html/code%20block%2043-5.png" width="672" /><img src="_main_files/figure-html/code%20block%2043-6.png" width="672" /><img src="_main_files/figure-html/code%20block%2043-7.png" width="672" /><img src="_main_files/figure-html/code%20block%2043-8.png" width="672" /><img src="_main_files/figure-html/code%20block%2043-9.png" width="672" /><img src="_main_files/figure-html/code%20block%2043-10.png" width="672" /><img src="_main_files/figure-html/code%20block%2043-11.png" width="672" /><img src="_main_files/figure-html/code%20block%2043-12.png" width="672" /><img src="_main_files/figure-html/code%20block%2043-13.png" width="672" /><img src="_main_files/figure-html/code%20block%2043-14.png" width="672" /><img src="_main_files/figure-html/code%20block%2043-15.png" width="672" /><img src="_main_files/figure-html/code%20block%2043-16.png" width="672" /><img src="_main_files/figure-html/code%20block%2043-17.png" width="672" /><img src="_main_files/figure-html/code%20block%2043-18.png" width="672" /><img src="_main_files/figure-html/code%20block%2043-19.png" width="672" /><img src="_main_files/figure-html/code%20block%2043-20.png" width="672" /></p>
</div>
</div>
<div id="select-genes-based-on-fdr-and-make-heatmap-3" class="section level5">
<h5><span class="header-section-number">0.3.9.5.2</span> Select genes based on FDR and make heatmap</h5>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">  gene.kept &lt;-<span class="st"> </span><span class="kw">rownames</span>(res_EV_EH)[res_EV_EH<span class="op">$</span>padj <span class="op">&lt;=</span><span class="st"> </span>p_cutoff <span class="op">&amp;</span><span class="st"> </span><span class="op">!</span><span class="kw">is.na</span>(res_EV_EH<span class="op">$</span>padj) <span class="op">&amp;</span><span class="st"> </span>( res_EV_EH<span class="op">$</span>log2FoldChange <span class="op">&lt;=</span><span class="st"> </span><span class="op">-</span><span class="fl">0.5</span> <span class="op">|</span><span class="st"> </span>res_EV_EH<span class="op">$</span>log2FoldChange <span class="op">&gt;=</span><span class="st"> </span><span class="fl">0.5</span>)]
  count.table.kept &lt;-<span class="st"> </span><span class="kw">log2</span>(count.table <span class="op">+</span><span class="st"> </span>epsilon)[gene.kept, ]
  <span class="kw">heatmap.2</span>(<span class="kw">as.matrix</span>(count.table.kept), 
            <span class="dt">scale=</span><span class="st">&quot;row&quot;</span>, 
            <span class="dt">hclust=</span><span class="cf">function</span>(x) <span class="kw">hclust</span>(x,<span class="dt">method=</span><span class="st">&quot;average&quot;</span>), 
            <span class="dt">distfun=</span><span class="cf">function</span>(x) <span class="kw">as.dist</span>((<span class="dv">1</span><span class="op">-</span><span class="kw">cor</span>(<span class="kw">t</span>(x)))<span class="op">/</span><span class="dv">2</span>), 
            <span class="dt">trace=</span><span class="st">&quot;none&quot;</span>, 
            <span class="dt">density=</span><span class="st">&quot;none&quot;</span>, 
            <span class="co">#labRow=&quot;&quot;,</span>
            <span class="dt">cexCol=</span><span class="fl">0.7</span>)</code></pre></div>
<p><img src="_main_files/figure-html/code%20black%2044-1.png" width="960" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">  <span class="kw">invisible</span>(<span class="kw">dev.off</span>())</code></pre></div>
</div>
<div id="do-functional-enrichment-3" class="section level5">
<h5><span class="header-section-number">0.3.9.5.3</span> Do functional enrichment</h5>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">res_EV_EH.df &lt;-<span class="st"> </span><span class="kw">na.omit</span>(<span class="kw">data.frame</span>(res_EV_EH))
induced.sign &lt;-<span class="st"> </span><span class="kw">rownames</span>(res_EV_EH.df)[res_EV_EH.df<span class="op">$</span>log2FoldChange <span class="op">&gt;=</span><span class="st"> </span>
<span class="st">    </span><span class="fl">0.5</span> <span class="op">&amp;</span><span class="st"> </span>res_EV_EH.df<span class="op">$</span>padj <span class="op">&lt;</span><span class="st"> </span>p_cutoff]
<span class="co"># head(induced.sign)</span>
<span class="co"># names(term.induced)</span>
<span class="cf">if</span> (<span class="kw">identical</span>(induced.sign, <span class="kw">character</span>(<span class="dv">0</span>))) {
    <span class="kw">cat</span>(<span class="st">&quot;No genes found that have induced expression&quot;</span>)
} <span class="cf">else</span> {
    term.induced &lt;-<span class="st"> </span><span class="kw">gprofiler</span>(<span class="dt">query =</span> induced.sign, 
        <span class="dt">organism =</span> <span class="st">&quot;hsapiens&quot;</span>)
    term.induced &lt;-<span class="st"> </span>term.induced[<span class="kw">order</span>(term.induced<span class="op">$</span>p.value), 
        ]
    <span class="co"># term.induced$p.value</span>
    <span class="kw">kable</span>(term.induced[<span class="dv">1</span><span class="op">:</span><span class="dv">10</span>, <span class="kw">c</span>(<span class="st">&quot;term.name&quot;</span>, 
        <span class="st">&quot;term.size&quot;</span>, <span class="st">&quot;query.size&quot;</span>, 
        <span class="st">&quot;overlap.size&quot;</span>, <span class="st">&quot;recall&quot;</span>, 
        <span class="st">&quot;precision&quot;</span>, <span class="st">&quot;p.value&quot;</span>, 
        <span class="st">&quot;intersection&quot;</span>)], <span class="dt">format.args =</span> <span class="kw">c</span>(<span class="dt">engeneer =</span> <span class="ot">TRUE</span>, 
        <span class="dt">digits =</span> <span class="dv">3</span>), <span class="dt">caption =</span> <span class="st">&quot;**Table: induced gene functional analysis wit gProfileR. ** &quot;</span>)
}</code></pre></div>
<table>
<caption>(#tab:code block 45)<strong>Table: induced gene functional analysis wit gProfileR. </strong></caption>
<thead>
<tr class="header">
<th></th>
<th align="left">term.name</th>
<th align="right">term.size</th>
<th align="right">query.size</th>
<th align="right">overlap.size</th>
<th align="right">recall</th>
<th align="right">precision</th>
<th align="right">p.value</th>
<th align="left">intersection</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>2</td>
<td align="left">positive regulation of tau-protein kinase activity</td>
<td align="right">4</td>
<td align="right">25</td>
<td align="right">2</td>
<td align="right">0.500</td>
<td align="right">0.080</td>
<td align="right">0.0200</td>
<td align="left">EGR1,NAB2</td>
</tr>
<tr class="even">
<td>3</td>
<td align="left">regulation of cellular protein metabolic process</td>
<td align="right">2438</td>
<td align="right">25</td>
<td align="right">12</td>
<td align="right">0.005</td>
<td align="right">0.480</td>
<td align="right">0.0344</td>
<td align="left">GADD45B,EIF5,EEF2K,GPLD1,EGR1,ID1,HSPA2,ARRDC4,DUSP2,NAB2,SH3BP5L,HSPA1A</td>
</tr>
<tr class="odd">
<td>4</td>
<td align="left">Attenuation phase</td>
<td align="right">14</td>
<td align="right">15</td>
<td align="right">2</td>
<td align="right">0.143</td>
<td align="right">0.133</td>
<td align="right">0.0439</td>
<td align="left">HSPA2,HSPA1A</td>
</tr>
<tr class="even">
<td>1</td>
<td align="left">regulation of tau-protein kinase activity</td>
<td align="right">6</td>
<td align="right">25</td>
<td align="right">2</td>
<td align="right">0.333</td>
<td align="right">0.080</td>
<td align="right">0.0499</td>
<td align="left">EGR1,NAB2</td>
</tr>
<tr class="odd">
<td>NA</td>
<td align="left">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td>NA.1</td>
<td align="left">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td>NA.2</td>
<td align="left">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td>NA.3</td>
<td align="left">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td>NA.4</td>
<td align="left">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td>NA.5</td>
<td align="left">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="left">NA</td>
</tr>
</tbody>
</table>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">repressed.sign &lt;-<span class="st"> </span><span class="kw">rownames</span>(res_EV_EH.df)[res_EV_EH.df<span class="op">$</span>log2FoldChange <span class="op">&lt;=</span><span class="st"> </span>
<span class="st">    </span><span class="op">-</span><span class="fl">0.5</span> <span class="op">&amp;</span><span class="st"> </span>res_EV_EH.df<span class="op">$</span>padj <span class="op">&lt;</span><span class="st"> </span>
<span class="st">    </span>p_cutoff]
<span class="cf">if</span> (<span class="kw">identical</span>(repressed.sign, <span class="kw">character</span>(<span class="dv">0</span>))) {
    <span class="kw">cat</span>(<span class="st">&quot;No genes found that have repressed expression&quot;</span>)
} <span class="cf">else</span> {
    term.repressed &lt;-<span class="st"> </span><span class="kw">gprofiler</span>(<span class="dt">query =</span> repressed.sign, 
        <span class="dt">organism =</span> <span class="st">&quot;hsapiens&quot;</span>)
    term.repressed &lt;-<span class="st"> </span>term.repressed[<span class="kw">order</span>(term.repressed<span class="op">$</span>p.value), 
        ]
    <span class="kw">kable</span>(term.repressed[<span class="dv">1</span><span class="op">:</span><span class="dv">10</span>, 
        <span class="kw">c</span>(<span class="st">&quot;term.name&quot;</span>, <span class="st">&quot;term.size&quot;</span>, 
            <span class="st">&quot;query.size&quot;</span>, <span class="st">&quot;overlap.size&quot;</span>, 
            <span class="st">&quot;recall&quot;</span>, <span class="st">&quot;precision&quot;</span>, 
            <span class="st">&quot;p.value&quot;</span>, <span class="st">&quot;intersection&quot;</span>)], 
        <span class="dt">format.args =</span> <span class="kw">c</span>(<span class="dt">engeneer =</span> <span class="ot">TRUE</span>, 
            <span class="dt">digits =</span> <span class="dv">3</span>), <span class="dt">caption =</span> <span class="st">&quot;**Table: repressed genes functional analysis with gProfileR. ** &quot;</span>)
}</code></pre></div>
<table>
<caption>(#tab:code block 45)<strong>Table: repressed genes functional analysis with gProfileR. </strong></caption>
<thead>
<tr class="header">
<th></th>
<th align="left">term.name</th>
<th align="right">term.size</th>
<th align="right">query.size</th>
<th align="right">overlap.size</th>
<th align="right">recall</th>
<th align="right">precision</th>
<th align="right">p.value</th>
<th align="left">intersection</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>28</td>
<td align="left">TNF signaling pathway</td>
<td align="right">108</td>
<td align="right">15</td>
<td align="right">7</td>
<td align="right">0.065</td>
<td align="right">0.467</td>
<td align="right">0.00e+00</td>
<td align="left">BIRC3,TRAF1,BCL3,ICAM1,NFKBIA,TNFAIP3,JUN</td>
</tr>
<tr class="even">
<td>29</td>
<td align="left">NF-kappa B signaling pathway</td>
<td align="right">93</td>
<td align="right">15</td>
<td align="right">6</td>
<td align="right">0.065</td>
<td align="right">0.400</td>
<td align="right">9.00e-07</td>
<td align="left">BIRC3,TRAF1,ICAM1,NFKBIA,RELB,TNFAIP3</td>
</tr>
<tr class="odd">
<td>30</td>
<td align="left">Epstein-Barr virus infection</td>
<td align="right">197</td>
<td align="right">15</td>
<td align="right">7</td>
<td align="right">0.036</td>
<td align="right">0.467</td>
<td align="right">2.70e-06</td>
<td align="left">TRAF1,ICAM1,NFKBIA,RELB,TNFAIP3,NFKBIE,JUN</td>
</tr>
<tr class="even">
<td>21</td>
<td align="left">I-kappaB/NF-kappaB complex</td>
<td align="right">7</td>
<td align="right">20</td>
<td align="right">3</td>
<td align="right">0.429</td>
<td align="right">0.150</td>
<td align="right">7.49e-05</td>
<td align="left">BCL3,NFKBIA,RELB</td>
</tr>
<tr class="odd">
<td>12</td>
<td align="left">regulation of DNA binding transcription factor activity</td>
<td align="right">400</td>
<td align="right">20</td>
<td align="right">7</td>
<td align="right">0.018</td>
<td align="right">0.350</td>
<td align="right">2.14e-04</td>
<td align="left">TRAF1,ICAM1,NFKBIA,TNFAIP3,BHLHE40,NFKBIE,JUN</td>
</tr>
<tr class="even">
<td>23</td>
<td align="left">TNF-alpha/NF-kappa B signaling complex (CHUK, KPNA3, NFKB2, NFKBIB, REL, IKBKG, NFKB1, NFKBIE, RELB, NFKBIA, RELA, TNIP2)</td>
<td align="right">12</td>
<td align="right">8</td>
<td align="right">3</td>
<td align="right">0.250</td>
<td align="right">0.375</td>
<td align="right">3.12e-04</td>
<td align="left">NFKBIA,RELB,NFKBIE</td>
</tr>
<tr class="odd">
<td>41</td>
<td align="left">TNFR1-induced NFkappaB signaling pathway</td>
<td align="right">26</td>
<td align="right">12</td>
<td align="right">3</td>
<td align="right">0.115</td>
<td align="right">0.250</td>
<td align="right">7.68e-04</td>
<td align="left">BIRC3,TRAF1,TNFAIP3</td>
</tr>
<tr class="even">
<td>19</td>
<td align="left">cellular response to tumor necrosis factor</td>
<td align="right">290</td>
<td align="right">20</td>
<td align="right">6</td>
<td align="right">0.021</td>
<td align="right">0.300</td>
<td align="right">7.94e-04</td>
<td align="left">BIRC3,TNFRSF9,TRAF1,ICAM1,NFKBIA,TNFAIP3</td>
</tr>
<tr class="odd">
<td>18</td>
<td align="left">response to tumor necrosis factor</td>
<td align="right">307</td>
<td align="right">20</td>
<td align="right">6</td>
<td align="right">0.020</td>
<td align="right">0.300</td>
<td align="right">1.11e-03</td>
<td align="left">BIRC3,TNFRSF9,TRAF1,ICAM1,NFKBIA,TNFAIP3</td>
</tr>
<tr class="even">
<td>40</td>
<td align="left">Regulation of TNFR1 signaling</td>
<td align="right">33</td>
<td align="right">12</td>
<td align="right">3</td>
<td align="right">0.091</td>
<td align="right">0.250</td>
<td align="right">1.61e-03</td>
<td align="left">BIRC3,TRAF1,TNFAIP3</td>
</tr>
</tbody>
</table>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># kable(head(term.induced[,c(&#39;p.value&#39;,</span>
<span class="co"># &#39;term.name&#39;,&#39;intersection&#39;)],</span>
<span class="co"># 10))</span></code></pre></div>
</div>
</div>
<div id="epo-with-vehicle-ev-vs-hmgb1-h" class="section level4">
<h4><span class="header-section-number">0.3.9.6</span> EPO with Vehicle (EV) vs HMGB1 (H)</h4>
<div id="plots-4" class="section level5 tabset tabset-fade tabset-pills">
<h5><span class="header-section-number">0.3.9.6.1</span> Plots</h5>
<div id="ma-plot-4" class="section level6">
<h6><span class="header-section-number">0.3.9.6.1.1</span> MA-plot</h6>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">res_EV_H &lt;-<span class="st"> </span><span class="kw">results</span>(wald.test, 
    <span class="dt">contrast =</span> <span class="kw">c</span>(<span class="st">&quot;condition&quot;</span>, <span class="st">&quot;EV&quot;</span>, 
        <span class="st">&quot;H&quot;</span>), <span class="dt">alpha =</span> p_cutoff, 
    <span class="dt">pAdjustMethod =</span> <span class="st">&quot;BH&quot;</span>)
res_EV_H_ashr &lt;-<span class="st"> </span><span class="kw">lfcShrink</span>(wald.test, 
    <span class="dt">contrast =</span> <span class="kw">c</span>(<span class="st">&quot;condition&quot;</span>, <span class="st">&quot;EV&quot;</span>, 
        <span class="st">&quot;H&quot;</span>), <span class="dt">res =</span> res_EV_H, <span class="dt">type =</span> <span class="st">&quot;ashr&quot;</span>)
<span class="kw">par</span>(<span class="dt">mfrow =</span> <span class="kw">c</span>(<span class="dv">1</span>, <span class="dv">2</span>))
<span class="kw">plotMA</span>(res_EV_H, <span class="dt">xlim =</span> xlim, <span class="dt">ylim =</span> ylim, 
    <span class="dt">main =</span> <span class="st">&quot;normal&quot;</span>)
<span class="kw">plotMA</span>(res_EV_H_ashr, <span class="dt">xlim =</span> xlim, 
    <span class="dt">ylim =</span> ylim, <span class="dt">main =</span> <span class="st">&quot;ashr&quot;</span>)</code></pre></div>
<p><img src="_main_files/figure-html/code%20block%2046-1.png" width="672" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">invisible</span>(<span class="kw">dev.off</span>())</code></pre></div>
</div>
<div id="historgram-of-adjusted-p-values-4" class="section level6">
<h6><span class="header-section-number">0.3.9.6.1.2</span> Historgram of adjusted p-values</h6>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">hist</span>(res_EV_H<span class="op">$</span>padj, <span class="dt">breaks =</span> <span class="dv">20</span>, 
    <span class="dt">col =</span> <span class="st">&quot;grey&quot;</span>, <span class="dt">main =</span> <span class="st">&quot;DESeq2 p-value distribution&quot;</span>, 
    <span class="dt">xlab =</span> <span class="st">&quot;DESeq2 P-value&quot;</span>, <span class="dt">ylab =</span> <span class="st">&quot;Number of genes&quot;</span>)</code></pre></div>
<p><img src="_main_files/figure-html/code%20block%2047-1.png" width="672" /></p>
</div>
<div id="volacano-plot-4" class="section level6">
<h6><span class="header-section-number">0.3.9.6.1.3</span> Volacano-plot</h6>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">resOrdered_EV_H &lt;-<span class="st"> </span>res_EV_H[<span class="kw">order</span>(res_EV_H<span class="op">$</span>padj), 
    ]
<span class="kw">write.csv</span>(<span class="kw">as.data.frame</span>(resOrdered_EV_H), 
    <span class="dt">file =</span> <span class="st">&quot;condition_EV_H_alpha0.1_results.csv&quot;</span>)
EV_H_p1 &lt;-<span class="st"> </span><span class="kw">enhanced_volcano_plots</span>(<span class="st">&quot;EV vs. H (padj=0.05,log2fc=0.5)&quot;</span>, 
    res_EV_H, <span class="fl">0.05</span>, fc_cutoff)
EV_H_p2 &lt;-<span class="st"> </span><span class="kw">enhanced_volcano_plots</span>(<span class="st">&quot;EV vs. H (padj=0.1,log2fc=0.5)&quot;</span>, 
    res_EV_H, p_cutoff, fc_cutoff)
<span class="kw">grid.arrange</span>(EV_H_p1, EV_H_p2, 
    <span class="dt">nrow =</span> <span class="dv">2</span>, <span class="dt">ncol =</span> <span class="dv">1</span>)
<span class="kw">grid.rect</span>(<span class="dt">gp =</span> <span class="kw">gpar</span>(<span class="dt">fill =</span> <span class="ot">NA</span>))</code></pre></div>
<p><img src="_main_files/figure-html/code%20block%2048-1.png" width="960" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">invisible</span>(<span class="kw">dev.off</span>())</code></pre></div>
</div>
<div id="plot-counts-4" class="section level6">
<h6><span class="header-section-number">0.3.9.6.1.4</span> Plot counts</h6>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">nBestFeatures =<span class="st"> </span><span class="dv">20</span>
ord &lt;-<span class="st"> </span><span class="kw">order</span>(res_EV_H<span class="op">$</span>padj, <span class="dt">decreasing =</span> <span class="ot">FALSE</span>)
<span class="cf">for</span> (i <span class="cf">in</span> <span class="kw">head</span>(ord, nBestFeatures)) {
    <span class="kw">print</span>(<span class="kw">plotCounts_gg</span>(i, <span class="dt">dds =</span> wald.test, 
        <span class="dt">intgroup =</span> <span class="kw">c</span>(<span class="st">&quot;condition&quot;</span>)))
}</code></pre></div>
<p><img src="_main_files/figure-html/code%20block%2049-1.png" width="672" /><img src="_main_files/figure-html/code%20block%2049-2.png" width="672" /><img src="_main_files/figure-html/code%20block%2049-3.png" width="672" /><img src="_main_files/figure-html/code%20block%2049-4.png" width="672" /><img src="_main_files/figure-html/code%20block%2049-5.png" width="672" /><img src="_main_files/figure-html/code%20block%2049-6.png" width="672" /><img src="_main_files/figure-html/code%20block%2049-7.png" width="672" /><img src="_main_files/figure-html/code%20block%2049-8.png" width="672" /><img src="_main_files/figure-html/code%20block%2049-9.png" width="672" /><img src="_main_files/figure-html/code%20block%2049-10.png" width="672" /><img src="_main_files/figure-html/code%20block%2049-11.png" width="672" /><img src="_main_files/figure-html/code%20block%2049-12.png" width="672" /><img src="_main_files/figure-html/code%20block%2049-13.png" width="672" /><img src="_main_files/figure-html/code%20block%2049-14.png" width="672" /><img src="_main_files/figure-html/code%20block%2049-15.png" width="672" /><img src="_main_files/figure-html/code%20block%2049-16.png" width="672" /><img src="_main_files/figure-html/code%20block%2049-17.png" width="672" /><img src="_main_files/figure-html/code%20block%2049-18.png" width="672" /><img src="_main_files/figure-html/code%20block%2049-19.png" width="672" /><img src="_main_files/figure-html/code%20block%2049-20.png" width="672" /></p>
</div>
</div>
<div id="select-genes-based-on-fdr-and-make-heatmap-4" class="section level5">
<h5><span class="header-section-number">0.3.9.6.2</span> Select genes based on FDR and make heatmap</h5>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">  gene.kept &lt;-<span class="st"> </span><span class="kw">rownames</span>(res_EV_H)[res_EV_H<span class="op">$</span>padj <span class="op">&lt;=</span><span class="st"> </span>p_cutoff <span class="op">&amp;</span><span class="st"> </span><span class="op">!</span><span class="kw">is.na</span>(res_EV_H<span class="op">$</span>padj) <span class="op">&amp;</span><span class="st"> </span>( res_EV_H<span class="op">$</span>log2FoldChange <span class="op">&lt;=</span><span class="st"> </span><span class="op">-</span><span class="fl">0.5</span> <span class="op">|</span><span class="st"> </span>res_EV_H<span class="op">$</span>log2FoldChange <span class="op">&gt;=</span><span class="st"> </span><span class="fl">0.5</span>)]
  count.table.kept &lt;-<span class="st"> </span><span class="kw">log2</span>(count.table <span class="op">+</span><span class="st"> </span>epsilon)[gene.kept, ]
  <span class="kw">heatmap.2</span>(<span class="kw">as.matrix</span>(count.table.kept), 
            <span class="dt">scale=</span><span class="st">&quot;row&quot;</span>, 
            <span class="dt">hclust=</span><span class="cf">function</span>(x) <span class="kw">hclust</span>(x,<span class="dt">method=</span><span class="st">&quot;average&quot;</span>), 
            <span class="dt">distfun=</span><span class="cf">function</span>(x) <span class="kw">as.dist</span>((<span class="dv">1</span><span class="op">-</span><span class="kw">cor</span>(<span class="kw">t</span>(x)))<span class="op">/</span><span class="dv">2</span>), 
            <span class="dt">trace=</span><span class="st">&quot;none&quot;</span>, 
            <span class="dt">density=</span><span class="st">&quot;none&quot;</span>, 
            <span class="co">#labRow=&quot;&quot;,</span>
            <span class="dt">cexCol=</span><span class="fl">0.7</span>)</code></pre></div>
<p><img src="_main_files/figure-html/code%20black%2050-1.png" width="960" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">  <span class="kw">invisible</span>(<span class="kw">dev.off</span>())</code></pre></div>
</div>
<div id="do-functional-enrichment-4" class="section level5">
<h5><span class="header-section-number">0.3.9.6.3</span> Do functional enrichment</h5>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">res_EV_H.df &lt;-<span class="st"> </span><span class="kw">na.omit</span>(<span class="kw">data.frame</span>(res_EV_H))
induced.sign &lt;-<span class="st"> </span><span class="kw">rownames</span>(res_EV_H.df)[res_EV_H.df<span class="op">$</span>log2FoldChange <span class="op">&gt;=</span><span class="st"> </span>
<span class="st">    </span><span class="fl">0.5</span> <span class="op">&amp;</span><span class="st"> </span>res_EV_H.df<span class="op">$</span>padj <span class="op">&lt;</span><span class="st"> </span>p_cutoff]
<span class="co"># head(induced.sign)</span>
<span class="co"># names(term.induced)</span>
<span class="cf">if</span> (<span class="kw">identical</span>(induced.sign, <span class="kw">character</span>(<span class="dv">0</span>))) {
    <span class="kw">cat</span>(<span class="st">&quot;No genes found that have induced expression&quot;</span>)
} <span class="cf">else</span> {
    term.induced &lt;-<span class="st"> </span><span class="kw">gprofiler</span>(<span class="dt">query =</span> induced.sign, 
        <span class="dt">organism =</span> <span class="st">&quot;hsapiens&quot;</span>)
    term.induced &lt;-<span class="st"> </span>term.induced[<span class="kw">order</span>(term.induced<span class="op">$</span>p.value), 
        ]
    <span class="co"># term.induced$p.value</span>
    <span class="kw">kable</span>(term.induced[<span class="dv">1</span><span class="op">:</span><span class="dv">10</span>, <span class="kw">c</span>(<span class="st">&quot;term.name&quot;</span>, 
        <span class="st">&quot;term.size&quot;</span>, <span class="st">&quot;query.size&quot;</span>, 
        <span class="st">&quot;overlap.size&quot;</span>, <span class="st">&quot;recall&quot;</span>, 
        <span class="st">&quot;precision&quot;</span>, <span class="st">&quot;p.value&quot;</span>, 
        <span class="st">&quot;intersection&quot;</span>)], <span class="dt">format.args =</span> <span class="kw">c</span>(<span class="dt">engeneer =</span> <span class="ot">TRUE</span>, 
        <span class="dt">digits =</span> <span class="dv">3</span>), <span class="dt">caption =</span> <span class="st">&quot;**Table: induced gene functional analysis wit gProfileR. ** &quot;</span>)
}</code></pre></div>
<table>
<caption>(#tab:code block 51)<strong>Table: induced gene functional analysis wit gProfileR. </strong></caption>
<thead>
<tr class="header">
<th></th>
<th align="left">term.name</th>
<th align="right">term.size</th>
<th align="right">query.size</th>
<th align="right">overlap.size</th>
<th align="right">recall</th>
<th align="right">precision</th>
<th align="right">p.value</th>
<th align="left">intersection</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>36</td>
<td align="left">Growth hormone receptor signaling</td>
<td align="right">24</td>
<td align="right">35</td>
<td align="right">5</td>
<td align="right">0.208</td>
<td align="right">0.143</td>
<td align="right">3.80e-06</td>
<td align="left">CISH,SOCS2,SOCS3,SOCS1,IRS2</td>
</tr>
<tr class="even">
<td>35</td>
<td align="left">Cytokine Signaling in Immune system</td>
<td align="right">685</td>
<td align="right">35</td>
<td align="right">12</td>
<td align="right">0.018</td>
<td align="right">0.343</td>
<td align="right">3.36e-04</td>
<td align="left">OSM,CISH,EGR1,SOCS2,CD80,CDKN1A,PIM1,EIF4A1,FOS,SOCS3,SOCS1,IRS2</td>
</tr>
<tr class="odd">
<td>39</td>
<td align="left">Interleukin-4 and 13 signaling</td>
<td align="right">111</td>
<td align="right">35</td>
<td align="right">6</td>
<td align="right">0.054</td>
<td align="right">0.171</td>
<td align="right">4.71e-04</td>
<td align="left">OSM,CDKN1A,PIM1,FOS,SOCS3,SOCS1</td>
</tr>
<tr class="even">
<td>37</td>
<td align="left">Signaling by Interleukins</td>
<td align="right">463</td>
<td align="right">35</td>
<td align="right">10</td>
<td align="right">0.022</td>
<td align="right">0.286</td>
<td align="right">5.17e-04</td>
<td align="left">OSM,CISH,SOCS2,CD80,CDKN1A,PIM1,FOS,SOCS3,SOCS1,IRS2</td>
</tr>
<tr class="odd">
<td>34</td>
<td align="left">Jak-STAT signaling pathway</td>
<td align="right">162</td>
<td align="right">35</td>
<td align="right">7</td>
<td align="right">0.043</td>
<td align="right">0.200</td>
<td align="right">6.45e-04</td>
<td align="left">OSM,CISH,SOCS2,CDKN1A,PIM1,SOCS3,SOCS1</td>
</tr>
<tr class="even">
<td>14</td>
<td align="left">negative regulation of kinase activity</td>
<td align="right">272</td>
<td align="right">66</td>
<td align="right">9</td>
<td align="right">0.033</td>
<td align="right">0.136</td>
<td align="right">1.01e-03</td>
<td align="left">GADD45B,CISH,SOCS2,CDKN1A,DUSP2,SH3BP5L,SOCS3,SOCS1,IRS2</td>
</tr>
<tr class="odd">
<td>32</td>
<td align="left">Prolactin signaling pathway</td>
<td align="right">70</td>
<td align="right">35</td>
<td align="right">5</td>
<td align="right">0.071</td>
<td align="right">0.143</td>
<td align="right">1.27e-03</td>
<td align="left">CISH,SOCS2,FOS,SOCS3,SOCS1</td>
</tr>
<tr class="even">
<td>38</td>
<td align="left">Interleukin-7 signaling</td>
<td align="right">36</td>
<td align="right">35</td>
<td align="right">4</td>
<td align="right">0.111</td>
<td align="right">0.114</td>
<td align="right">1.74e-03</td>
<td align="left">CISH,SOCS2,SOCS1,IRS2</td>
</tr>
<tr class="odd">
<td>13</td>
<td align="left">negative regulation of transferase activity</td>
<td align="right">298</td>
<td align="right">66</td>
<td align="right">9</td>
<td align="right">0.030</td>
<td align="right">0.136</td>
<td align="right">2.18e-03</td>
<td align="left">GADD45B,CISH,SOCS2,CDKN1A,DUSP2,SH3BP5L,SOCS3,SOCS1,IRS2</td>
</tr>
<tr class="even">
<td>31</td>
<td align="left">protein kinase inhibitor activity</td>
<td align="right">95</td>
<td align="right">66</td>
<td align="right">6</td>
<td align="right">0.063</td>
<td align="right">0.091</td>
<td align="right">2.52e-03</td>
<td align="left">CISH,SOCS2,CDKN1A,SH3BP5L,SOCS3,SOCS1</td>
</tr>
</tbody>
</table>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">repressed.sign &lt;-<span class="st"> </span><span class="kw">rownames</span>(res_EV_H.df)[res_EV_H.df<span class="op">$</span>log2FoldChange <span class="op">&lt;=</span><span class="st"> </span>
<span class="st">    </span><span class="op">-</span><span class="fl">0.5</span> <span class="op">&amp;</span><span class="st"> </span>res_EV_H.df<span class="op">$</span>padj <span class="op">&lt;</span><span class="st"> </span>p_cutoff]
<span class="cf">if</span> (<span class="kw">identical</span>(repressed.sign, <span class="kw">character</span>(<span class="dv">0</span>))) {
    <span class="kw">cat</span>(<span class="st">&quot;No genes found that have repressed expression&quot;</span>)
} <span class="cf">else</span> {
    term.repressed &lt;-<span class="st"> </span><span class="kw">gprofiler</span>(<span class="dt">query =</span> repressed.sign, 
        <span class="dt">organism =</span> <span class="st">&quot;hsapiens&quot;</span>)
    term.repressed &lt;-<span class="st"> </span>term.repressed[<span class="kw">order</span>(term.repressed<span class="op">$</span>p.value), 
        ]
    <span class="kw">kable</span>(term.repressed[<span class="dv">1</span><span class="op">:</span><span class="dv">10</span>, 
        <span class="kw">c</span>(<span class="st">&quot;term.name&quot;</span>, <span class="st">&quot;term.size&quot;</span>, 
            <span class="st">&quot;query.size&quot;</span>, <span class="st">&quot;overlap.size&quot;</span>, 
            <span class="st">&quot;recall&quot;</span>, <span class="st">&quot;precision&quot;</span>, 
            <span class="st">&quot;p.value&quot;</span>, <span class="st">&quot;intersection&quot;</span>)], 
        <span class="dt">format.args =</span> <span class="kw">c</span>(<span class="dt">engeneer =</span> <span class="ot">TRUE</span>, 
            <span class="dt">digits =</span> <span class="dv">3</span>), <span class="dt">caption =</span> <span class="st">&quot;**Table: repressed genes functional analysis with gProfileR. ** &quot;</span>)
}</code></pre></div>
<table>
<caption>(#tab:code block 51)<strong>Table: repressed genes functional analysis with gProfileR. </strong></caption>
<thead>
<tr class="header">
<th></th>
<th align="left">term.name</th>
<th align="right">term.size</th>
<th align="right">query.size</th>
<th align="right">overlap.size</th>
<th align="right">recall</th>
<th align="right">precision</th>
<th align="right">p.value</th>
<th align="left">intersection</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>16</td>
<td align="left">TNF signaling pathway</td>
<td align="right">108</td>
<td align="right">17</td>
<td align="right">7</td>
<td align="right">0.065</td>
<td align="right">0.412</td>
<td align="right">1.00e-07</td>
<td align="left">BIRC3,TRAF1,BCL3,ICAM1,NFKBIA,TNFAIP3,JUN</td>
</tr>
<tr class="even">
<td>15</td>
<td align="left">NF-kappa B signaling pathway</td>
<td align="right">93</td>
<td align="right">17</td>
<td align="right">5</td>
<td align="right">0.054</td>
<td align="right">0.294</td>
<td align="right">9.21e-05</td>
<td align="left">BIRC3,TRAF1,ICAM1,NFKBIA,TNFAIP3</td>
</tr>
<tr class="odd">
<td>4</td>
<td align="left">cellular response to tumor necrosis factor</td>
<td align="right">290</td>
<td align="right">24</td>
<td align="right">7</td>
<td align="right">0.024</td>
<td align="right">0.292</td>
<td align="right">9.55e-05</td>
<td align="left">BIRC3,TNFRSF9,TRAF1,ICAM1,NFKBIA,TNFAIP3,ZFP36L2</td>
</tr>
<tr class="even">
<td>3</td>
<td align="left">response to tumor necrosis factor</td>
<td align="right">307</td>
<td align="right">24</td>
<td align="right">7</td>
<td align="right">0.023</td>
<td align="right">0.292</td>
<td align="right">1.41e-04</td>
<td align="left">BIRC3,TNFRSF9,TRAF1,ICAM1,NFKBIA,TNFAIP3,ZFP36L2</td>
</tr>
<tr class="odd">
<td>7</td>
<td align="left">liver development</td>
<td align="right">139</td>
<td align="right">24</td>
<td align="right">5</td>
<td align="right">0.036</td>
<td align="right">0.208</td>
<td align="right">1.47e-03</td>
<td align="left">NFKBIA,TNFAIP3,SLCO2B1,CITED2,JUN</td>
</tr>
<tr class="even">
<td>21</td>
<td align="left">TNFR1-induced NFkappaB signaling pathway</td>
<td align="right">26</td>
<td align="right">15</td>
<td align="right">3</td>
<td align="right">0.115</td>
<td align="right">0.200</td>
<td align="right">1.53e-03</td>
<td align="left">BIRC3,TRAF1,TNFAIP3</td>
</tr>
<tr class="odd">
<td>6</td>
<td align="left">hepaticobiliary system development</td>
<td align="right">142</td>
<td align="right">24</td>
<td align="right">5</td>
<td align="right">0.035</td>
<td align="right">0.208</td>
<td align="right">1.64e-03</td>
<td align="left">NFKBIA,TNFAIP3,SLCO2B1,CITED2,JUN</td>
</tr>
<tr class="even">
<td>22</td>
<td align="left">Regulation of TNFR1 signaling</td>
<td align="right">33</td>
<td align="right">15</td>
<td align="right">3</td>
<td align="right">0.091</td>
<td align="right">0.200</td>
<td align="right">3.20e-03</td>
<td align="left">BIRC3,TRAF1,TNFAIP3</td>
</tr>
<tr class="odd">
<td>13</td>
<td align="left">Epstein-Barr virus infection</td>
<td align="right">197</td>
<td align="right">17</td>
<td align="right">5</td>
<td align="right">0.025</td>
<td align="right">0.294</td>
<td align="right">3.60e-03</td>
<td align="left">TRAF1,ICAM1,NFKBIA,TNFAIP3,JUN</td>
</tr>
<tr class="even">
<td>5</td>
<td align="left">tumor necrosis factor-mediated signaling pathway</td>
<td align="right">174</td>
<td align="right">24</td>
<td align="right">5</td>
<td align="right">0.029</td>
<td align="right">0.208</td>
<td align="right">4.46e-03</td>
<td align="left">BIRC3,TNFRSF9,TRAF1,NFKBIA,TNFAIP3</td>
</tr>
</tbody>
</table>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># kable(head(term.induced[,c(&#39;p.value&#39;,</span>
<span class="co"># &#39;term.name&#39;,&#39;intersection&#39;)],</span>
<span class="co"># 10))</span></code></pre></div>
</div>
</div>
<div id="epo-with-hmgb1-eh-vs-hmgb1-h" class="section level4">
<h4><span class="header-section-number">0.3.9.7</span> EPO with HMGB1 (EH) vs HMGB1 (H)</h4>
<div id="plots-5" class="section level5 tabset tabset-fade tabset-pills">
<h5><span class="header-section-number">0.3.9.7.1</span> Plots</h5>
<div id="ma-plot-5" class="section level6">
<h6><span class="header-section-number">0.3.9.7.1.1</span> MA-plot</h6>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">res_EH_H &lt;-<span class="st"> </span><span class="kw">results</span>(wald.test, 
    <span class="dt">contrast =</span> <span class="kw">c</span>(<span class="st">&quot;condition&quot;</span>, <span class="st">&quot;EH&quot;</span>, 
        <span class="st">&quot;H&quot;</span>), <span class="dt">alpha =</span> p_cutoff, 
    <span class="dt">pAdjustMethod =</span> <span class="st">&quot;BH&quot;</span>)
res_EH_H_ashr &lt;-<span class="st"> </span><span class="kw">lfcShrink</span>(wald.test, 
    <span class="dt">contrast =</span> <span class="kw">c</span>(<span class="st">&quot;condition&quot;</span>, <span class="st">&quot;EH&quot;</span>, 
        <span class="st">&quot;H&quot;</span>), <span class="dt">res =</span> res_EH_H, <span class="dt">type =</span> <span class="st">&quot;ashr&quot;</span>)
<span class="kw">par</span>(<span class="dt">mfrow =</span> <span class="kw">c</span>(<span class="dv">1</span>, <span class="dv">2</span>))
<span class="kw">plotMA</span>(res_EH_H, <span class="dt">xlim =</span> xlim, <span class="dt">ylim =</span> ylim, 
    <span class="dt">main =</span> <span class="st">&quot;normal&quot;</span>)
<span class="kw">plotMA</span>(res_EH_H_ashr, <span class="dt">xlim =</span> xlim, 
    <span class="dt">ylim =</span> ylim, <span class="dt">main =</span> <span class="st">&quot;ashr&quot;</span>)</code></pre></div>
<p><img src="_main_files/figure-html/code%20block%2052-1.png" width="672" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">invisible</span>(<span class="kw">dev.off</span>())</code></pre></div>
</div>
<div id="historgram-of-adjusted-p-values-5" class="section level6">
<h6><span class="header-section-number">0.3.9.7.1.2</span> Historgram of adjusted p-values</h6>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">hist</span>(res_EH_H<span class="op">$</span>padj, <span class="dt">breaks =</span> <span class="dv">20</span>, 
    <span class="dt">col =</span> <span class="st">&quot;grey&quot;</span>, <span class="dt">main =</span> <span class="st">&quot;DESeq2 p-value distribution&quot;</span>, 
    <span class="dt">xlab =</span> <span class="st">&quot;DESeq2 P-value&quot;</span>, <span class="dt">ylab =</span> <span class="st">&quot;Number of genes&quot;</span>)</code></pre></div>
<p><img src="_main_files/figure-html/code%20block%2053-1.png" width="672" /></p>
</div>
<div id="volacano-plot-5" class="section level6">
<h6><span class="header-section-number">0.3.9.7.1.3</span> Volacano-plot</h6>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">resOrdered_EH_H &lt;-<span class="st"> </span>res_EH_H[<span class="kw">order</span>(res_EH_H<span class="op">$</span>padj), 
    ]
<span class="kw">write.csv</span>(<span class="kw">as.data.frame</span>(resOrdered_EH_H), 
    <span class="dt">file =</span> <span class="st">&quot;condition_EH_H_alpha0.1_results.csv&quot;</span>)
EH_H_p1 &lt;-<span class="st"> </span><span class="kw">enhanced_volcano_plots</span>(<span class="st">&quot;EH vs. H (padj=0.05,log2fc=0.5)&quot;</span>, 
    res_EH_H, <span class="fl">0.05</span>, fc_cutoff)
EH_H_p2 &lt;-<span class="st"> </span><span class="kw">enhanced_volcano_plots</span>(<span class="st">&quot;EH vs. H (padj=0.1,log2fc=0.5)&quot;</span>, 
    res_EH_H, p_cutoff, fc_cutoff)
<span class="kw">grid.arrange</span>(EH_H_p1, EH_H_p2, 
    <span class="dt">nrow =</span> <span class="dv">2</span>, <span class="dt">ncol =</span> <span class="dv">1</span>)
<span class="kw">grid.rect</span>(<span class="dt">gp =</span> <span class="kw">gpar</span>(<span class="dt">fill =</span> <span class="ot">NA</span>))</code></pre></div>
<p><img src="_main_files/figure-html/code%20block%2054-1.png" width="960" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">invisible</span>(<span class="kw">dev.off</span>())</code></pre></div>
</div>
<div id="plot-counts-5" class="section level6">
<h6><span class="header-section-number">0.3.9.7.1.4</span> Plot counts</h6>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">nBestFeatures =<span class="st"> </span><span class="dv">20</span>
ord &lt;-<span class="st"> </span><span class="kw">order</span>(res_EH_H<span class="op">$</span>padj, <span class="dt">decreasing =</span> <span class="ot">FALSE</span>)
<span class="cf">for</span> (i <span class="cf">in</span> <span class="kw">head</span>(ord, nBestFeatures)) {
    <span class="kw">print</span>(<span class="kw">plotCounts_gg</span>(i, <span class="dt">dds =</span> wald.test, 
        <span class="dt">intgroup =</span> <span class="kw">c</span>(<span class="st">&quot;condition&quot;</span>)))
}</code></pre></div>
<p><img src="_main_files/figure-html/code%20block%2055-1.png" width="672" /><img src="_main_files/figure-html/code%20block%2055-2.png" width="672" /><img src="_main_files/figure-html/code%20block%2055-3.png" width="672" /><img src="_main_files/figure-html/code%20block%2055-4.png" width="672" /><img src="_main_files/figure-html/code%20block%2055-5.png" width="672" /><img src="_main_files/figure-html/code%20block%2055-6.png" width="672" /><img src="_main_files/figure-html/code%20block%2055-7.png" width="672" /><img src="_main_files/figure-html/code%20block%2055-8.png" width="672" /><img src="_main_files/figure-html/code%20block%2055-9.png" width="672" /><img src="_main_files/figure-html/code%20block%2055-10.png" width="672" /><img src="_main_files/figure-html/code%20block%2055-11.png" width="672" /><img src="_main_files/figure-html/code%20block%2055-12.png" width="672" /><img src="_main_files/figure-html/code%20block%2055-13.png" width="672" /><img src="_main_files/figure-html/code%20block%2055-14.png" width="672" /><img src="_main_files/figure-html/code%20block%2055-15.png" width="672" /><img src="_main_files/figure-html/code%20block%2055-16.png" width="672" /><img src="_main_files/figure-html/code%20block%2055-17.png" width="672" /><img src="_main_files/figure-html/code%20block%2055-18.png" width="672" /><img src="_main_files/figure-html/code%20block%2055-19.png" width="672" /><img src="_main_files/figure-html/code%20block%2055-20.png" width="672" /></p>
</div>
</div>
<div id="select-genes-based-on-fdr-and-make-heatmap-5" class="section level5">
<h5><span class="header-section-number">0.3.9.7.2</span> Select genes based on FDR and make heatmap</h5>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">  gene.kept &lt;-<span class="st"> </span><span class="kw">rownames</span>(res_EH_H)[res_EH_H<span class="op">$</span>padj <span class="op">&lt;=</span><span class="st"> </span>p_cutoff <span class="op">&amp;</span><span class="st"> </span><span class="op">!</span><span class="kw">is.na</span>(res_EH_H<span class="op">$</span>padj) <span class="op">&amp;</span><span class="st"> </span>( res_EH_H<span class="op">$</span>log2FoldChange <span class="op">&lt;=</span><span class="st"> </span><span class="op">-</span><span class="fl">0.5</span> <span class="op">|</span><span class="st"> </span>res_EH_H<span class="op">$</span>log2FoldChange <span class="op">&gt;=</span><span class="st"> </span><span class="fl">0.5</span>)]
  count.table.kept &lt;-<span class="st"> </span><span class="kw">log2</span>(count.table <span class="op">+</span><span class="st"> </span>epsilon)[gene.kept, ]
  <span class="kw">heatmap.2</span>(<span class="kw">as.matrix</span>(count.table.kept), 
            <span class="dt">scale=</span><span class="st">&quot;row&quot;</span>, 
            <span class="dt">hclust=</span><span class="cf">function</span>(x) <span class="kw">hclust</span>(x,<span class="dt">method=</span><span class="st">&quot;average&quot;</span>), 
            <span class="dt">distfun=</span><span class="cf">function</span>(x) <span class="kw">as.dist</span>((<span class="dv">1</span><span class="op">-</span><span class="kw">cor</span>(<span class="kw">t</span>(x)))<span class="op">/</span><span class="dv">2</span>), 
            <span class="dt">trace=</span><span class="st">&quot;none&quot;</span>, 
            <span class="dt">density=</span><span class="st">&quot;none&quot;</span>, 
            <span class="co">#labRow=&quot;&quot;,</span>
            <span class="dt">cexCol=</span><span class="fl">0.7</span>)</code></pre></div>
<p><img src="_main_files/figure-html/code%20black%2056-1.png" width="960" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">  <span class="kw">invisible</span>(<span class="kw">dev.off</span>())</code></pre></div>
</div>
<div id="do-functional-enrichment-5" class="section level5">
<h5><span class="header-section-number">0.3.9.7.3</span> Do functional enrichment</h5>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">res_EH_H.df &lt;-<span class="st"> </span><span class="kw">na.omit</span>(<span class="kw">data.frame</span>(res_EH_H))
induced.sign &lt;-<span class="st"> </span><span class="kw">rownames</span>(res_EH_H.df)[res_EH_H.df<span class="op">$</span>log2FoldChange <span class="op">&gt;=</span><span class="st"> </span>
<span class="st">    </span><span class="fl">0.5</span> <span class="op">&amp;</span><span class="st"> </span>res_EH_H.df<span class="op">$</span>padj <span class="op">&lt;</span><span class="st"> </span>p_cutoff]
<span class="co"># head(induced.sign)</span>
<span class="co"># names(term.induced)</span>
<span class="cf">if</span> (<span class="kw">identical</span>(induced.sign, <span class="kw">character</span>(<span class="dv">0</span>))) {
    <span class="kw">cat</span>(<span class="st">&quot;No genes found that have induced expression&quot;</span>)
} <span class="cf">else</span> {
    term.induced &lt;-<span class="st"> </span><span class="kw">gprofiler</span>(<span class="dt">query =</span> induced.sign, 
        <span class="dt">organism =</span> <span class="st">&quot;hsapiens&quot;</span>)
    term.induced &lt;-<span class="st"> </span>term.induced[<span class="kw">order</span>(term.induced<span class="op">$</span>p.value), 
        ]
    <span class="co"># term.induced$p.value</span>
    <span class="kw">kable</span>(term.induced[<span class="dv">1</span><span class="op">:</span><span class="dv">10</span>, <span class="kw">c</span>(<span class="st">&quot;term.name&quot;</span>, 
        <span class="st">&quot;term.size&quot;</span>, <span class="st">&quot;query.size&quot;</span>, 
        <span class="st">&quot;overlap.size&quot;</span>, <span class="st">&quot;recall&quot;</span>, 
        <span class="st">&quot;precision&quot;</span>, <span class="st">&quot;p.value&quot;</span>, 
        <span class="st">&quot;intersection&quot;</span>)], <span class="dt">format.args =</span> <span class="kw">c</span>(<span class="dt">engeneer =</span> <span class="ot">TRUE</span>, 
        <span class="dt">digits =</span> <span class="dv">3</span>), <span class="dt">caption =</span> <span class="st">&quot;**Table: induced gene functional analysis wit gProfileR. ** &quot;</span>)
}</code></pre></div>
<table>
<caption>(#tab:code block 57)<strong>Table: induced gene functional analysis wit gProfileR. </strong></caption>
<thead>
<tr class="header">
<th></th>
<th align="left">term.name</th>
<th align="right">term.size</th>
<th align="right">query.size</th>
<th align="right">overlap.size</th>
<th align="right">recall</th>
<th align="right">precision</th>
<th align="right">p.value</th>
<th align="left">intersection</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>3</td>
<td align="left">Jak-STAT signaling pathway</td>
<td align="right">162</td>
<td align="right">6</td>
<td align="right">4</td>
<td align="right">0.025</td>
<td align="right">0.667</td>
<td align="right">0.000146</td>
<td align="left">OSM,CISH,SOCS2,PIM1</td>
</tr>
<tr class="even">
<td>4</td>
<td align="left">Cytokine Signaling in Immune system</td>
<td align="right">685</td>
<td align="right">7</td>
<td align="right">5</td>
<td align="right">0.007</td>
<td align="right">0.714</td>
<td align="right">0.004300</td>
<td align="left">OSM,CISH,EGR1,SOCS2,PIM1</td>
</tr>
<tr class="odd">
<td>7</td>
<td align="left">Growth hormone receptor signaling</td>
<td align="right">24</td>
<td align="right">7</td>
<td align="right">2</td>
<td align="right">0.083</td>
<td align="right">0.286</td>
<td align="right">0.020700</td>
<td align="left">CISH,SOCS2</td>
</tr>
<tr class="even">
<td>5</td>
<td align="left">Signaling by Interleukins</td>
<td align="right">463</td>
<td align="right">7</td>
<td align="right">4</td>
<td align="right">0.009</td>
<td align="right">0.571</td>
<td align="right">0.023100</td>
<td align="left">OSM,CISH,SOCS2,PIM1</td>
</tr>
<tr class="odd">
<td>6</td>
<td align="left">Interleukin-7 signaling</td>
<td align="right">36</td>
<td align="right">7</td>
<td align="right">2</td>
<td align="right">0.056</td>
<td align="right">0.286</td>
<td align="right">0.047100</td>
<td align="left">CISH,SOCS2</td>
</tr>
<tr class="even">
<td>1</td>
<td align="left">EGR-EP300 complex</td>
<td align="right">2</td>
<td align="right">2</td>
<td align="right">1</td>
<td align="right">0.500</td>
<td align="right">0.500</td>
<td align="right">0.049800</td>
<td align="left">EGR1</td>
</tr>
<tr class="odd">
<td>2</td>
<td align="left">RCP-Rab11 complex</td>
<td align="right">2</td>
<td align="right">2</td>
<td align="right">1</td>
<td align="right">0.500</td>
<td align="right">0.500</td>
<td align="right">0.049800</td>
<td align="left">RAB11FIP1</td>
</tr>
<tr class="even">
<td>NA</td>
<td align="left">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td>NA.1</td>
<td align="left">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td>NA.2</td>
<td align="left">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="left">NA</td>
</tr>
</tbody>
</table>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">repressed.sign &lt;-<span class="st"> </span><span class="kw">rownames</span>(res_EH_H.df)[res_EH_H.df<span class="op">$</span>log2FoldChange <span class="op">&lt;=</span><span class="st"> </span>
<span class="st">    </span><span class="op">-</span><span class="fl">0.5</span> <span class="op">&amp;</span><span class="st"> </span>res_EH_H.df<span class="op">$</span>padj <span class="op">&lt;</span><span class="st"> </span>p_cutoff]
<span class="cf">if</span> (<span class="kw">identical</span>(repressed.sign, <span class="kw">character</span>(<span class="dv">0</span>))) {
    <span class="kw">cat</span>(<span class="st">&quot;No genes found that have repressed expression&quot;</span>)
} <span class="cf">else</span> {
    term.repressed &lt;-<span class="st"> </span><span class="kw">gprofiler</span>(<span class="dt">query =</span> repressed.sign, 
        <span class="dt">organism =</span> <span class="st">&quot;hsapiens&quot;</span>)
    term.repressed &lt;-<span class="st"> </span>term.repressed[<span class="kw">order</span>(term.repressed<span class="op">$</span>p.value), 
        ]
    <span class="kw">kable</span>(term.repressed[<span class="dv">1</span><span class="op">:</span><span class="dv">10</span>, 
        <span class="kw">c</span>(<span class="st">&quot;term.name&quot;</span>, <span class="st">&quot;term.size&quot;</span>, 
            <span class="st">&quot;query.size&quot;</span>, <span class="st">&quot;overlap.size&quot;</span>, 
            <span class="st">&quot;recall&quot;</span>, <span class="st">&quot;precision&quot;</span>, 
            <span class="st">&quot;p.value&quot;</span>, <span class="st">&quot;intersection&quot;</span>)], 
        <span class="dt">format.args =</span> <span class="kw">c</span>(<span class="dt">engeneer =</span> <span class="ot">TRUE</span>, 
            <span class="dt">digits =</span> <span class="dv">3</span>), <span class="dt">caption =</span> <span class="st">&quot;**Table: repressed genes functional analysis with gProfileR. ** &quot;</span>)
}</code></pre></div>
<pre><code>## No genes found that have repressed expression</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># kable(head(term.induced[,c(&#39;p.value&#39;,</span>
<span class="co"># &#39;term.name&#39;,&#39;intersection&#39;)],</span>
<span class="co"># 10))</span></code></pre></div>

</div>
</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>


    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"google": false,
"linkedin": false,
"weibo": false,
"instapper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
