---
title: '&quot;Clustering and differential analysis of CFU-E cells with different conditions&quot;'
author: "Ronak Shah"
date: "July 27th 2017"
output:
  html_document:
    code_download: yes
    code_folding: hide
    highlight: pygments
    keep_md: yes
    theme: paper
    toc: yes
    toc_float: yes
    toc_depth: 10
  word_document:
    toc: yes
    toc_depth: 10
  pdf_document:
    toc: yes
    toc_depth: 10
---

```{r setup, include=FALSE}
library(knitr)
opts_chunk$set(tidy.opts=list(width.cutoff=30,blank=FALSE), tidy=TRUE, echo = TRUE)
devtools::source_gist("c83e078bf8c81b035e32c3fc0cf04ee8", 
                      filename = 'render_toc.R')
```

## Table of Contents

```{r toc, echo=FALSE} 
render_toc("deseq2_analysis.Rmd")
```

```{r code block 1, message=FALSE, warning=FALSE}
#Load libraries
set.seed(0)
source("https://bioconductor.org/biocLite.R")
library("DESeq2")
library("RColorBrewer")
library("vsn")
library("pheatmap")
library("EnhancedVolcano")
library("gridExtra")
library("grid")
library("ggplot2")
library("gplots")
library("affy")
library("reshape2")
library("gProfileR")
library("ComplexHeatmap")
library("kableExtra")
```
***
## Summary of Experiment:

1. Flow-sorted Colony-Forming Units of Erythroid (CFU-E) cells. The cells are then starved for six hours to stop the Erythropoietin (EPO). Then these cells are treated with:
    - EPO + Vehicle (Peripheral Blood Smear (PBS))
    - EPO + High mobility group box 1 (HMGB1) protein
    - HMGB1

2. The treatment occurs for sixty minutes and then RNA was extracted. Three biological replicates are used for reproducibility.

3. Single-end stranded RNA-seq performed on wild-type EPO starved cells and treated with 1,2,3 as stated above.

4. Labels:
    - WT => Untreated
    - EV => EPO + Vehicle
    - EH => EPO + HMGB1
    - H => HMGB1
    - E1 => Samples from Patient 1
    - E2 => Samples from Patient 2
    - E3 => Samples from Patient 3

## Exploring data with DESeq2

<!-- ### Basic DESeq2 setup -->

```{r code block 2, message=FALSE,warning=FALSE,cache=TRUE}
  directory <- "/Users/rshah22/Documents/Projects/RNAseq_Erythrocid/counts_analysis/"
  files<-read.table(paste(directory,"deseq2_sampleinfo.txt",sep=""),header=T)
  condition<-read.csv(paste(directory,"deseq2_phenotype.txt",sep=""),header = T)
  sampleTable <- data.frame(sampleName = files$sampleName,
                            fileName = files$fileName,
                            condition.type = condition$condition,
                            condition.treatment = condition$treatment,
                            condition.experiment = condition$experiment)


  ddsHTSeq <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTable,
                                        directory = directory,
                                        design= ~condition.type + condition.experiment)
  #ddsHTSeq
  ddsHTSeq$condition.type <- factor(ddsHTSeq$condition.type, levels = c("WT","EV","EH","H"))
  ddsHTSeq$condition.experiment <- factor(ddsHTSeq$condition.experiment, levels = c("E1","E2","E3"))

```


### Running DESeq2

```{r code block 3, message=FALSE, warning=FALSE,cache=TRUE}
  dds <- DESeq(ddsHTSeq)
```


### Make counts table to explore that data
```{r code block 4, warning=FALSE,cache=TRUE}
  count.table <- counts(dds)
  count.table<-count.table[-c ((grep("^HIST|RNA|^RP11|^LINC|^RPL|^RNVU",rownames(count.table)))),]
  stats.per.sample <- data.frame(t(do.call(cbind, lapply(count.table, summary))))
  stats.per.sample$libsum <- apply(count.table, 2, sum) ## libsum
  stats.per.sample$perc05 <- apply(count.table, 2, quantile, 0.05)
  stats.per.sample$perc10 <- apply(count.table, 2, quantile, 0.10)
  stats.per.sample$perc90 <- apply(count.table, 2, quantile, 0.90)
  stats.per.sample$perc95 <- apply(count.table, 2, quantile, 0.95)
  stats.per.sample$zeros <- apply(count.table==0, 2, sum)
  stats.per.sample$percent.zeros <- 100*stats.per.sample$zeros/nrow(count.table)
  kable(stats.per.sample[sample(1:ncol(count.table), size = 10),],
      caption = "**Table: statistics per sample. ** We only display a random selection of 10 samples. ") %>% kable_styling(bootstrap_options = c("striped", "hover", "condensed", "responsive"))
```

***
<!--### Assign color based on sample type -->
```{r code block 5, warning=FALSE}   
  col.sampletype <- c("WT"="#9E0142","EV"="#D0384D","EH"="#EE6445","H"="#FA9C58") 
  expDesign<-condition
  expDesign$color <- col.sampletype[as.vector(expDesign$condition)]
```

### Histograms of counts per gene {.tabset .tabset-fade .tabset-pills}
#### matrix
```{r code block 6, warning=FALSE}
  plot.new()
  hist(as.matrix(count.table), col="blue", border="white", breaks=100)
  invisible(dev.off())
```

#### truncated matrix
```{r code block 7, warning=FALSE}
  plot.new()
  hist(as.matrix(count.table), col="blue", border="white",
      breaks=20000, xlim=c(0,500), main="Counts per gene",
      xlab="Counts (truncated axis)", ylab="Number of genes", 
      las=1, cex.axis=0.7)
  invisible(dev.off())
```     

#### log2-transformation
```{r code block 8, warning=FALSE}
  epsilon <- 1 # pseudo-count to avoid problems with log(0)
  plot.new()
  hist(as.matrix(log2(count.table + epsilon)), breaks=100, col="blue", border="white",
      main="Log2-transformed counts per gene", xlab="log2(counts+1)", ylab="Number of genes", 
      las=1, cex.axis=0.7)
  invisible(dev.off())
```

#### box-plot
```{r code block 9, warning=FALSE}
  plot.new()
  boxplot(log2(count.table + epsilon), col=expDesign$color, pch=".", 
          horizontal=TRUE, cex.axis=0.5,
          las=1, ylab="Samples", xlab="log2(Counts +1)")
  invisible(dev.off())
```
#### density-plots
```{r code block 10, warning=FALSE}
  plot.new()
  plotDensity(log2(count.table + epsilon), lty=1, col=expDesign$color, lwd=2) 
  grid() 
  legend("topright", legend=names(col.sampletype), col=col.sampletype, lwd=2)
  invisible(dev.off())
```



#### scatter-plots
-  all points are aligned along the diagonal, with a relatively wider dispersion at the bottom, corresponding to small number fluctuations.
- comparing across samples types there are points discarding from the diagonal.
```{r code block 11, warning=FALSE, cache=TRUE}
  plotFun <- function(x,y){ 
    dns <- densCols(x,y); 
    points(x,y, col=dns, pch=".", panel.first=grid());  
    abline(a=0, b=1, col="brown")
  }
  plot.new()
  pairs(log2(count.table[,sample(ncol(count.table), 12)] + epsilon), 
        panel=plotFun, lower.panel = NULL)
  invisible(dev.off())
```

### Eliminate undetected genes
```{r code block 12, warning=FALSE}
  prop.null <- apply(count.table, 2, function(x) 100*mean(x==0))
  plot.new()
  barplot(prop.null, main="Percentage of null counts per sample", 
        horiz=TRUE, cex.names=0.5, las=1, 
        col=expDesign$color, ylab='Samples', xlab='% of null counts')
  invisible(dev.off())
  count.table <- count.table[rowSums(count.table) > 0,]
```

###  Re-run DESeq2 & Normalization 
```{r code block 13, message=FALSE, warning=FALSE}
    dds0 <- DESeqDataSetFromMatrix(countData = count.table, colData = expDesign, design = ~ condition + experiment)
    dds.norm <-  estimateSizeFactors(dds0)
  #sizeFactors(dds.norm)
```

#### Plot normalized data {.tabset .tabset-fade .tabset-pills}
##### box-plot
```{r code block 14, warning=FALSE}
  par(mfrow=c(1,2),cex.lab=0.7)
  boxplot(log2(counts(dds.norm)+epsilon),  col=col.sampletype, cex.axis=0.7, 
        las=1, xlab="log2(counts)", horizontal=TRUE, main="Raw counts")
  boxplot(log2(counts(dds.norm, normalized=TRUE)+epsilon),  col=col.sampletype, cex.axis=0.7, 
        las=1, xlab="log2(normalized counts)", horizontal=TRUE, main="Normalized counts") 
  invisible(dev.off())
```

##### density-plot
```{r code block 15, warning=FALSE}
  par(mfrow=c(1,2),cex.lab=0.7)
  plotDensity(log2(counts(dds.norm)+epsilon),  col=col.sampletype, 
            xlab="log2(counts)", cex.lab=0.7)
  plotDensity(log2(counts(dds.norm, normalized=TRUE)+epsilon), col=col.sampletype, 
            xlab="log2(normalized counts)", cex.lab=0.7)
  invisible(dev.off())
```

### Count variance is related to mean
```{r code block 16, warning=FALSE}
  ## Computing mean and variance
  norm.counts <- counts(dds.norm, normalized=TRUE)
  mean.counts <- rowMeans(norm.counts)
  variance.counts <- apply(norm.counts, 1, var)

## sum(mean.counts==0) # Number of completely undetected genes

  norm.counts.stats <- data.frame(
    min=apply(norm.counts, 2, min),
    mean=apply(norm.counts, 2, mean),
    median=apply(norm.counts, 2, median),
    max=apply(norm.counts, 2, max),
    zeros=apply(norm.counts==0, 2, sum),
    percent.zeros=100*apply(norm.counts==0, 2, sum)/nrow(norm.counts),
    perc05=apply(norm.counts, 2, quantile, 0.05),
    perc10=apply(norm.counts, 2, quantile, 0.10),
    perc90=apply(norm.counts, 2, quantile, 0.90),
    perc95=apply(norm.counts, 2, quantile, 0.95)
  )
  kable(norm.counts.stats,caption = "**Table: statistics for count. ** ") %>% kable_styling(bootstrap_options = c("striped", "hover", "condensed", "responsive"))
  ## Mean and variance relationship
  mean.var.col <- densCols(x=log2(mean.counts), y=log2(variance.counts))
  plot.new()
  plot(x=log2(mean.counts), y=log2(variance.counts), pch=16, cex=0.5, 
       col=mean.var.col, main="Mean-variance relationship",
       xlab="Mean log2(normalized counts) per gene",
      ylab="Variance of log2(normalized counts)",
      panel.first = grid())
  abline(a=0, b=1, col="brown")
  invisible(dev.off())
```

### Estimated Dispersion for each gene

- Shows the mean of normalized counts (x axis) and dispersion estimate for each genes
```{r code block 17, message=FALSE, warning=FALSE}
  ## Performing estimation of dispersion parameter
  dds.disp <- estimateDispersions(dds.norm)
  plotDispEsts(dds.disp)
  invisible(dev.off())
```

### PCA and heatmap on all samples using reguralized log transformation {.tabset .tabset-fade .tabset-pills}
 
 - Transforms the count data to the log2 scale in a way which minimizes differences between samples for rows with small counts, and which normalizes with respect to library size.

#### PCA using rlog
```{r code block 18, warning=FALSE}
  rld <- rlog(dds.disp, blind=TRUE)
  hmcol <- colorRampPalette(brewer.pal(11,"Spectral"))(12)
  plotPCA(rld, ntop=5000, intgroup=c("condition","experiment")) + scale_color_manual(values = hmcol )
  invisible(dev.off())
```

#### Heatmap of count matrix for first 50 genes
```{r code block 19, warning=FALSE,fig.width=12,fig.height=10}
  select <- order(rowMedians(counts(dds.disp,normalized=TRUE)),
                  decreasing=TRUE)[1:50]
  df <- as.data.frame(colData(dds.disp)[,c("condition","experiment")])
#Heatmap of count matrix
  pheatmap(assay(rld)[select,], cluster_rows=TRUE, show_rownames=TRUE,
          cluster_cols=TRUE, annotation_col=df)
  invisible(dev.off())
```

#### Heatmap of sample distance based on rlog
```{r code block 20, warning=FALSE}
  sampleDists <- dist(t(assay(rld)))
  sampleDistMatrix <- as.matrix(sampleDists)
  rownames(sampleDistMatrix) <- paste(rld$condition, rld$experiment, sep="-")
  colnames(sampleDistMatrix) <- NULL
  colors <- colorRampPalette( rev(brewer.pal(9, "Blues")) )(255)
  pheatmap(sampleDistMatrix,
           clustering_distance_rows=sampleDists,
           clustering_distance_cols=sampleDists,
           col=colors)
  invisible(dev.off())
```

### Perform differential expression call

#### Set standard cutoff and Enhanced Volcano Plots
```{r code block 21, warning=FALSE}
  p_cutoff=0.1
  fc_cutoff=0.5
  xlim <- c(1,1e5)
  ylim <- c(-6,6)
  wald.test <- nbinomWaldTest(dds.disp)
  enhanced_volcano_plots<-function(title,results,pvalue,foldchange){
    p = EnhancedVolcano(results,
                        lab = rownames(results),
                        x = "log2FoldChange",
                        y = "padj",
                        xlab = bquote(~Log[2]~ "fold change"),
                        ylab = bquote(~-Log[10]~adjusted~italic(P)),
                        pCutoff = pvalue,
                        FCcutoff = foldchange,
                        xlim = c(-6,6),
                        transcriptLabSize = 3.0,
                        title = title,
                        colAlpha = 1,
                        legend=c("NS","Log2 FC","Adjusted p-value",
                                 "Adjusted p-value & Log2 FC"),
                        legendPosition = "bottom",
                        legendLabSize = 10,
                        legendIconSize = 3.0,
                        DrawConnectors = TRUE,
                        widthConnectors = 0.5,
                        colConnectors = "black")
    return(p)
  }
  plotCounts_gg <- function(i, dds, intgroup) {
    group <- if (length(intgroup) == 1) {
    colData(dds)[[intgroup]]
    } else if (length(intgroup) == 2) {
    lvls <- as.vector(t(outer(levels(colData(dds)[[intgroup[1]]]),
    levels(colData(dds)[[intgroup[2]]]), function(x,
    y)
    paste(x, y, sep = " : "))))
    droplevels(factor(apply(
    as.data.frame(colData(dds)[,
    intgroup, drop = FALSE]), 1, paste, collapse = " : "
    ),
    levels = lvls))
    } else {
    factor(apply(as.data.frame(colData(dds)[, intgroup, drop = FALSE]),
    1, paste, collapse = " : "))
    }
    data <-
    plotCounts(dds,
    gene = i,
    intgroup = intgroup,
    returnData = TRUE)
    data <- cbind(data, data.frame('group' = group))
    main <- rownames(dds)[i]
    
    ggplot(data, aes(x = group, y = count)) + geom_boxplot() + ylab('Normalized count') + ggtitle(main) + coord_trans(y = "log2") + scale_x_discrete(limits=c("WT","EV","EH","H"))
  }
  #for Kable
  rowlim=10
```

#### Untreated (WT) vs EPO with Vehicle (EV) 

##### Plots {.tabset .tabset-fade .tabset-pills}

###### MA-plot
```{r code block 22, message=FALSE, warning=FALSE}
  res_WT_EV <- results(wald.test, contrast=c("condition","WT","EV"),alpha=p_cutoff,pAdjustMethod="BH")
  res_WT_EV_ashr <- lfcShrink(wald.test, contrast=c("condition","WT","EV"),res=res_WT_EV,type="ashr")
  par(mfrow=c(1,2))
  plotMA(res_WT_EV,xlim=xlim, ylim=ylim,main="normal")
  plotMA(res_WT_EV_ashr,xlim=xlim, ylim=ylim,main="ashr")
  invisible(dev.off())
```

###### Historgram of adjusted p-values
```{r code block 23, message=FALSE, warning=FALSE}
  hist(res_WT_EV$padj, breaks=20, col="grey", main="DESeq2 p-value distribution", xlab="DESeq2 P-value", ylab="Number of genes")
```

###### Volacano-plot
```{r code block 24, message=FALSE, warning=FALSE, fig.width=10, fig.height=12}
  resOrdered_WT_EV <- res_WT_EV[order(res_WT_EV$padj),]
  write.csv(as.data.frame(resOrdered_WT_EV), file="condition_WT_EV_alpha0.1_results.csv")
  WT_EV_p1<-enhanced_volcano_plots("WT vs. EV (padj=0.05,log2fc=0.5)",res_WT_EV,0.05,fc_cutoff)
  WT_EV_p2<-enhanced_volcano_plots("WT vs. EV (padj=0.1,log2fc=0.5)",res_WT_EV,p_cutoff,fc_cutoff)
  grid.arrange(WT_EV_p1, WT_EV_p2, nrow=2, ncol=1)
  grid.rect(gp=gpar(fill=NA))
  invisible(dev.off())
```

###### Plot counts
```{r code block 25, message=FALSE, warning=FALSE}
  nBestFeatures = 20
  ord <- order(res_WT_EV$padj, decreasing = FALSE)
  for (i in head(ord, nBestFeatures)) {
    print(plotCounts_gg(i, dds = wald.test, intgroup = c("condition")))
  }
```

##### Select genes based on FDR and make heatmap

```{r code black 26, message=FALSE,warning=FALSE,fig.width=10, fig.height=12}
  gene.kept <- rownames(res_WT_EV)[res_WT_EV$padj <= p_cutoff & !is.na(res_WT_EV$padj) & ( res_WT_EV$log2FoldChange <= -0.5 | res_WT_EV$log2FoldChange >= 0.5)]
  count.table.kept <- log2(count.table + epsilon)[gene.kept, ]
  heatmap.2(as.matrix(count.table.kept), 
            scale="row", 
            hclust=function(x) hclust(x,method="average"), 
            distfun=function(x) as.dist((1-cor(t(x)))/2), 
            trace="none", 
            density="none", 
            #labRow="",
            cexCol=0.7)
  invisible(dev.off())
```

##### Do functional enrichment

```{r code block 27, warning=FALSE}
  res_WT_EV.df <- na.omit(data.frame(res_WT_EV))
  induced.sign <- rownames(res_WT_EV.df)[res_WT_EV.df$log2FoldChange >= 0.5 &  res_WT_EV.df$padj < p_cutoff]
  # head(induced.sign)
  # names(term.induced)
  if(identical(induced.sign, character(0))){
    cat("No genes found that have induced expression")
  } else {
  term.induced <- gprofiler(query=induced.sign, organism="hsapiens")
  term.induced <- term.induced[order(term.induced$p.value),]
  na.omit(term.induced)
  # term.induced$p.value
  if(nrow(term.induced)>=10){rowlim=10}else{rowlim=nrow(term.induced)}
  kable(term.induced[1:rowlim,c("term.name",
                        "term.size",
                        "query.size",
                        "overlap.size",
                        "recall",
                        "precision",
                        "p.value", 
                        "intersection")], 
        format.args=c(engeneer=TRUE, digits=3), caption="**Table: induced gene functional analysis wit gProfileR. ** ") %>% kable_styling(bootstrap_options = c("striped", "hover", "condensed", "responsive"))
  }
  repressed.sign <- rownames(res_WT_EV.df)[res_WT_EV.df$log2FoldChange <= -0.5 &  res_WT_EV.df$padj < p_cutoff]
  if(identical(repressed.sign, character(0))){
    cat("No genes found that have repressed expression")
  } else {
  term.repressed <- gprofiler(query=repressed.sign, organism="hsapiens")
  term.repressed <- term.repressed[order(term.repressed$p.value),]
  na.omit(term.repressed)
  if(nrow(term.repressed)>=10){rowlim=10}else{rowlim=nrow(term.repressed)}
  kable(term.repressed[1:rowlim,c("term.name",
                        "term.size",
                        "query.size",
                        "overlap.size",
                        "recall",
                        "precision",
                        "p.value", 
                        "intersection")], 
        format.args=c(engeneer=TRUE, digits=3), caption="**Table: repressed genes functional analysis with gProfileR. ** ") %>% kable_styling(bootstrap_options = c("striped", "hover", "condensed", "responsive"))
  }
  #kable(head(term.induced[,c("p.value", "term.name","intersection")], 10))
```

#### Untreated (WT) vs EPO with HMGB1 (EH)

##### Plots {.tabset .tabset-fade .tabset-pills}
###### MA-plot
```{r code block 28, message=FALSE, warning=FALSE}
  res_WT_EH <- results(wald.test, contrast=c("condition","WT","EH"),alpha=p_cutoff,pAdjustMethod="BH")
  res_WT_EH_ashr <- lfcShrink(wald.test, contrast=c("condition","WT","EH"),res=res_WT_EV,type="ashr")
  par(mfrow=c(1,2))
  plotMA(res_WT_EH,xlim=xlim, ylim=ylim,main="normal")
  plotMA(res_WT_EH_ashr,xlim=xlim, ylim=ylim,main="ashr")
  invisible(dev.off())
```

###### Historgram of adjusted p-values
```{r code block 29, message=FALSE, warning=FALSE}
  hist(res_WT_EH$padj, breaks=20, col="grey", main="DESeq2 p-value distribution", xlab="DESeq2 P-value", ylab="Number of genes")
```

###### Volacano-plot
```{r code block 30, message=FALSE, warning=FALSE,fig.width=10, fig.height=12}
  resOrdered_WT_EH <- res_WT_EH[order(res_WT_EH$padj),]
  write.csv(as.data.frame(resOrdered_WT_EH), file="condition_WT_EH_alpha0.1_results.csv")
  WT_EH_p1<-enhanced_volcano_plots("WT vs. EH (padj=0.05,log2fc=0.5)",res_WT_EH,0.05,fc_cutoff)
  WT_EH_p2<-enhanced_volcano_plots("WT vs. EH (padj=0.1,log2fc=0.5)",res_WT_EH,p_cutoff,fc_cutoff)
  grid.arrange(WT_EH_p1, WT_EH_p2, nrow=2, ncol=1)
  grid.rect(gp=gpar(fill=NA))
  invisible(dev.off())
```

###### Plot counts
```{r code block 31, message=FALSE, warning=FALSE}
  nBestFeatures = 20
  ord <- order(res_WT_EH$padj, decreasing = FALSE)
  for (i in head(ord, nBestFeatures)) {
    print(plotCounts_gg(i, dds = wald.test, intgroup = c("condition")))
  }
```

##### Select genes based on FDR and make heatmap

```{r code black 32, message=FALSE,warning=FALSE,fig.width=10, fig.height=12}
  gene.kept <- rownames(res_WT_EH)[res_WT_EH$padj <= p_cutoff & !is.na(res_WT_EH$padj) & ( res_WT_EH$log2FoldChange <= -0.5 | res_WT_EH$log2FoldChange >= 0.5)]
  count.table.kept <- log2(count.table + epsilon)[gene.kept, ]
  heatmap.2(as.matrix(count.table.kept), 
            scale="row", 
            hclust=function(x) hclust(x,method="average"), 
            distfun=function(x) as.dist((1-cor(t(x)))/2), 
            trace="none", 
            density="none", 
            #labRow="",
            cexCol=0.7)
  invisible(dev.off())
```

##### Do functional enrichment

```{r code block 33, warning=FALSE}
  res_WT_EH.df <- na.omit(data.frame(res_WT_EH))
  induced.sign <- rownames(res_WT_EV.df)[res_WT_EH.df$log2FoldChange >= 0.5 &  res_WT_EH.df$padj < p_cutoff]
  # head(induced.sign)
  # names(term.induced)
  if(identical(induced.sign, character(0))){
    cat("No genes found that have induced expression")
  } else {
  term.induced <- gprofiler(query=induced.sign, organism="hsapiens")
  term.induced <- term.induced[order(term.induced$p.value),]
  # term.induced$p.value
  na.omit(term.induced)
  if(nrow(term.induced)>=10){rowlim=10}else{rowlim=nrow(term.induced)}
  kable(term.induced[1:rowlim,c("term.name",
                        "term.size",
                        "query.size",
                        "overlap.size",
                        "recall",
                        "precision",
                        "p.value", 
                        "intersection")], 
        format.args=c(engeneer=TRUE, digits=3), caption="**Table: induced gene functional analysis wit gProfileR. ** ") %>% kable_styling(bootstrap_options = c("striped", "hover", "condensed", "responsive"))
  }
  repressed.sign <- rownames(res_WT_EH.df)[res_WT_EH.df$log2FoldChange <= -0.5 &  res_WT_EH.df$padj < p_cutoff]
  if(identical(repressed.sign, character(0))){
    cat("No genes found that have repressed expression")
  } else {
  term.repressed <- gprofiler(query=repressed.sign, organism="hsapiens")
  term.repressed <- term.repressed[order(term.repressed$p.value),]
  na.omit(term.repressed)
  if(nrow(term.repressed)>=10){rowlim=10}else{rowlim=nrow(term.repressed)}
  kable(term.repressed[1:rowlim,c("term.name",
                        "term.size",
                        "query.size",
                        "overlap.size",
                        "recall",
                        "precision",
                        "p.value", 
                        "intersection")], 
        format.args=c(engeneer=TRUE, digits=3), caption="**Table: repressed genes functional analysis with gProfileR. ** ") %>% kable_styling(bootstrap_options = c("striped", "hover", "condensed", "responsive"))
  }
  #kable(head(term.induced[,c("p.value", "term.name","intersection")], 10))
```

***

#### Untreated (WT) vs HMGB1 (H) 

##### Plots {.tabset .tabset-fade .tabset-pills}

###### MA-plot
```{r code block 34, message=FALSE, warning=FALSE}
  res_WT_H <- results(wald.test, contrast=c("condition","WT","H"),alpha=p_cutoff,pAdjustMethod="BH")
  res_WT_H_ashr <- lfcShrink(wald.test, contrast=c("condition","WT","H"),res=res_WT_H,type="ashr")
  par(mfrow=c(1,2))
  plotMA(res_WT_H,xlim=xlim, ylim=ylim,main="normal")
  plotMA(res_WT_H_ashr,xlim=xlim, ylim=ylim,main="ashr")
  invisible(dev.off())
```

###### Historgram of adjusted p-values
```{r code block 35, message=FALSE, warning=FALSE}
  hist(res_WT_H$padj, breaks=20, col="grey", main="DESeq2 p-value distribution", xlab="DESeq2 P-value", ylab="Number of genes")
```

###### Volacano-plot
```{r code block 36, message=FALSE, warning=FALSE, fig.width=10, fig.height=12}
  resOrdered_WT_H <- res_WT_H[order(res_WT_H$padj),]
  write.csv(as.data.frame(resOrdered_WT_H), file="condition_WT_H_alpha0.1_results.csv")
  WT_H_p1<-enhanced_volcano_plots("WT vs. H (padj=0.05,log2fc=0.5)",res_WT_H,0.05,fc_cutoff)
  WT_H_p2<-enhanced_volcano_plots("WT vs. H (padj=0.1,log2fc=0.5)",res_WT_H,p_cutoff,fc_cutoff)
  grid.arrange(WT_H_p1, WT_H_p2, nrow=2, ncol=1)
  grid.rect(gp=gpar(fill=NA))
  invisible(dev.off())
```

###### Plot counts
```{r code block 37, message=FALSE, warning=FALSE}
  nBestFeatures = 20
  ord <- order(res_WT_H$padj, decreasing = FALSE)
  for (i in head(ord, nBestFeatures)) {
    print(plotCounts_gg(i, dds = wald.test, intgroup = c("condition")))
  }
```

##### Select genes based on FDR and make heatmap

```{r code black 38, message=FALSE,warning=FALSE,fig.width=10, fig.height=12}
  gene.kept <- rownames(res_WT_H)[res_WT_H$padj <= p_cutoff & !is.na(res_WT_H$padj) & ( res_WT_H$log2FoldChange <= -0.5 | res_WT_H$log2FoldChange >= 0.5)]
  count.table.kept <- log2(count.table + epsilon)[gene.kept, ]
  heatmap.2(as.matrix(count.table.kept), 
            scale="row", 
            hclust=function(x) hclust(x,method="average"), 
            distfun=function(x) as.dist((1-cor(t(x)))/2), 
            trace="none", 
            density="none", 
            #labRow="",
            cexCol=0.7)
  invisible(dev.off())
```

##### Do functional enrichment

```{r code block 39, warning=FALSE}
  res_WT_H.df <- na.omit(data.frame(res_WT_H))
  induced.sign <- rownames(res_WT_H.df)[res_WT_H.df$log2FoldChange >= 0.5 &  res_WT_H.df$padj < p_cutoff]
  # head(induced.sign)
  # names(term.induced)
  if(identical(induced.sign, character(0))){
    cat("No genes found that have induced expression")
  } else {
  term.induced <- gprofiler(query=induced.sign, organism="hsapiens")
  term.induced <- term.induced[order(term.induced$p.value),]
  na.omit(term.induced)
  if(nrow(term.induced)>=10){rowlim=10}else{rowlim=nrow(term.induced)}
  # term.induced$p.value
  if(length(term.induced$term.name) == 0){
    cat("No terms found that have induced expression using the genelist")
  }
  else{
  kable(term.induced[1:rowlim,c("term.name",
                        "term.size",
                        "query.size",
                        "overlap.size",
                        "recall",
                        "precision",
                        "p.value", 
                        "intersection")], 
        format.args=c(engeneer=TRUE, digits=3), caption="**Table: induced gene functional analysis with gProfileR. ** ") %>% kable_styling(bootstrap_options = c("striped", "hover", "condensed", "responsive"))
  }
  }
  repressed.sign <- rownames(res_WT_H.df)[res_WT_H.df$log2FoldChange <= -0.5 &  res_WT_H.df$padj < p_cutoff]
  if(identical(repressed.sign, character(0))){
    cat("No genes found that have repressed expression")
  } else {
  term.repressed <- gprofiler(query=repressed.sign, organism="hsapiens")
  term.repressed <- term.repressed[order(term.repressed$p.value),]
  na.omit(term.repressed)
  if(nrow(term.repressed)>=10){rowlim=10}else{rowlim=nrow(term.repressed)}
  kable(term.repressed[1:rowlim,c("term.name",
                        "term.size",
                        "query.size",
                        "overlap.size",
                        "recall",
                        "precision",
                        "p.value", 
                        "intersection")], 
        format.args=c(engeneer=TRUE, digits=3), caption="**Table: repressed genes functional analysis with gProfileR. ** ") %>% kable_styling(bootstrap_options = c("striped", "hover", "condensed", "responsive"))
  }
  #kable(head(term.induced[,c("p.value", "term.name","intersection")], 10))
```

#### EPO with Vehicle (EV) vs EPO with HMGB1 (EH) 

##### Plots {.tabset .tabset-fade .tabset-pills}

###### MA-plot
```{r code block 40, message=FALSE, warning=FALSE}
  res_EV_EH <- results(wald.test, contrast=c("condition","EV","EH"),alpha=p_cutoff,pAdjustMethod="BH")
  res_EV_EH_ashr <- lfcShrink(wald.test, contrast=c("condition","EV","EH"),res=res_EV_EH,type="ashr")
  par(mfrow=c(1,2))
  plotMA(res_EV_EH,xlim=xlim, ylim=ylim,main="normal")
  plotMA(res_EV_EH_ashr,xlim=xlim, ylim=ylim,main="ashr")
  invisible(dev.off())
```

###### Historgram of adjusted p-values
```{r code block 41, message=FALSE, warning=FALSE}
  hist(res_EV_EH$padj, breaks=20, col="grey", main="DESeq2 p-value distribution", xlab="DESeq2 P-value", ylab="Number of genes")
```

###### Volacano-plot
```{r code block 42, message=FALSE, warning=FALSE, fig.width=10, fig.height=12}
  resOrdered_EV_EH <- res_EV_EH[order(res_EV_EH$padj),]
  write.csv(as.data.frame(resOrdered_EV_EH), file="condition_EV_EH_alpha0.1_results.csv")
  EV_EH_p1<-enhanced_volcano_plots("EV vs. EH (padj=0.05,log2fc=0.5)",res_EV_EH,0.05,fc_cutoff)
  EV_EH_p2<-enhanced_volcano_plots("EV vs. EH (padj=0.1,log2fc=0.5)",res_EV_EH,p_cutoff,fc_cutoff)
  grid.arrange(EV_EH_p1, EV_EH_p2, nrow=2, ncol=1)
  grid.rect(gp=gpar(fill=NA))
  invisible(dev.off())
```

###### Plot counts
```{r code block 43, message=FALSE, warning=FALSE}
  nBestFeatures = 20
  ord <- order(res_EV_EH$padj, decreasing = FALSE)
  for (i in head(ord, nBestFeatures)) {
    print(plotCounts_gg(i, dds = wald.test, intgroup = c("condition")))
  }
```

##### Select genes based on FDR and make heatmap

```{r code black 44, message=FALSE,warning=FALSE,fig.width=10, fig.height=12}
  gene.kept <- rownames(res_EV_EH)[res_EV_EH$padj <= p_cutoff & !is.na(res_EV_EH$padj) & ( res_EV_EH$log2FoldChange <= -0.5 | res_EV_EH$log2FoldChange >= 0.5)]
  count.table.kept <- log2(count.table + epsilon)[gene.kept, ]
  heatmap.2(as.matrix(count.table.kept), 
            scale="row", 
            hclust=function(x) hclust(x,method="average"), 
            distfun=function(x) as.dist((1-cor(t(x)))/2), 
            trace="none", 
            density="none", 
            #labRow="",
            cexCol=0.7)
  invisible(dev.off())
```

##### Do functional enrichment

```{r code block 45, warning=FALSE}
  res_EV_EH.df <- na.omit(data.frame(res_EV_EH))
  induced.sign <- rownames(res_EV_EH.df)[res_EV_EH.df$log2FoldChange >= 0.5 &  res_EV_EH.df$padj < p_cutoff]
  # head(induced.sign)
  # names(term.induced)
  if(identical(induced.sign, character(0))){
    cat("No genes found that have induced expression")
  } else {
  term.induced <- gprofiler(query=induced.sign, organism="hsapiens")
  term.induced <- term.induced[order(term.induced$p.value),]
  na.omit(term.induced)
  if(nrow(term.induced)>=10){rowlim=10}else{rowlim=nrow(term.induced)}
  # term.induced$p.value
  kable(term.induced[1:rowlim,c("term.name",
                        "term.size",
                        "query.size",
                        "overlap.size",
                        "recall",
                        "precision",
                        "p.value", 
                        "intersection")], 
        format.args=c(engeneer=TRUE, digits=3), caption="**Table: induced gene functional analysis with gProfileR. ** ") %>% kable_styling(bootstrap_options = c("striped", "hover", "condensed", "responsive"))
  }
  repressed.sign <- rownames(res_EV_EH.df)[res_EV_EH.df$log2FoldChange <= -0.5 &  res_EV_EH.df$padj < p_cutoff]
  if(identical(repressed.sign, character(0))){
    cat("No genes found that have repressed expression")
  } else {
  term.repressed <- gprofiler(query=repressed.sign, organism="hsapiens")
  term.repressed <- term.repressed[order(term.repressed$p.value),]
  na.omit(term.repressed)
  if(nrow(term.repressed)>=10){rowlim=10}else{rowlim=nrow(term.repressed)}
  kable(term.repressed[1:rowlim,c("term.name",
                        "term.size",
                        "query.size",
                        "overlap.size",
                        "recall",
                        "precision",
                        "p.value", 
                        "intersection")], 
        format.args=c(engeneer=TRUE, digits=3), caption="**Table: repressed genes functional analysis with gProfileR. ** ") %>% kable_styling(bootstrap_options = c("striped", "hover", "condensed", "responsive"),full_width = FALSE)
  }
  #kable(head(term.induced[,c("p.value", "term.name","intersection")], 10))
```

#### EPO with Vehicle (EV) vs HMGB1 (H) 

##### Plots {.tabset .tabset-fade .tabset-pills}

###### MA-plot
```{r code block 46, message=FALSE, warning=FALSE}
  res_EV_H <- results(wald.test, contrast=c("condition","EV","H"),alpha=p_cutoff,pAdjustMethod="BH")
  res_EV_H_ashr <- lfcShrink(wald.test, contrast=c("condition","EV","H"),res=res_EV_H,type="ashr")
  par(mfrow=c(1,2))
  plotMA(res_EV_H,xlim=xlim, ylim=ylim,main="normal")
  plotMA(res_EV_H_ashr,xlim=xlim, ylim=ylim,main="ashr")
  invisible(dev.off())
```

###### Historgram of adjusted p-values
```{r code block 47, message=FALSE, warning=FALSE}
  hist(res_EV_H$padj, breaks=20, col="grey", main="DESeq2 p-value distribution", xlab="DESeq2 P-value", ylab="Number of genes")
```

###### Volacano-plot
```{r code block 48, message=FALSE, warning=FALSE, fig.width=10, fig.height=12}
  resOrdered_EV_H <- res_EV_H[order(res_EV_H$padj),]
  write.csv(as.data.frame(resOrdered_EV_H), file="condition_EV_H_alpha0.1_results.csv")
  EV_H_p1<-enhanced_volcano_plots("EV vs. H (padj=0.05,log2fc=0.5)",res_EV_H,0.05,fc_cutoff)
  EV_H_p2<-enhanced_volcano_plots("EV vs. H (padj=0.1,log2fc=0.5)",res_EV_H,p_cutoff,fc_cutoff)
  grid.arrange(EV_H_p1, EV_H_p2, nrow=2, ncol=1)
  grid.rect(gp=gpar(fill=NA))
  invisible(dev.off())
```

###### Plot counts
```{r code block 49, message=FALSE, warning=FALSE}
  nBestFeatures = 20
  ord <- order(res_EV_H$padj, decreasing = FALSE)
  for (i in head(ord, nBestFeatures)) {
    print(plotCounts_gg(i, dds = wald.test, intgroup = c("condition")))
  }
```

##### Select genes based on FDR and make heatmap

```{r code black 50, message=FALSE,warning=FALSE,fig.width=10, fig.height=12}
  gene.kept <- rownames(res_EV_H)[res_EV_H$padj <= p_cutoff & !is.na(res_EV_H$padj) & ( res_EV_H$log2FoldChange <= -0.5 | res_EV_H$log2FoldChange >= 0.5)]
  count.table.kept <- log2(count.table + epsilon)[gene.kept, ]
  heatmap.2(as.matrix(count.table.kept), 
            scale="row", 
            hclust=function(x) hclust(x,method="average"), 
            distfun=function(x) as.dist((1-cor(t(x)))/2), 
            trace="none", 
            density="none", 
            #labRow="",
            cexCol=0.7)
  invisible(dev.off())
```

##### Do functional enrichment

```{r code block 51, warning=FALSE}
  res_EV_H.df <- na.omit(data.frame(res_EV_H))
  induced.sign <- rownames(res_EV_H.df)[res_EV_H.df$log2FoldChange >= 0.5 &  res_EV_H.df$padj < p_cutoff]
  # head(induced.sign)
  # names(term.induced)
  if(identical(induced.sign, character(0))){
    cat("No genes found that have induced expression")
  } else {
  term.induced <- gprofiler(query=induced.sign, organism="hsapiens")
  term.induced <- term.induced[order(term.induced$p.value),]
  # term.induced$p.value
  na.omit(term.induced)
  if(nrow(term.induced)>=10){rowlim=10}else{rowlim=nrow(term.induced)}
  kable(term.induced[1:rowlim,c("term.name",
                        "term.size",
                        "query.size",
                        "overlap.size",
                        "recall",
                        "precision",
                        "p.value", 
                        "intersection")], 
        format.args=c(engeneer=TRUE, digits=3), caption="**Table: induced gene functional analysis with gProfileR. ** ") %>% kable_styling(bootstrap_options = c("striped", "hover", "condensed", "responsive"))
  }
  repressed.sign <- rownames(res_EV_H.df)[res_EV_H.df$log2FoldChange <= -0.5 &  res_EV_H.df$padj < p_cutoff]
  if(identical(repressed.sign, character(0))){
    cat("No genes found that have repressed expression")
  } else {
  term.repressed <- gprofiler(query=repressed.sign, organism="hsapiens")
  term.repressed <- term.repressed[order(term.repressed$p.value),]
  na.omit(term.repressed)
  if(nrow(term.repressed)>=10){rowlim=10}else{rowlim=nrow(term.repressed)}
  kable(term.repressed[1:rowlim,c("term.name",
                        "term.size",
                        "query.size",
                        "overlap.size",
                        "recall",
                        "precision",
                        "p.value", 
                        "intersection")], 
        format.args=c(engeneer=TRUE, digits=3), caption="**Table: repressed genes functional analysis with gProfileR. ** ") %>% kable_styling(bootstrap_options = c("striped", "hover", "condensed", "responsive"))
  }
  #kable(head(term.induced[,c("p.value", "term.name","intersection")], 10))
```

#### EPO with HMGB1 (EH) vs HMGB1 (H) 

##### Plots {.tabset .tabset-fade .tabset-pills}

###### MA-plot
```{r code block 52, message=FALSE, warning=FALSE}
  res_EH_H <- results(wald.test, contrast=c("condition","EH","H"),alpha=p_cutoff,pAdjustMethod="BH")
  res_EH_H_ashr <- lfcShrink(wald.test, contrast=c("condition","EH","H"),res=res_EH_H,type="ashr")
  par(mfrow=c(1,2))
  plotMA(res_EH_H,xlim=xlim, ylim=ylim,main="normal")
  plotMA(res_EH_H_ashr,xlim=xlim, ylim=ylim,main="ashr")
  invisible(dev.off())
```

###### Historgram of adjusted p-values
```{r code block 53, message=FALSE, warning=FALSE}
  hist(res_EH_H$padj, breaks=20, col="grey", main="DESeq2 p-value distribution", xlab="DESeq2 P-value", ylab="Number of genes")
```

###### Volacano-plot
```{r code block 54, message=FALSE, warning=FALSE, fig.width=10, fig.height=12}
  resOrdered_EH_H <- res_EH_H[order(res_EH_H$padj),]
  write.csv(as.data.frame(resOrdered_EH_H), file="condition_EH_H_alpha0.1_results.csv")
  EH_H_p1<-enhanced_volcano_plots("EH vs. H (padj=0.05,log2fc=0.5)",res_EH_H,0.05,fc_cutoff)
  EH_H_p2<-enhanced_volcano_plots("EH vs. H (padj=0.1,log2fc=0.5)",res_EH_H,p_cutoff,fc_cutoff)
  grid.arrange(EH_H_p1, EH_H_p2, nrow=2, ncol=1)
  grid.rect(gp=gpar(fill=NA))
  invisible(dev.off())
```

###### Plot counts
```{r code block 55, message=FALSE, warning=FALSE}
  nBestFeatures = 20
  ord <- order(res_EH_H$padj, decreasing = FALSE)
  for (i in head(ord, nBestFeatures)) {
    print(plotCounts_gg(i, dds = wald.test, intgroup = c("condition")))
  }
```

##### Select genes based on FDR and make heatmap

```{r code black 56, message=FALSE,warning=FALSE,fig.width=10, fig.height=12}
  gene.kept <- rownames(res_EH_H)[res_EH_H$padj <= p_cutoff & !is.na(res_EH_H$padj) & ( res_EH_H$log2FoldChange <= -0.5 | res_EH_H$log2FoldChange >= 0.5)]
  count.table.kept <- log2(count.table + epsilon)[gene.kept, ]
  heatmap.2(as.matrix(count.table.kept), 
            scale="row", 
            hclust=function(x) hclust(x,method="average"), 
            distfun=function(x) as.dist((1-cor(t(x)))/2), 
            trace="none", 
            density="none", 
            #labRow="",
            cexCol=0.7)
  invisible(dev.off())
```

##### Do functional enrichment

```{r code block 57, warning=FALSE}
  res_EH_H.df <- na.omit(data.frame(res_EH_H))
  induced.sign <- rownames(res_EH_H.df)[res_EH_H.df$log2FoldChange >= 0.5 &  res_EH_H.df$padj < p_cutoff]
  # head(induced.sign)
  # names(term.induced)
  if(identical(induced.sign, character(0))){
    cat("No genes found that have induced expression")
  } else {
  term.induced <- gprofiler(query=induced.sign, organism="hsapiens")
  term.induced <- term.induced[order(term.induced$p.value),]
  # term.induced$p.value
  na.omit(term.induced)
  if(nrow(term.induced)>=10){rowlim=10}else{rowlim=nrow(term.induced)}
  kable(term.induced[1:rowlim,c("term.name",
                        "term.size",
                        "query.size",
                        "overlap.size",
                        "recall",
                        "precision",
                        "p.value", 
                        "intersection")], 
        format.args=c(engeneer=TRUE, digits=3), caption="**Table: induced gene functional analysis with gProfileR. ** ") %>% kable_styling(bootstrap_options = c("striped", "hover", "condensed", "responsive"))
  }
  repressed.sign <- rownames(res_EH_H.df)[res_EH_H.df$log2FoldChange <= -0.5 &  res_EH_H.df$padj < p_cutoff]
  if(identical(repressed.sign, character(0))){
    cat("No genes found that have repressed expression")
  } else {
  term.repressed <- gprofiler(query=repressed.sign, organism="hsapiens")
  term.repressed <- term.repressed[order(term.repressed$p.value),]
  na.omit(term.repressed)
  if(nrow(term.repressed)>=10){rowlim=10}else{rowlim=nrow(term.repressed)}
  kable(term.repressed[1:rowlim,c("term.name",
                        "term.size",
                        "query.size",
                        "overlap.size",
                        "recall",
                        "precision",
                        "p.value", 
                        "intersection")], 
        format.args=c(engeneer=TRUE, digits=3), caption="**Table: repressed genes functional analysis with gProfileR. ** ") %>% kable_styling(bootstrap_options = c("striped", "hover", "condensed", "responsive"))
  }
  #kable(head(term.induced[,c("p.value", "term.name","intersection")], 10))
```
